<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.537">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Emil Johansson">

<title>Group 2 report: Characterization of Alzheimer’s disease</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="group2_final_files/libs/clipboard/clipboard.min.js"></script>
<script src="group2_final_files/libs/quarto-html/quarto.js"></script>
<script src="group2_final_files/libs/quarto-html/popper.min.js"></script>
<script src="group2_final_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="group2_final_files/libs/quarto-html/anchor.min.js"></script>
<link href="group2_final_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="group2_final_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="group2_final_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="group2_final_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="group2_final_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Group 2 report: Characterization of Alzheimer’s disease</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">A project for the course Statistics in Bioinformatics</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Emil Johansson </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">Friday the 29th of November 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#qc" id="toc-qc" class="nav-link" data-scroll-target="#qc"><span class="header-section-number">2</span> QC</a></li>
  <li><a href="#differential-expression-analysis" id="toc-differential-expression-analysis" class="nav-link" data-scroll-target="#differential-expression-analysis"><span class="header-section-number">3</span> Differential expression analysis</a>
  <ul>
  <li><a href="#design-matrix" id="toc-design-matrix" class="nav-link" data-scroll-target="#design-matrix"><span class="header-section-number">3.1</span> Design matrix</a></li>
  <li><a href="#hypothesis-correction" id="toc-hypothesis-correction" class="nav-link" data-scroll-target="#hypothesis-correction"><span class="header-section-number">3.2</span> Hypothesis correction</a></li>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot"><span class="header-section-number">3.3</span> Volcano plot</a></li>
  </ul></li>
  <li><a href="#protein-protein-interaction-analysis" id="toc-protein-protein-interaction-analysis" class="nav-link" data-scroll-target="#protein-protein-interaction-analysis"><span class="header-section-number">4</span> Protein-protein interaction analysis</a></li>
  <li><a href="#enrichment-analysis" id="toc-enrichment-analysis" class="nav-link" data-scroll-target="#enrichment-analysis"><span class="header-section-number">5</span> Enrichment analysis</a></li>
  <li><a href="#random-forest-classification" id="toc-random-forest-classification" class="nav-link" data-scroll-target="#random-forest-classification"><span class="header-section-number">6</span> Random forest classification</a>
  <ul>
  <li><a href="#prepare-data" id="toc-prepare-data" class="nav-link" data-scroll-target="#prepare-data"><span class="header-section-number">6.1</span> Prepare data</a></li>
  <li><a href="#performance-evaluation" id="toc-performance-evaluation" class="nav-link" data-scroll-target="#performance-evaluation"><span class="header-section-number">6.2</span> Performance evaluation</a></li>
  </ul></li>
  <li><a href="#spearman-correlation-analysis" id="toc-spearman-correlation-analysis" class="nav-link" data-scroll-target="#spearman-correlation-analysis"><span class="header-section-number">7</span> Spearman correlation analysis</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">8</span> Session info</a></li>
  </ul>
</nav>
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Libraries we need</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>libs <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tidyverse"</span>, <span class="st">"readxl"</span>, <span class="st">"limma"</span>, <span class="st">"ggrepel"</span>, <span class="st">"magrittr"</span>, <span class="st">"kableExtra"</span>,</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"patchwork"</span>, <span class="st">"DT"</span>, <span class="st">"tidymodels"</span>, <span class="st">"ggbeeswarm"</span>, <span class="st">"gt"</span>, <span class="st">"skimr"</span>, <span class="st">"GGally"</span>,</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"visdat"</span>, <span class="st">"corrr"</span>, <span class="st">"ggsignif"</span>, <span class="st">"vip"</span>, <span class="st">"themis"</span>, <span class="st">"keras"</span>, <span class="st">"xgboost"</span>, <span class="st">"kknn"</span>,</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tensorflow"</span>, <span class="st">"xlsx"</span>, <span class="st">"HDAnalyzeR"</span>, <span class="st">"plotly"</span>, <span class="st">"umap"</span>, <span class="st">"uwot"</span>, <span class="st">"ggplotify"</span>,</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cowplot"</span>, <span class="st">"ggvenn"</span>, <span class="st">"UpSetR"</span>, <span class="st">"ComplexUpset"</span>, <span class="st">"DESeq2"</span>, <span class="st">"pheatmap"</span>, <span class="st">"randomForest"</span>,</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"caret"</span>, <span class="st">"edgeR"</span>, <span class="st">"STRINGdb"</span>, <span class="st">"igraph"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"enrichplot"</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Install missing libraries</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>installed_libs <span class="ot">&lt;-</span> libs <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(installed_libs <span class="sc">==</span> <span class="cn">FALSE</span>)) {</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(libs[<span class="sc">!</span>installed_libs])</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(libs, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="qc" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> QC</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(STRINGdb)</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene expression count data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>count_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">matches</span>(<span class="st">"_raw$"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Round counts to integers</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"C"</span>), <span class="fu">starts_with</span>(<span class="st">"A"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span><span class="fu">round</span>(.))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>genes_before_filtering <span class="ot">&lt;-</span> <span class="fu">nrow</span>(count_data_filtered)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of genes before filtering:"</span>, genes_before_filtering, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of genes before filtering: 19185 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> count_data_filtered)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(dge) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meanLog2CPM, <span class="at">main =</span> <span class="st">"Histogram of Mean Log2 CPM"</span>, <span class="at">xlab =</span> <span class="st">"Mean Log2 CPM"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/filter%20low%20counts%20using%20mean%20log2%20CPM-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data_filtered[meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span>, ]</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>genes_after_filtering <span class="ot">&lt;-</span> <span class="fu">nrow</span>(count_data_filtered)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of genes after filtering:"</span>, genes_after_filtering, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of genes after filtering: 14950 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(count_data_filtered),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^C"</span>, <span class="fu">colnames</span>(count_data_filtered)), <span class="st">"HC"</span>, <span class="st">"AD"</span>),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>)  <span class="co"># Manually adding batch information</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> count_data_filtered)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># design &lt;- model.matrix(~ condition + batch, data = sample_conditions)</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition <span class="sc">+</span> <span class="fu">factor</span>(batch), <span class="at">data =</span> sample_conditions)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, design, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/normalization%20with%20Voom-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(v<span class="sc">$</span>E)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     C1_raw   C2_raw   C3_raw   C4_raw   C5_raw   C6_raw   C7_raw   C8_raw
1  1.666028 3.622681 2.402353 2.242940 2.175301 1.945160 2.220551 2.047241
2  8.171797 7.319737 8.634353 8.026213 9.276020 8.897405 7.848728 7.295732
12 6.987940 2.022997 3.397751 3.657259 6.019456 7.878962 2.560113 3.316176
14 6.421959 6.029584 6.000448 6.510198 5.319996 6.775631 6.584768 6.530001
16 8.762372 8.293814 8.435144 8.724345 7.865274 8.882593 8.895612 8.728100
18 8.144870 7.275008 8.455469 8.478941 8.341384 8.366975 8.453630 8.680521
     A9_raw  A10_raw  A11_raw  A12_raw  A13_raw  A14_raw  A15_raw  A16_raw
1  1.906348 1.627370 2.073706 2.429696 2.914332 2.243341 2.639876 2.290996
2  8.228932 8.609914 7.917652 8.579893 8.510049 8.863482 8.725027 8.051962
12 4.499693 5.975793 3.873325 6.067364 7.633415 7.161695 5.411179 8.052767
14 6.392968 6.333542 6.492932 6.165991 5.714636 6.341330 6.336129 6.563178
16 8.703734 8.418269 8.892265 8.288686 8.648926 8.520732 8.270540 8.844795
18 8.301662 8.294551 8.288104 8.189206 8.289339 8.255371 8.281214 8.574186
    A17_raw
1  2.772543
2  8.536880
12 5.083795
14 6.130582
16 8.484317
18 8.402370</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(v<span class="sc">$</span>E, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">main =</span> <span class="st">"Distribution of Voom Normalized Counts"</span>, <span class="at">xlab =</span> <span class="st">"Voom Normalized Counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/normalization%20with%20Voom-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dds &lt;- DESeqDataSetFromMatrix(countData = count_data_filtered, </span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                               colData = sample_conditions, </span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                               design = ~ condition + factor(batch))</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize counts using variance stabilizing transformation (vst)</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># normCounts &lt;- vst(dds, blind = TRUE)</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co"># assay(normCounts)[1:5, 1:5]</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(assay(normCounts), breaks = 50, main = "Distribution of VST Normalized Counts", xlab = "Normalized Counts")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(v<span class="sc">$</span>E)  <span class="co"># Convert v$E to a data frame</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>voom_normalized<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(voom_normalized)  <span class="co"># Add GeneID as a column</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>gene_info_voom <span class="ot">&lt;-</span> <span class="fu">merge</span>(voom_normalized, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>gene_info_voom <span class="ot">&lt;-</span> gene_info_voom[, <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>, <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_info_voom), <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>)))]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(gene_info_voom, file = "voom_normalized.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sampleDist &lt;- cor(assay(normCounts), method = "spearman")</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">cor</span>(v<span class="sc">$</span>E, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>sampleColor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Accent"</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sampleColor) <span class="ot">&lt;-</span> <span class="fu">unique</span>(sample_conditions<span class="sc">$</span>condition)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDist,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist),</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist),</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Condition =</span> sample_conditions<span class="sc">$</span>condition, <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample),</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> <span class="fu">list</span>(<span class="at">Storage =</span> sampleColor),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Sample Correlation Heatmap by Condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20condition%20heatmap-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pcaRes &lt;- prcomp(t(assay(normCounts)))</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>pcaRes <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(v<span class="sc">$</span>E))</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>varExp <span class="ot">&lt;-</span> <span class="fu">round</span>(pcaRes<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pcaRes<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>pcaDF <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pcaRes<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pcaRes<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">Condition =</span> sample_conditions<span class="sc">$</span>condition,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sample =</span> sample_conditions<span class="sc">$</span>sample</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>pcaPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> pcaDF,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Condition, <span class="at">label =</span> Sample)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, varExp[<span class="dv">1</span>], <span class="st">" %)"</span>) ,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, varExp[<span class="dv">2</span>], <span class="st">" %)"</span>) ,</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Plot of Samples"</span>) <span class="sc">+</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Accent"</span>)) <span class="sc">+</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>)  <span class="co"># Added fixed ratio to ensure equal scaling between PC1 and PC2</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pcaPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20PCA%20by%20condition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a heatmap showing Batch 1 and Batch 2 samples separately</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>sampleDist_batch <span class="ot">&lt;-</span> <span class="fu">cor</span>(v<span class="sc">$</span>E, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>batchColor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Set2"</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(batchColor) <span class="ot">&lt;-</span> <span class="fu">unique</span>(sample_conditions<span class="sc">$</span>batch)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDist_batch,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist_batch),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist_batch),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Batch =</span> <span class="fu">factor</span>(sample_conditions<span class="sc">$</span>batch), <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> <span class="fu">list</span>(<span class="at">Batch =</span> batchColor),</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Sample Correlation Heatmap by Batch"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20batch%20heatmap-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA Plot by Batch</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>pcaRes_batch <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(v<span class="sc">$</span>E))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>varExp_batch <span class="ot">&lt;-</span> <span class="fu">round</span>(pcaRes_batch<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pcaRes_batch<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>pcaDF_batch <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pcaRes_batch<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pcaRes_batch<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">Batch =</span> <span class="fu">factor</span>(sample_conditions<span class="sc">$</span>batch),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sample =</span> sample_conditions<span class="sc">$</span>sample</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>pcaPlot_batch <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> pcaDF_batch,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Batch, <span class="at">label =</span> Sample)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, varExp_batch[<span class="dv">1</span>], <span class="st">" %)"</span>),</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, varExp_batch[<span class="dv">2</span>], <span class="st">" %)"</span>),</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Plot of Samples by Batch"</span>) <span class="sc">+</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Set2"</span>)) <span class="sc">+</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pcaPlot_batch)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20PCA%20by%20Batch-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove identified outliers</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co"># outliers &lt;- c("C1_raw", "C2_raw", "C4_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"C1_raw"</span>, <span class="st">"C5_raw"</span>, <span class="st">"C6_raw"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>remaining_samples <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">colnames</span>(count_data_filtered), outliers)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data_filtered[, remaining_samples]</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> sample_conditions[<span class="sc">!</span>sample_conditions<span class="sc">$</span>sample <span class="sc">%in%</span> outliers, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="differential-expression-analysis" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Differential expression analysis</h1>
<section id="design-matrix" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="design-matrix"><span class="header-section-number">3.1</span> Design matrix</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define design matrix including batch effect</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co"># designMatrix &lt;- model.matrix(~ 0 + condition + batch, data = sample_conditions)</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>designMatrix <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> condition <span class="sc">+</span> <span class="fu">factor</span>(batch), <span class="at">data =</span> sample_conditions)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(designMatrix) <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">colnames</span>(designMatrix))  <span class="co"># Ensure valid column names for contrast matrix</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Design Matrix with Batch Effect:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Design Matrix with Batch Effect:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(designMatrix))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  conditionAD conditionHC factor.batch.1
2           0           1              0
3           0           1              0
4           0           1              1
7           0           1              1
8           0           1              1
9           1           0              0</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define contrast matrix</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>contrastMatrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">AD_vs_HC =</span> conditionAD <span class="sc">-</span> conditionHC, <span class="at">levels =</span> designMatrix)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Contrast Matrix:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Contrast Matrix:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(contrastMatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                Contrasts
Levels           AD_vs_HC
  conditionAD           1
  conditionHC          -1
  factor.batch.1        0</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DGEList and estimate dispersion</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(count_data_filtered)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, designMatrix, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model using likelihood ratio test (LRT), now including batch in the design</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, designMatrix)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast =</span> contrastMatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="hypothesis-correction" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="hypothesis-correction"><span class="header-section-number">3.2</span> Hypothesis correction</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrt, <span class="at">n =</span> <span class="fu">nrow</span>(count_data_filtered))</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res<span class="sc">$</span>table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Summary of p-values:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Summary of p-values:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(res<span class="sc">$</span>table<span class="sc">$</span>PValue))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.1672  0.4128  0.4423  0.7062  1.0000 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Summary of FDRs:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Summary of FDRs:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(res<span class="sc">$</span>table<span class="sc">$</span>FDR))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.0000056 0.6683192 0.8253958 0.7729789 0.9415187 1.0000000 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>sigRes <span class="ot">&lt;-</span> <span class="fu">subset</span>(res<span class="sc">$</span>table, FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of significant genes:"</span>, <span class="fu">nrow</span>(sigRes), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of significant genes: 63 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(sigRes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">logFC</th>
<th style="text-align: right;">logCPM</th>
<th style="text-align: right;">LR</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1356</td>
<td style="text-align: right;">2.029851</td>
<td style="text-align: right;">5.348685</td>
<td style="text-align: right;">39.25040</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000056</td>
</tr>
<tr class="even">
<td style="text-align: left;">960</td>
<td style="text-align: right;">2.024781</td>
<td style="text-align: right;">5.760368</td>
<td style="text-align: right;">32.46153</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000909</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5920</td>
<td style="text-align: right;">1.133116</td>
<td style="text-align: right;">3.334786</td>
<td style="text-align: right;">28.76463</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="even">
<td style="text-align: left;">10437</td>
<td style="text-align: right;">1.655241</td>
<td style="text-align: right;">2.208464</td>
<td style="text-align: right;">28.46395</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7098</td>
<td style="text-align: right;">1.032912</td>
<td style="text-align: right;">1.682106</td>
<td style="text-align: right;">27.89196</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="even">
<td style="text-align: left;">929</td>
<td style="text-align: right;">1.585105</td>
<td style="text-align: right;">3.429787</td>
<td style="text-align: right;">27.75458</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>sigRes<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sigRes)  <span class="co"># Add GeneID as a column to the results</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">merge</span>(sigRes, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> gene_info[, <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>, <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_info), <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>)))]</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(gene_info, file = "sig_genes.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="volcano-plot" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="volcano-plot"><span class="header-section-number">3.3</span> Volcano plot</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols to the results</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>table<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res<span class="sc">$</span>table)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>gene_info_all <span class="ot">&lt;-</span> <span class="fu">merge</span>(res<span class="sc">$</span>table, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano plot using gene symbols for all genes</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>volcanoPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gene_info_all, <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR),</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">color =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">"darkred"</span>, <span class="st">"grey"</span>),</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>, GeneSymbol, <span class="st">""</span>))) <span class="sc">+</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="st">"Fold Change, Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">""</span>)) <span class="sc">+</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="st">"Adjusted P value, Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">""</span>)) <span class="sc">+</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkred"</span>, <span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">max.overlaps =</span> <span class="dv">10</span>)</span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcanoPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/visualize%20results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="protein-protein-interaction-analysis" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Protein-protein interaction analysis</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize STRINGdb</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>(<span class="at">version =</span> <span class="st">"11"</span>, <span class="at">species =</span> <span class="dv">9606</span>, <span class="at">score_threshold =</span> <span class="dv">400</span>, <span class="at">input_directory =</span> <span class="st">""</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> gene_info<span class="sc">$</span>GeneSymbol</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">GeneSymbol =</span> gene_symbols, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>mapped_genes <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(gene_symbols, <span class="st">"GeneSymbol"</span>, <span class="at">removeUnmappedRows =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Warning:  we couldn't map to STRING 1% of your identifiers</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get PPI Network for the mapped genes</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>ppi_network <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_interactions</span>(mapped_genes<span class="sc">$</span>STRING_id)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>string_db<span class="sc">$</span><span class="fu">plot_network</span>(mapped_genes<span class="sc">$</span>STRING_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Parameter add_link not available in version 11.0 (please use 11.0b or later)"</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/protein-protein%20interation%20analysis-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>gene_symbols_vector <span class="ot">&lt;-</span> gene_symbols<span class="sc">$</span>GeneSymbol  <span class="co"># Extract the gene symbols as a vector</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>mapped_genes_entrez <span class="ot">&lt;-</span> <span class="fu">bitr</span>(gene_symbols_vector,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fromType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">toType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>goSEA <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> mapped_genes_entrez<span class="sc">$</span>ENTREZID,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">keyType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="enrichment-analysis" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Enrichment analysis</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(goSEA, <span class="at">colorEdge =</span> <span class="cn">TRUE</span>, <span class="at">cex_label_gene =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(goSEA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>goSEA <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(goSEA)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">treeplot</span>(goSEA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="random-forest-classification" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Random forest classification</h1>
<section id="prepare-data" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="prepare-data"><span class="header-section-number">6.1</span> Prepare data</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and prepare the data</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>data_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data_raw <span class="sc">%&gt;%</span></span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneSymbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GeneSymbol)) <span class="sc">%&gt;%</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"GeneSymbol"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="dv">0</span>, .)) <span class="sc">%&gt;%</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(as.integer)</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare metadata</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>metadata_raw <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata_raw <span class="sc">%&gt;%</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Index =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"Index"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sample =</span> ID, <span class="at">disease =</span> Disease) <span class="sc">%&gt;%</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">gsub</span>(<span class="st">"_raw"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, individual, disease) <span class="sc">%&gt;%</span></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">factor</span>(individual), <span class="at">disease =</span> <span class="fu">factor</span>(disease))</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure the data is in a matrix form with gene symbols as rownames</span></span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data)</span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DESeqDataSet</span></span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> countData,</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata,</span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> disease</span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate size factors</span></span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate gene-wise dispersions</span></span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateDispersionsGeneEst</span>(dds)</span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Directly use the gene-wise estimates</span></span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a><span class="fu">dispersions</span>(dds) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(dds)<span class="sc">$</span>dispGeneEst</span>
<span id="cb52-39"><a href="#cb52-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-40"><a href="#cb52-40" aria-hidden="true" tabindex="-1"></a>normCounts <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Define model fitting procedure</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="co">#NB: you will also need the packages statmod and e1071 for this analysis</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co">#New score function</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">mean</span>(x)</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co">#New filter function</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>filter <span class="ot">&lt;-</span> <span class="cf">function</span>(score, x, y){</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(data) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>selection <span class="ot">&lt;-</span> score <span class="sc">&gt;</span> <span class="fu">quantile</span>(score, <span class="fl">0.9</span>) <span class="sc">&amp;</span> meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(score) <span class="sc">%in%</span> <span class="fu">names</span>(score)[selection]</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create training set</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>trainingSet <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">assay</span>(normCounts)))</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Train and validate decision tree (C5.0)</span></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>dtControl <span class="ot">&lt;-</span> <span class="fu">sbfControl</span>(</span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">functions =</span> rfSBF,</span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"loocv"</span>,</span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveDetails =</span> <span class="cn">TRUE</span>,</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>dtModel <span class="ot">&lt;-</span> <span class="fu">sbf</span>(</span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  trainingSet,</span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>  metadata<span class="sc">$</span>disease,</span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">sbfControl =</span> dtControl)</span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Show model</span></span>
<span id="cb54-16"><a href="#cb54-16" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(dtModel$fit)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="performance-evaluation" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="performance-evaluation"><span class="header-section-number">6.2</span> Performance evaluation</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Print confusion matrix from validation</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(dtModel<span class="sc">$</span>pred<span class="sc">$</span>predictions<span class="sc">$</span>pred, metadata<span class="sc">$</span>disease)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction AD Control
   AD       5       6
   Control  4       2
                                          
               Accuracy : 0.4118          
                 95% CI : (0.1844, 0.6708)
    No Information Rate : 0.5294          
    P-Value [Acc &gt; NIR] : 0.8878          
                                          
                  Kappa : -0.1972         
                                          
 Mcnemar's Test P-Value : 0.7518          
                                          
            Sensitivity : 0.5556          
            Specificity : 0.2500          
         Pos Pred Value : 0.4545          
         Neg Pred Value : 0.3333          
             Prevalence : 0.5294          
         Detection Rate : 0.2941          
   Detection Prevalence : 0.6471          
      Balanced Accuracy : 0.4028          
                                          
       'Positive' Class : AD              
                                          </code></pre>
</div>
</div>
</section>
</section>
<section id="spearman-correlation-analysis" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Spearman correlation analysis</h1>
<p>Loading the data for the correlaton analysis</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/significant_genes.tsv"</span>)</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Performing Spearman correlation of the 63 upregulated genes on selected clinical variables of interest and comparing the correlation between Alzheimer’s disease and control samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create correlation data and plot</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>create_correlation_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(data, metadata, gene_symbols, variable, title) {</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  correlation_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(GeneSymbol <span class="sc">%in%</span> gene_symbols) <span class="sc">%&gt;%</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>GeneSymbol, <span class="at">names_to =</span> <span class="st">"ID"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(metadata, <span class="at">by =</span> <span class="st">"ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(GeneSymbol, Disease) <span class="sc">%&gt;%</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">correlation =</span> <span class="fu">cor</span>(value, <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(variable), <span class="at">method =</span> <span class="st">"spearman"</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the order of genes based on their correlation in AD</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  ad_correlations <span class="ot">&lt;-</span> correlation_data <span class="sc">%&gt;%</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Disease <span class="sc">==</span> <span class="st">"AD"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(correlation)) <span class="sc">%&gt;%</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(GeneSymbol)</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure that all GeneSymbols are included, even those not present in AD data</span></span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a>  full_gene_list <span class="ot">&lt;-</span> <span class="fu">unique</span>(correlation_data<span class="sc">$</span>GeneSymbol)</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a>  ad_order <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(ad_correlations, full_gene_list)) <span class="co"># This ensures all genes are included and in order</span></span>
<span id="cb58-20"><a href="#cb58-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-21"><a href="#cb58-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the levels of GeneSymbol according to their order in AD</span></span>
<span id="cb58-22"><a href="#cb58-22" aria-hidden="true" tabindex="-1"></a>  correlation_data<span class="sc">$</span>GeneSymbol <span class="ot">&lt;-</span> <span class="fu">factor</span>(correlation_data<span class="sc">$</span>GeneSymbol, <span class="at">levels =</span> ad_order)</span>
<span id="cb58-23"><a href="#cb58-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-24"><a href="#cb58-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot the heatmap with genes ordered by their correlation in AD</span></span>
<span id="cb58-25"><a href="#cb58-25" aria-hidden="true" tabindex="-1"></a>  heatmap_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(correlation_data, <span class="fu">aes</span>(<span class="at">x =</span> Disease, <span class="at">y =</span> GeneSymbol, <span class="at">fill =</span> correlation)) <span class="sc">+</span></span>
<span id="cb58-26"><a href="#cb58-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb58-27"><a href="#cb58-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">midpoint =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb58-28"><a href="#cb58-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb58-29"><a href="#cb58-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb58-30"><a href="#cb58-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Correlation"</span>, <span class="at">y =</span> <span class="st">"Gene (ordered by AD correlation)"</span>, <span class="at">x =</span> <span class="st">"Disease"</span>, <span class="at">title =</span> title)</span>
<span id="cb58-31"><a href="#cb58-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-32"><a href="#cb58-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(heatmap_plot)</span>
<span id="cb58-33"><a href="#cb58-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb58-34"><a href="#cb58-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-35"><a href="#cb58-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Variables of interest</span></span>
<span id="cb58-36"><a href="#cb58-36" aria-hidden="true" tabindex="-1"></a>variables_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Age"</span>, <span class="st">"PMI"</span>, <span class="st">"Mean_plaque"</span>, <span class="st">"CDR"</span>)</span>
<span id="cb58-37"><a href="#cb58-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-38"><a href="#cb58-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot list</span></span>
<span id="cb58-39"><a href="#cb58-39" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb58-40"><a href="#cb58-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-41"><a href="#cb58-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over variables and create plots</span></span>
<span id="cb58-42"><a href="#cb58-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (variable <span class="cf">in</span> variables_of_interest) {</span>
<span id="cb58-43"><a href="#cb58-43" aria-hidden="true" tabindex="-1"></a>  plot_list[[variable]] <span class="ot">&lt;-</span> <span class="fu">create_correlation_plot</span>(data, metadata, de_results<span class="sc">$</span>GeneSymbol, variable, <span class="fu">paste</span>(<span class="st">"Correlation with"</span>, variable))</span>
<span id="cb58-44"><a href="#cb58-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb58-45"><a href="#cb58-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-46"><a href="#cb58-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb58-47"><a href="#cb58-47" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plot_list, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb58-48"><a href="#cb58-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-49"><a href="#cb58-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the combined plot</span></span>
<span id="cb58-50"><a href="#cb58-50" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="session-info" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Session info</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Stockholm
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          enrichplot_1.22.0          
 [3] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1       
 [5] clusterProfiler_4.10.1      igraph_2.0.3               
 [7] STRINGdb_2.14.3             edgeR_4.0.16               
 [9] caret_6.0-94                lattice_0.22-6             
[11] randomForest_4.7-1.1        pheatmap_1.0.12            
[13] DESeq2_1.42.1               SummarizedExperiment_1.32.0
[15] Biobase_2.62.0              MatrixGenerics_1.14.0      
[17] matrixStats_1.4.1           GenomicRanges_1.54.1       
[19] GenomeInfoDb_1.38.8         IRanges_2.36.0             
[21] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[23] ComplexUpset_1.3.3          UpSetR_1.4.0               
[25] ggvenn_0.1.10               cowplot_1.1.3              
[27] ggplotify_0.1.2             uwot_0.2.2                 
[29] Matrix_1.6-4                umap_0.2.10.0              
[31] plotly_4.10.4               HDAnalyzeR_1.0.0           
[33] xlsx_0.6.5                  tensorflow_2.16.0          
[35] kknn_1.3.1                  xgboost_1.7.8.1            
[37] keras_2.15.0                themis_1.0.2               
[39] vip_0.4.1                   ggsignif_0.6.4             
[41] corrr_0.4.4                 visdat_0.6.0               
[43] GGally_2.2.1                skimr_2.1.5                
[45] gt_0.11.0                   ggbeeswarm_0.7.2           
[47] yardstick_1.3.1             workflowsets_1.1.0         
[49] workflows_1.1.4             tune_1.2.1                 
[51] rsample_1.2.1               recipes_1.1.0              
[53] parsnip_1.2.1               modeldata_1.4.0            
[55] infer_1.0.7                 dials_1.3.0                
[57] scales_1.3.0                broom_1.0.6                
[59] tidymodels_1.2.0            DT_0.33                    
[61] patchwork_1.3.0             kableExtra_1.4.0           
[63] magrittr_2.0.3              ggrepel_0.9.6              
[65] limma_3.58.1                readxl_1.4.3               
[67] lubridate_1.9.3             forcats_1.0.0              
[69] stringr_1.5.1               dplyr_1.1.4                
[71] purrr_1.0.2                 readr_2.1.5                
[73] tidyr_1.3.1                 tibble_3.2.1               
[75] ggplot2_3.5.1               tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             nnet_7.3-19             Biostrings_2.70.3      
  [4] vctrs_0.6.5             proxy_0.4-27            digest_0.6.37          
  [7] png_0.1-8               tfruns_1.5.3            parallelly_1.38.0      
 [10] MASS_7.3-60             reshape2_1.4.4          foreach_1.5.2          
 [13] qvalue_2.34.0           withr_3.0.1             xfun_0.49              
 [16] ggfun_0.1.6             survival_3.7-0          memoise_2.0.1          
 [19] gson_0.1.0              systemfonts_1.1.0       tidytree_0.4.6         
 [22] gtools_3.9.5            KEGGREST_1.42.0         httr_1.4.7             
 [25] globals_0.16.3          hash_2.2.6.3            rstudioapi_0.16.0      
 [28] archive_1.1.9           generics_0.1.3          DOSE_3.28.2            
 [31] base64enc_0.1-3         curl_5.2.2              repr_1.1.7             
 [34] zlibbioc_1.48.2         ggraph_2.2.1            polyclip_1.10-7        
 [37] GenomeInfoDbData_1.2.11 SparseArray_1.2.4       evaluate_1.0.0         
 [40] S4Arrays_1.2.1          hms_1.1.3               colorspace_2.1-1       
 [43] reticulate_1.39.0       viridis_0.6.5           ggtree_3.10.1          
 [46] future.apply_1.11.2     lhs_1.2.0               shadowtext_0.1.4       
 [49] ROSE_0.0-4              class_7.3-22            pillar_1.9.0           
 [52] nlme_3.1-166            iterators_1.0.14        GPfit_1.0-8            
 [55] caTools_1.18.3          compiler_4.3.2          RSpectra_0.16-2        
 [58] stringi_1.8.4           gower_1.0.1             plyr_1.8.9             
 [61] crayon_1.5.3            abind_1.4-8             gridGraphics_0.5-1     
 [64] chron_2.3-61            locfit_1.5-9.10         graphlayouts_1.1.1     
 [67] bit_4.0.5               fastmatch_1.1-4         whisker_0.4.1          
 [70] codetools_0.2-20        openssl_2.2.1           e1071_1.7-16           
 [73] splines_4.3.2           Rcpp_1.0.13             DiceDesign_1.10        
 [76] HDO.db_0.99.1           cellranger_1.1.0        knitr_1.48             
 [79] blob_1.2.4              utf8_1.2.4              fs_1.6.4               
 [82] listenv_0.9.1           sqldf_0.4-11            statmod_1.5.0          
 [85] tzdb_0.4.0              svglite_2.1.3           tweenr_2.0.3           
 [88] pkgconfig_2.0.3         tools_4.3.2             cachem_1.1.0           
 [91] RSQLite_2.3.7           viridisLite_0.4.2       DBI_1.2.3              
 [94] fastmap_1.2.0           rmarkdown_2.28          ggstats_0.6.0          
 [97] rpart_4.1.23            farver_2.1.2            tidygraph_1.3.1        
[100] scatterpie_0.2.4        gsubfn_0.7              yaml_2.3.10            
[103] cli_3.6.3               lifecycle_1.0.4         askpass_1.2.0          
[106] lava_1.8.0              backports_1.5.0         BiocParallel_1.36.0    
[109] timechange_0.3.0        gtable_0.3.5            parallel_4.3.2         
[112] pROC_1.18.5             ape_5.8                 jsonlite_1.8.8         
[115] bitops_1.0-8            bit64_4.0.5             yulab.utils_0.1.7      
[118] proto_1.0.0             GOSemSim_2.28.1         zeallot_0.1.0          
[121] timeDate_4032.109       lazyeval_0.2.2          htmltools_0.5.8.1      
[124] rJava_1.0-11            GO.db_3.18.0            glue_1.7.0             
[127] XVector_0.42.0          RCurl_1.98-1.16         treeio_1.26.0          
[130] gridExtra_2.3           R6_2.5.1                gplots_3.1.3.1         
[133] labeling_0.4.3          xlsxjars_0.6.1          aplot_0.2.3            
[136] ipred_0.9-15            DelayedArray_0.28.0     tidyselect_1.2.1       
[139] vipor_0.4.7             plotrix_3.8-4           ggforce_0.4.2          
[142] xml2_1.3.6              future_1.34.0           ModelMetrics_1.2.2.2   
[145] munsell_0.5.1           KernSmooth_2.23-24      furrr_0.3.1            
[148] data.table_1.16.0       htmlwidgets_1.6.4       fgsea_1.28.0           
[151] rlang_1.1.4             ggnewscale_0.5.0        fansi_1.0.6            
[154] hardhat_1.4.0           beeswarm_0.4.0          prodlim_2024.06.25     </code></pre>
</div>
</div>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb61" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Group 2 report: Characterization of Alzheimer's disease"</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "A project for the course Statistics in Bioinformatics"</span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Emil Johansson"</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> last-modified</span></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="an">date-format:</span><span class="co"> "dddd [the] Do [of] MMMM YYYY"</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: lumen</span></span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a><span class="co">    title-block-banner: true</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a><span class="co">    smooth-scroll: true</span></span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 4</span></span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right</span></span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true</span></span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 4</span></span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb61-19"><a href="#cb61-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb61-20"><a href="#cb61-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb61-21"><a href="#cb61-21" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: kable</span></span>
<span id="cb61-22"><a href="#cb61-22" aria-hidden="true" tabindex="-1"></a><span class="co">    standalone: true</span></span>
<span id="cb61-23"><a href="#cb61-23" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-align: left</span></span>
<span id="cb61-24"><a href="#cb61-24" aria-hidden="true" tabindex="-1"></a><span class="co">    figure:</span></span>
<span id="cb61-25"><a href="#cb61-25" aria-hidden="true" tabindex="-1"></a><span class="co">      caption: "Figure {number}: {label}"</span></span>
<span id="cb61-26"><a href="#cb61-26" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb61-27"><a href="#cb61-27" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: inline</span></span>
<span id="cb61-28"><a href="#cb61-28" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb61-29"><a href="#cb61-29" aria-hidden="true" tabindex="-1"></a><span class="co">  echo: true</span></span>
<span id="cb61-30"><a href="#cb61-30" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb61-31"><a href="#cb61-31" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb61-32"><a href="#cb61-32" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb61-33"><a href="#cb61-33" aria-hidden="true" tabindex="-1"></a><span class="co">  markdown: </span></span>
<span id="cb61-34"><a href="#cb61-34" aria-hidden="true" tabindex="-1"></a><span class="co">    wrap: 72</span></span>
<span id="cb61-35"><a href="#cb61-35" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb61-36"><a href="#cb61-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-37"><a href="#cb61-37" aria-hidden="true" tabindex="-1"></a><span class="fu"># Setup</span></span>
<span id="cb61-38"><a href="#cb61-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-41"><a href="#cb61-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-42"><a href="#cb61-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Libraries we need</span></span>
<span id="cb61-43"><a href="#cb61-43" aria-hidden="true" tabindex="-1"></a>libs <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb61-44"><a href="#cb61-44" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tidyverse"</span>, <span class="st">"readxl"</span>, <span class="st">"limma"</span>, <span class="st">"ggrepel"</span>, <span class="st">"magrittr"</span>, <span class="st">"kableExtra"</span>,</span>
<span id="cb61-45"><a href="#cb61-45" aria-hidden="true" tabindex="-1"></a>  <span class="st">"patchwork"</span>, <span class="st">"DT"</span>, <span class="st">"tidymodels"</span>, <span class="st">"ggbeeswarm"</span>, <span class="st">"gt"</span>, <span class="st">"skimr"</span>, <span class="st">"GGally"</span>,</span>
<span id="cb61-46"><a href="#cb61-46" aria-hidden="true" tabindex="-1"></a>  <span class="st">"visdat"</span>, <span class="st">"corrr"</span>, <span class="st">"ggsignif"</span>, <span class="st">"vip"</span>, <span class="st">"themis"</span>, <span class="st">"keras"</span>, <span class="st">"xgboost"</span>, <span class="st">"kknn"</span>,</span>
<span id="cb61-47"><a href="#cb61-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tensorflow"</span>, <span class="st">"xlsx"</span>, <span class="st">"HDAnalyzeR"</span>, <span class="st">"plotly"</span>, <span class="st">"umap"</span>, <span class="st">"uwot"</span>, <span class="st">"ggplotify"</span>,</span>
<span id="cb61-48"><a href="#cb61-48" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cowplot"</span>, <span class="st">"ggvenn"</span>, <span class="st">"UpSetR"</span>, <span class="st">"ComplexUpset"</span>, <span class="st">"DESeq2"</span>, <span class="st">"pheatmap"</span>, <span class="st">"randomForest"</span>,</span>
<span id="cb61-49"><a href="#cb61-49" aria-hidden="true" tabindex="-1"></a>  <span class="st">"caret"</span>, <span class="st">"edgeR"</span>, <span class="st">"STRINGdb"</span>, <span class="st">"igraph"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"enrichplot"</span></span>
<span id="cb61-50"><a href="#cb61-50" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb61-51"><a href="#cb61-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-52"><a href="#cb61-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Install missing libraries</span></span>
<span id="cb61-53"><a href="#cb61-53" aria-hidden="true" tabindex="-1"></a>installed_libs <span class="ot">&lt;-</span> libs <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb61-54"><a href="#cb61-54" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(installed_libs <span class="sc">==</span> <span class="cn">FALSE</span>)) {</span>
<span id="cb61-55"><a href="#cb61-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(libs[<span class="sc">!</span>installed_libs])</span>
<span id="cb61-56"><a href="#cb61-56" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-57"><a href="#cb61-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-58"><a href="#cb61-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb61-59"><a href="#cb61-59" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(libs, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span>
<span id="cb61-60"><a href="#cb61-60" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-61"><a href="#cb61-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-62"><a href="#cb61-62" aria-hidden="true" tabindex="-1"></a><span class="fu"># QC</span></span>
<span id="cb61-63"><a href="#cb61-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-64"><a href="#cb61-64" aria-hidden="true" tabindex="-1"></a><span class="in">```{r load-libraries}</span></span>
<span id="cb61-65"><a href="#cb61-65" aria-hidden="true" tabindex="-1"></a><span class="in">library(dplyr)</span></span>
<span id="cb61-66"><a href="#cb61-66" aria-hidden="true" tabindex="-1"></a><span class="in">library(DESeq2)</span></span>
<span id="cb61-67"><a href="#cb61-67" aria-hidden="true" tabindex="-1"></a><span class="in">library(edgeR)</span></span>
<span id="cb61-68"><a href="#cb61-68" aria-hidden="true" tabindex="-1"></a><span class="in">library(limma)</span></span>
<span id="cb61-69"><a href="#cb61-69" aria-hidden="true" tabindex="-1"></a><span class="in">library(pheatmap)</span></span>
<span id="cb61-70"><a href="#cb61-70" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggplot2)</span></span>
<span id="cb61-71"><a href="#cb61-71" aria-hidden="true" tabindex="-1"></a><span class="in">library(ggrepel)</span></span>
<span id="cb61-72"><a href="#cb61-72" aria-hidden="true" tabindex="-1"></a><span class="in">library(RColorBrewer)</span></span>
<span id="cb61-73"><a href="#cb61-73" aria-hidden="true" tabindex="-1"></a><span class="in">library(clusterProfiler)</span></span>
<span id="cb61-74"><a href="#cb61-74" aria-hidden="true" tabindex="-1"></a><span class="in">library(org.Hs.eg.db)</span></span>
<span id="cb61-75"><a href="#cb61-75" aria-hidden="true" tabindex="-1"></a><span class="in">library(enrichplot)</span></span>
<span id="cb61-76"><a href="#cb61-76" aria-hidden="true" tabindex="-1"></a><span class="in">library(STRINGdb)</span></span>
<span id="cb61-77"><a href="#cb61-77" aria-hidden="true" tabindex="-1"></a><span class="in">library(igraph)</span></span>
<span id="cb61-78"><a href="#cb61-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-79"><a href="#cb61-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-80"><a href="#cb61-80" aria-hidden="true" tabindex="-1"></a><span class="in">```{r load data}</span></span>
<span id="cb61-81"><a href="#cb61-81" aria-hidden="true" tabindex="-1"></a><span class="in"># Load gene expression count data</span></span>
<span id="cb61-82"><a href="#cb61-82" aria-hidden="true" tabindex="-1"></a><span class="in">count_data &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1) %&gt;%</span></span>
<span id="cb61-83"><a href="#cb61-83" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(matches("_raw$"))</span></span>
<span id="cb61-84"><a href="#cb61-84" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-85"><a href="#cb61-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-86"><a href="#cb61-86" aria-hidden="true" tabindex="-1"></a><span class="in">```{r preprocess data}</span></span>
<span id="cb61-87"><a href="#cb61-87" aria-hidden="true" tabindex="-1"></a><span class="in"># Round counts to integers</span></span>
<span id="cb61-88"><a href="#cb61-88" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data %&gt;% </span></span>
<span id="cb61-89"><a href="#cb61-89" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(starts_with("C"), starts_with("A")) %&gt;%</span></span>
<span id="cb61-90"><a href="#cb61-90" aria-hidden="true" tabindex="-1"></a><span class="in">  mutate_all(~round(.))</span></span>
<span id="cb61-91"><a href="#cb61-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-92"><a href="#cb61-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-93"><a href="#cb61-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filter low counts using mean log2 CPM}</span></span>
<span id="cb61-94"><a href="#cb61-94" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data</span></span>
<span id="cb61-95"><a href="#cb61-95" aria-hidden="true" tabindex="-1"></a><span class="in">genes_before_filtering &lt;- nrow(count_data_filtered)</span></span>
<span id="cb61-96"><a href="#cb61-96" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of genes before filtering:", genes_before_filtering, "\n")</span></span>
<span id="cb61-97"><a href="#cb61-97" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(counts = count_data_filtered)</span></span>
<span id="cb61-98"><a href="#cb61-98" aria-hidden="true" tabindex="-1"></a><span class="in"># dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb61-99"><a href="#cb61-99" aria-hidden="true" tabindex="-1"></a><span class="in">meanLog2CPM &lt;- rowMeans(log2(cpm(dge) + 1))</span></span>
<span id="cb61-100"><a href="#cb61-100" aria-hidden="true" tabindex="-1"></a><span class="in">hist(meanLog2CPM, main = "Histogram of Mean Log2 CPM", xlab = "Mean Log2 CPM")</span></span>
<span id="cb61-101"><a href="#cb61-101" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data_filtered[meanLog2CPM &gt; 1, ]</span></span>
<span id="cb61-102"><a href="#cb61-102" aria-hidden="true" tabindex="-1"></a><span class="in">genes_after_filtering &lt;- nrow(count_data_filtered)</span></span>
<span id="cb61-103"><a href="#cb61-103" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of genes after filtering:", genes_after_filtering, "\n")</span></span>
<span id="cb61-104"><a href="#cb61-104" aria-hidden="true" tabindex="-1"></a><span class="in">sample_conditions &lt;- data.frame(</span></span>
<span id="cb61-105"><a href="#cb61-105" aria-hidden="true" tabindex="-1"></a><span class="in">  sample = colnames(count_data_filtered),</span></span>
<span id="cb61-106"><a href="#cb61-106" aria-hidden="true" tabindex="-1"></a><span class="in">  condition = ifelse(grepl("^C", colnames(count_data_filtered)), "HC", "AD"),</span></span>
<span id="cb61-107"><a href="#cb61-107" aria-hidden="true" tabindex="-1"></a><span class="in">  batch = c(0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1)  # Manually adding batch information</span></span>
<span id="cb61-108"><a href="#cb61-108" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb61-109"><a href="#cb61-109" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-110"><a href="#cb61-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-111"><a href="#cb61-111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r normalization with Voom}</span></span>
<span id="cb61-112"><a href="#cb61-112" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(counts = count_data_filtered)</span></span>
<span id="cb61-113"><a href="#cb61-113" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb61-114"><a href="#cb61-114" aria-hidden="true" tabindex="-1"></a><span class="in"># design &lt;- model.matrix(~ condition + batch, data = sample_conditions)</span></span>
<span id="cb61-115"><a href="#cb61-115" aria-hidden="true" tabindex="-1"></a><span class="in">design &lt;- model.matrix(~ condition + factor(batch), data = sample_conditions)</span></span>
<span id="cb61-116"><a href="#cb61-116" aria-hidden="true" tabindex="-1"></a><span class="in">v &lt;- voom(dge, design, plot = TRUE)</span></span>
<span id="cb61-117"><a href="#cb61-117" aria-hidden="true" tabindex="-1"></a><span class="in">head(v$E)</span></span>
<span id="cb61-118"><a href="#cb61-118" aria-hidden="true" tabindex="-1"></a><span class="in">hist(v$E, breaks = 50, main = "Distribution of Voom Normalized Counts", xlab = "Voom Normalized Counts")</span></span>
<span id="cb61-119"><a href="#cb61-119" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-120"><a href="#cb61-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-121"><a href="#cb61-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r normalization with DESeq2}</span></span>
<span id="cb61-122"><a href="#cb61-122" aria-hidden="true" tabindex="-1"></a><span class="in"># dds &lt;- DESeqDataSetFromMatrix(countData = count_data_filtered, </span></span>
<span id="cb61-123"><a href="#cb61-123" aria-hidden="true" tabindex="-1"></a><span class="in">#                               colData = sample_conditions, </span></span>
<span id="cb61-124"><a href="#cb61-124" aria-hidden="true" tabindex="-1"></a><span class="in">#                               design = ~ condition + factor(batch))</span></span>
<span id="cb61-125"><a href="#cb61-125" aria-hidden="true" tabindex="-1"></a><span class="in"># Normalize counts using variance stabilizing transformation (vst)</span></span>
<span id="cb61-126"><a href="#cb61-126" aria-hidden="true" tabindex="-1"></a><span class="in"># normCounts &lt;- vst(dds, blind = TRUE)</span></span>
<span id="cb61-127"><a href="#cb61-127" aria-hidden="true" tabindex="-1"></a><span class="in"># assay(normCounts)[1:5, 1:5]</span></span>
<span id="cb61-128"><a href="#cb61-128" aria-hidden="true" tabindex="-1"></a><span class="in"># hist(assay(normCounts), breaks = 50, main = "Distribution of VST Normalized Counts", xlab = "Normalized Counts")</span></span>
<span id="cb61-129"><a href="#cb61-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-130"><a href="#cb61-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-131"><a href="#cb61-131" aria-hidden="true" tabindex="-1"></a><span class="in">```{r save normalized results}</span></span>
<span id="cb61-132"><a href="#cb61-132" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb61-133"><a href="#cb61-133" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb61-134"><a href="#cb61-134" aria-hidden="true" tabindex="-1"></a><span class="in">voom_normalized &lt;- as.data.frame(v$E)  # Convert v$E to a data frame</span></span>
<span id="cb61-135"><a href="#cb61-135" aria-hidden="true" tabindex="-1"></a><span class="in">voom_normalized$GeneID &lt;- rownames(voom_normalized)  # Add GeneID as a column</span></span>
<span id="cb61-136"><a href="#cb61-136" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_voom &lt;- merge(voom_normalized, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb61-137"><a href="#cb61-137" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_voom &lt;- gene_info_voom[, c("GeneID", "GeneSymbol", setdiff(names(gene_info_voom), c("GeneID", "GeneSymbol")))]</span></span>
<span id="cb61-138"><a href="#cb61-138" aria-hidden="true" tabindex="-1"></a><span class="in"># write.table(gene_info_voom, file = "voom_normalized.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span>
<span id="cb61-139"><a href="#cb61-139" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-140"><a href="#cb61-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-141"><a href="#cb61-141" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: condition heatmap}</span></span>
<span id="cb61-142"><a href="#cb61-142" aria-hidden="true" tabindex="-1"></a><span class="in"># sampleDist &lt;- cor(assay(normCounts), method = "spearman")</span></span>
<span id="cb61-143"><a href="#cb61-143" aria-hidden="true" tabindex="-1"></a><span class="in">sampleDist &lt;- cor(v$E, method = "spearman")</span></span>
<span id="cb61-144"><a href="#cb61-144" aria-hidden="true" tabindex="-1"></a><span class="in">sampleColor &lt;- brewer.pal(3, "Accent")[1:2]</span></span>
<span id="cb61-145"><a href="#cb61-145" aria-hidden="true" tabindex="-1"></a><span class="in">names(sampleColor) &lt;- unique(sample_conditions$condition)</span></span>
<span id="cb61-146"><a href="#cb61-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-147"><a href="#cb61-147" aria-hidden="true" tabindex="-1"></a><span class="in">pheatmap(sampleDist,</span></span>
<span id="cb61-148"><a href="#cb61-148" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_rows = as.dist(1 - sampleDist),</span></span>
<span id="cb61-149"><a href="#cb61-149" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_cols = as.dist(1 - sampleDist),</span></span>
<span id="cb61-150"><a href="#cb61-150" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_col = data.frame(Condition = sample_conditions$condition, row.names = sample_conditions$sample),</span></span>
<span id="cb61-151"><a href="#cb61-151" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_colors = list(Storage = sampleColor),</span></span>
<span id="cb61-152"><a href="#cb61-152" aria-hidden="true" tabindex="-1"></a><span class="in">  main = "Sample Correlation Heatmap by Condition")</span></span>
<span id="cb61-153"><a href="#cb61-153" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-154"><a href="#cb61-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: PCA by condition}</span></span>
<span id="cb61-155"><a href="#cb61-155" aria-hidden="true" tabindex="-1"></a><span class="in"># pcaRes &lt;- prcomp(t(assay(normCounts)))</span></span>
<span id="cb61-156"><a href="#cb61-156" aria-hidden="true" tabindex="-1"></a><span class="in">pcaRes &lt;- prcomp(t(v$E))</span></span>
<span id="cb61-157"><a href="#cb61-157" aria-hidden="true" tabindex="-1"></a><span class="in">varExp &lt;- round(pcaRes$sdev^2 / sum(pcaRes$sdev^2) * 100)</span></span>
<span id="cb61-158"><a href="#cb61-158" aria-hidden="true" tabindex="-1"></a><span class="in">pcaDF &lt;- data.frame(</span></span>
<span id="cb61-159"><a href="#cb61-159" aria-hidden="true" tabindex="-1"></a><span class="in">  PC1 = pcaRes$x[, 1],</span></span>
<span id="cb61-160"><a href="#cb61-160" aria-hidden="true" tabindex="-1"></a><span class="in">  PC2 = pcaRes$x[, 2],</span></span>
<span id="cb61-161"><a href="#cb61-161" aria-hidden="true" tabindex="-1"></a><span class="in">  Condition = sample_conditions$condition,</span></span>
<span id="cb61-162"><a href="#cb61-162" aria-hidden="true" tabindex="-1"></a><span class="in">  Sample = sample_conditions$sample</span></span>
<span id="cb61-163"><a href="#cb61-163" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb61-164"><a href="#cb61-164" aria-hidden="true" tabindex="-1"></a><span class="in">pcaPlot &lt;- ggplot(</span></span>
<span id="cb61-165"><a href="#cb61-165" aria-hidden="true" tabindex="-1"></a><span class="in">  data = pcaDF,</span></span>
<span id="cb61-166"><a href="#cb61-166" aria-hidden="true" tabindex="-1"></a><span class="in">  mapping = aes(x = PC1, y = PC2, color = Condition, label = Sample)</span></span>
<span id="cb61-167"><a href="#cb61-167" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb61-168"><a href="#cb61-168" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(size = 3) +</span></span>
<span id="cb61-169"><a href="#cb61-169" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 4) +</span></span>
<span id="cb61-170"><a href="#cb61-170" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x = paste0("PC1 (", varExp[1], " %)") ,</span></span>
<span id="cb61-171"><a href="#cb61-171" aria-hidden="true" tabindex="-1"></a><span class="in">       y = paste0("PC2 (", varExp[2], " %)") ,</span></span>
<span id="cb61-172"><a href="#cb61-172" aria-hidden="true" tabindex="-1"></a><span class="in">       title = "PCA Plot of Samples") +</span></span>
<span id="cb61-173"><a href="#cb61-173" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb61-174"><a href="#cb61-174" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(axis.text = element_text(size = 12), legend.text = element_text(size = 10)) +</span></span>
<span id="cb61-175"><a href="#cb61-175" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_color_manual(values = brewer.pal(3, "Accent")) +</span></span>
<span id="cb61-176"><a href="#cb61-176" aria-hidden="true" tabindex="-1"></a><span class="in">  coord_fixed(ratio = 1)  # Added fixed ratio to ensure equal scaling between PC1 and PC2</span></span>
<span id="cb61-177"><a href="#cb61-177" aria-hidden="true" tabindex="-1"></a><span class="in">print(pcaPlot)</span></span>
<span id="cb61-178"><a href="#cb61-178" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-179"><a href="#cb61-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-180"><a href="#cb61-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: batch heatmap}</span></span>
<span id="cb61-181"><a href="#cb61-181" aria-hidden="true" tabindex="-1"></a><span class="in"># Create a heatmap showing Batch 1 and Batch 2 samples separately</span></span>
<span id="cb61-182"><a href="#cb61-182" aria-hidden="true" tabindex="-1"></a><span class="in">sampleDist_batch &lt;- cor(v$E, method = "spearman")</span></span>
<span id="cb61-183"><a href="#cb61-183" aria-hidden="true" tabindex="-1"></a><span class="in">batchColor &lt;- brewer.pal(3, "Set2")[1:2]</span></span>
<span id="cb61-184"><a href="#cb61-184" aria-hidden="true" tabindex="-1"></a><span class="in">names(batchColor) &lt;- unique(sample_conditions$batch)</span></span>
<span id="cb61-185"><a href="#cb61-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-186"><a href="#cb61-186" aria-hidden="true" tabindex="-1"></a><span class="in">pheatmap(sampleDist_batch,</span></span>
<span id="cb61-187"><a href="#cb61-187" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_rows = as.dist(1 - sampleDist_batch),</span></span>
<span id="cb61-188"><a href="#cb61-188" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_cols = as.dist(1 - sampleDist_batch),</span></span>
<span id="cb61-189"><a href="#cb61-189" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_col = data.frame(Batch = factor(sample_conditions$batch), row.names = sample_conditions$sample),</span></span>
<span id="cb61-190"><a href="#cb61-190" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_colors = list(Batch = batchColor),</span></span>
<span id="cb61-191"><a href="#cb61-191" aria-hidden="true" tabindex="-1"></a><span class="in">  main = "Sample Correlation Heatmap by Batch")</span></span>
<span id="cb61-192"><a href="#cb61-192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-193"><a href="#cb61-193" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: PCA by Batch}</span></span>
<span id="cb61-194"><a href="#cb61-194" aria-hidden="true" tabindex="-1"></a><span class="in"># PCA Plot by Batch</span></span>
<span id="cb61-195"><a href="#cb61-195" aria-hidden="true" tabindex="-1"></a><span class="in">pcaRes_batch &lt;- prcomp(t(v$E))</span></span>
<span id="cb61-196"><a href="#cb61-196" aria-hidden="true" tabindex="-1"></a><span class="in">varExp_batch &lt;- round(pcaRes_batch$sdev^2 / sum(pcaRes_batch$sdev^2) * 100)</span></span>
<span id="cb61-197"><a href="#cb61-197" aria-hidden="true" tabindex="-1"></a><span class="in">pcaDF_batch &lt;- data.frame(</span></span>
<span id="cb61-198"><a href="#cb61-198" aria-hidden="true" tabindex="-1"></a><span class="in">  PC1 = pcaRes_batch$x[, 1],</span></span>
<span id="cb61-199"><a href="#cb61-199" aria-hidden="true" tabindex="-1"></a><span class="in">  PC2 = pcaRes_batch$x[, 2],</span></span>
<span id="cb61-200"><a href="#cb61-200" aria-hidden="true" tabindex="-1"></a><span class="in">  Batch = factor(sample_conditions$batch),</span></span>
<span id="cb61-201"><a href="#cb61-201" aria-hidden="true" tabindex="-1"></a><span class="in">  Sample = sample_conditions$sample</span></span>
<span id="cb61-202"><a href="#cb61-202" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb61-203"><a href="#cb61-203" aria-hidden="true" tabindex="-1"></a><span class="in">pcaPlot_batch &lt;- ggplot(</span></span>
<span id="cb61-204"><a href="#cb61-204" aria-hidden="true" tabindex="-1"></a><span class="in">  data = pcaDF_batch,</span></span>
<span id="cb61-205"><a href="#cb61-205" aria-hidden="true" tabindex="-1"></a><span class="in">  mapping = aes(x = PC1, y = PC2, color = Batch, label = Sample)</span></span>
<span id="cb61-206"><a href="#cb61-206" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb61-207"><a href="#cb61-207" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(size = 3) +</span></span>
<span id="cb61-208"><a href="#cb61-208" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 4) +</span></span>
<span id="cb61-209"><a href="#cb61-209" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x = paste0("PC1 (", varExp_batch[1], " %)"),</span></span>
<span id="cb61-210"><a href="#cb61-210" aria-hidden="true" tabindex="-1"></a><span class="in">       y = paste0("PC2 (", varExp_batch[2], " %)"),</span></span>
<span id="cb61-211"><a href="#cb61-211" aria-hidden="true" tabindex="-1"></a><span class="in">       title = "PCA Plot of Samples by Batch") +</span></span>
<span id="cb61-212"><a href="#cb61-212" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb61-213"><a href="#cb61-213" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(axis.text = element_text(size = 12), legend.text = element_text(size = 10)) +</span></span>
<span id="cb61-214"><a href="#cb61-214" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_color_manual(values = brewer.pal(3, "Set2")) +</span></span>
<span id="cb61-215"><a href="#cb61-215" aria-hidden="true" tabindex="-1"></a><span class="in">  coord_fixed(ratio = 1)</span></span>
<span id="cb61-216"><a href="#cb61-216" aria-hidden="true" tabindex="-1"></a><span class="in">print(pcaPlot_batch)</span></span>
<span id="cb61-217"><a href="#cb61-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-218"><a href="#cb61-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-219"><a href="#cb61-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: outlier removal}</span></span>
<span id="cb61-220"><a href="#cb61-220" aria-hidden="true" tabindex="-1"></a><span class="in"># Remove identified outliers</span></span>
<span id="cb61-221"><a href="#cb61-221" aria-hidden="true" tabindex="-1"></a><span class="in"># outliers &lt;- c("C1_raw", "C2_raw", "C4_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb61-222"><a href="#cb61-222" aria-hidden="true" tabindex="-1"></a><span class="in">outliers &lt;- c("C1_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb61-223"><a href="#cb61-223" aria-hidden="true" tabindex="-1"></a><span class="in">remaining_samples &lt;- setdiff(colnames(count_data_filtered), outliers)</span></span>
<span id="cb61-224"><a href="#cb61-224" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data_filtered[, remaining_samples]</span></span>
<span id="cb61-225"><a href="#cb61-225" aria-hidden="true" tabindex="-1"></a><span class="in">sample_conditions &lt;- sample_conditions[!sample_conditions$sample %in% outliers, ]</span></span>
<span id="cb61-226"><a href="#cb61-226" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-227"><a href="#cb61-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-228"><a href="#cb61-228" aria-hidden="true" tabindex="-1"></a><span class="fu"># Differential expression analysis</span></span>
<span id="cb61-229"><a href="#cb61-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-230"><a href="#cb61-230" aria-hidden="true" tabindex="-1"></a><span class="fu">## Design matrix</span></span>
<span id="cb61-231"><a href="#cb61-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-232"><a href="#cb61-232" aria-hidden="true" tabindex="-1"></a><span class="in">```{r statistical analyses}</span></span>
<span id="cb61-233"><a href="#cb61-233" aria-hidden="true" tabindex="-1"></a><span class="in"># Define design matrix including batch effect</span></span>
<span id="cb61-234"><a href="#cb61-234" aria-hidden="true" tabindex="-1"></a><span class="in"># designMatrix &lt;- model.matrix(~ 0 + condition + batch, data = sample_conditions)</span></span>
<span id="cb61-235"><a href="#cb61-235" aria-hidden="true" tabindex="-1"></a><span class="in">designMatrix &lt;- model.matrix(~ 0 + condition + factor(batch), data = sample_conditions)</span></span>
<span id="cb61-236"><a href="#cb61-236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-237"><a href="#cb61-237" aria-hidden="true" tabindex="-1"></a><span class="in">colnames(designMatrix) &lt;- make.names(colnames(designMatrix))  # Ensure valid column names for contrast matrix</span></span>
<span id="cb61-238"><a href="#cb61-238" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Design Matrix with Batch Effect:\n")</span></span>
<span id="cb61-239"><a href="#cb61-239" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(designMatrix))</span></span>
<span id="cb61-240"><a href="#cb61-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-241"><a href="#cb61-241" aria-hidden="true" tabindex="-1"></a><span class="in"># Define contrast matrix</span></span>
<span id="cb61-242"><a href="#cb61-242" aria-hidden="true" tabindex="-1"></a><span class="in">contrastMatrix &lt;- makeContrasts(AD_vs_HC = conditionAD - conditionHC, levels = designMatrix)</span></span>
<span id="cb61-243"><a href="#cb61-243" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Contrast Matrix:\n")</span></span>
<span id="cb61-244"><a href="#cb61-244" aria-hidden="true" tabindex="-1"></a><span class="in">print(contrastMatrix)</span></span>
<span id="cb61-245"><a href="#cb61-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-246"><a href="#cb61-246" aria-hidden="true" tabindex="-1"></a><span class="in"># Prepare DGEList and estimate dispersion</span></span>
<span id="cb61-247"><a href="#cb61-247" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(count_data_filtered)</span></span>
<span id="cb61-248"><a href="#cb61-248" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb61-249"><a href="#cb61-249" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- estimateDisp(dge, designMatrix, robust = TRUE)</span></span>
<span id="cb61-250"><a href="#cb61-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-251"><a href="#cb61-251" aria-hidden="true" tabindex="-1"></a><span class="in"># Fit the model using likelihood ratio test (LRT), now including batch in the design</span></span>
<span id="cb61-252"><a href="#cb61-252" aria-hidden="true" tabindex="-1"></a><span class="in">fit &lt;- glmFit(dge, designMatrix)</span></span>
<span id="cb61-253"><a href="#cb61-253" aria-hidden="true" tabindex="-1"></a><span class="in">lrt &lt;- glmLRT(fit, contrast = contrastMatrix)</span></span>
<span id="cb61-254"><a href="#cb61-254" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-255"><a href="#cb61-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-256"><a href="#cb61-256" aria-hidden="true" tabindex="-1"></a><span class="fu">## Hypothesis correction</span></span>
<span id="cb61-257"><a href="#cb61-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-258"><a href="#cb61-258" aria-hidden="true" tabindex="-1"></a><span class="in">```{r hypothesis testing}</span></span>
<span id="cb61-259"><a href="#cb61-259" aria-hidden="true" tabindex="-1"></a><span class="in">res &lt;- topTags(lrt, n = nrow(count_data_filtered))</span></span>
<span id="cb61-260"><a href="#cb61-260" aria-hidden="true" tabindex="-1"></a><span class="in">res$table$FDR &lt;- p.adjust(res$table$PValue, method = "BH")</span></span>
<span id="cb61-261"><a href="#cb61-261" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Summary of p-values:\n")</span></span>
<span id="cb61-262"><a href="#cb61-262" aria-hidden="true" tabindex="-1"></a><span class="in">print(summary(res$table$PValue))</span></span>
<span id="cb61-263"><a href="#cb61-263" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Summary of FDRs:\n")</span></span>
<span id="cb61-264"><a href="#cb61-264" aria-hidden="true" tabindex="-1"></a><span class="in">print(summary(res$table$FDR))</span></span>
<span id="cb61-265"><a href="#cb61-265" aria-hidden="true" tabindex="-1"></a><span class="in">sigRes &lt;- subset(res$table, FDR &lt; 0.05 &amp; abs(logFC) &gt; 1)</span></span>
<span id="cb61-266"><a href="#cb61-266" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of significant genes:", nrow(sigRes), "\n")</span></span>
<span id="cb61-267"><a href="#cb61-267" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::kable(head(sigRes))</span></span>
<span id="cb61-268"><a href="#cb61-268" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-269"><a href="#cb61-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-270"><a href="#cb61-270" aria-hidden="true" tabindex="-1"></a><span class="in">```{r add gene symbols}</span></span>
<span id="cb61-271"><a href="#cb61-271" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb61-272"><a href="#cb61-272" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb61-273"><a href="#cb61-273" aria-hidden="true" tabindex="-1"></a><span class="in">sigRes$GeneID &lt;- rownames(sigRes)  # Add GeneID as a column to the results</span></span>
<span id="cb61-274"><a href="#cb61-274" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- merge(sigRes, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb61-275"><a href="#cb61-275" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- gene_info[, c("GeneID", "GeneSymbol", setdiff(names(gene_info), c("GeneID", "GeneSymbol")))]</span></span>
<span id="cb61-276"><a href="#cb61-276" aria-hidden="true" tabindex="-1"></a><span class="in"># write.table(gene_info, file = "sig_genes.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span>
<span id="cb61-277"><a href="#cb61-277" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-278"><a href="#cb61-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-279"><a href="#cb61-279" aria-hidden="true" tabindex="-1"></a><span class="fu">## Volcano plot</span></span>
<span id="cb61-280"><a href="#cb61-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-281"><a href="#cb61-281" aria-hidden="true" tabindex="-1"></a><span class="in">```{r visualize results}</span></span>
<span id="cb61-282"><a href="#cb61-282" aria-hidden="true" tabindex="-1"></a><span class="in"># Add gene symbols to the results</span></span>
<span id="cb61-283"><a href="#cb61-283" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb61-284"><a href="#cb61-284" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb61-285"><a href="#cb61-285" aria-hidden="true" tabindex="-1"></a><span class="in">res$table$GeneID &lt;- rownames(res$table)</span></span>
<span id="cb61-286"><a href="#cb61-286" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_all &lt;- merge(res$table, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb61-287"><a href="#cb61-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-288"><a href="#cb61-288" aria-hidden="true" tabindex="-1"></a><span class="in"># Volcano plot using gene symbols for all genes</span></span>
<span id="cb61-289"><a href="#cb61-289" aria-hidden="true" tabindex="-1"></a><span class="in">volcanoPlot &lt;- ggplot(gene_info_all, aes(x = logFC, y = -log10(FDR),</span></span>
<span id="cb61-290"><a href="#cb61-290" aria-hidden="true" tabindex="-1"></a><span class="in">                                         color = ifelse(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, "darkred", "grey"),</span></span>
<span id="cb61-291"><a href="#cb61-291" aria-hidden="true" tabindex="-1"></a><span class="in">                                         label = ifelse(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, GeneSymbol, ""))) +</span></span>
<span id="cb61-292"><a href="#cb61-292" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(alpha = 0.5, size = 2) +</span></span>
<span id="cb61-293"><a href="#cb61-293" aria-hidden="true" tabindex="-1"></a><span class="in">  xlab(expression("Fold Change, Log"[2]*"")) +</span></span>
<span id="cb61-294"><a href="#cb61-294" aria-hidden="true" tabindex="-1"></a><span class="in">  ylab(expression("Adjusted P value, Log"[10]*"")) +</span></span>
<span id="cb61-295"><a href="#cb61-295" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_vline(xintercept = c(-1, 1), linetype = "dotted", linewidth = 1) +</span></span>
<span id="cb61-296"><a href="#cb61-296" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_hline(yintercept = -log10(0.05), linetype = "dotted", linewidth = 1) +</span></span>
<span id="cb61-297"><a href="#cb61-297" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb61-298"><a href="#cb61-298" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(legend.position = "none") +</span></span>
<span id="cb61-299"><a href="#cb61-299" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_colour_manual(values = c("darkred", "grey")) +</span></span>
<span id="cb61-300"><a href="#cb61-300" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 3, max.overlaps = 10)</span></span>
<span id="cb61-301"><a href="#cb61-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-302"><a href="#cb61-302" aria-hidden="true" tabindex="-1"></a><span class="in">print(volcanoPlot)</span></span>
<span id="cb61-303"><a href="#cb61-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-304"><a href="#cb61-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-305"><a href="#cb61-305" aria-hidden="true" tabindex="-1"></a><span class="fu"># Protein-protein interaction analysis</span></span>
<span id="cb61-306"><a href="#cb61-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-307"><a href="#cb61-307" aria-hidden="true" tabindex="-1"></a><span class="in">```{r protein-protein interation analysis}</span></span>
<span id="cb61-308"><a href="#cb61-308" aria-hidden="true" tabindex="-1"></a><span class="in"># Initialize STRINGdb</span></span>
<span id="cb61-309"><a href="#cb61-309" aria-hidden="true" tabindex="-1"></a><span class="in">string_db &lt;- STRINGdb$new(version = "11", species = 9606, score_threshold = 400, input_directory = "")</span></span>
<span id="cb61-310"><a href="#cb61-310" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- gene_info$GeneSymbol</span></span>
<span id="cb61-311"><a href="#cb61-311" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- data.frame(GeneSymbol = gene_symbols, stringsAsFactors = FALSE)</span></span>
<span id="cb61-312"><a href="#cb61-312" aria-hidden="true" tabindex="-1"></a><span class="in">mapped_genes &lt;- string_db$map(gene_symbols, "GeneSymbol", removeUnmappedRows = TRUE)</span></span>
<span id="cb61-313"><a href="#cb61-313" aria-hidden="true" tabindex="-1"></a><span class="in"># Get PPI Network for the mapped genes</span></span>
<span id="cb61-314"><a href="#cb61-314" aria-hidden="true" tabindex="-1"></a><span class="in">ppi_network &lt;- string_db$get_interactions(mapped_genes$STRING_id)</span></span>
<span id="cb61-315"><a href="#cb61-315" aria-hidden="true" tabindex="-1"></a><span class="in">string_db$plot_network(mapped_genes$STRING_id)</span></span>
<span id="cb61-316"><a href="#cb61-316" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-317"><a href="#cb61-317" aria-hidden="true" tabindex="-1"></a><span class="in">```{r enrichGO SEA}</span></span>
<span id="cb61-318"><a href="#cb61-318" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols_vector &lt;- gene_symbols$GeneSymbol  # Extract the gene symbols as a vector</span></span>
<span id="cb61-319"><a href="#cb61-319" aria-hidden="true" tabindex="-1"></a><span class="in">mapped_genes_entrez &lt;- bitr(gene_symbols_vector,</span></span>
<span id="cb61-320"><a href="#cb61-320" aria-hidden="true" tabindex="-1"></a><span class="in">                            fromType = "SYMBOL",</span></span>
<span id="cb61-321"><a href="#cb61-321" aria-hidden="true" tabindex="-1"></a><span class="in">                            toType = "ENTREZID",</span></span>
<span id="cb61-322"><a href="#cb61-322" aria-hidden="true" tabindex="-1"></a><span class="in">                            OrgDb = org.Hs.eg.db)</span></span>
<span id="cb61-323"><a href="#cb61-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-324"><a href="#cb61-324" aria-hidden="true" tabindex="-1"></a><span class="in">goSEA &lt;- enrichGO(gene = mapped_genes_entrez$ENTREZID,</span></span>
<span id="cb61-325"><a href="#cb61-325" aria-hidden="true" tabindex="-1"></a><span class="in">                  OrgDb = org.Hs.eg.db,</span></span>
<span id="cb61-326"><a href="#cb61-326" aria-hidden="true" tabindex="-1"></a><span class="in">                  keyType = "ENTREZID",</span></span>
<span id="cb61-327"><a href="#cb61-327" aria-hidden="true" tabindex="-1"></a><span class="in">                  ont = "BP",</span></span>
<span id="cb61-328"><a href="#cb61-328" aria-hidden="true" tabindex="-1"></a><span class="in">                  pAdjustMethod = "BH",</span></span>
<span id="cb61-329"><a href="#cb61-329" aria-hidden="true" tabindex="-1"></a><span class="in">                  pvalueCutoff = 0.05,</span></span>
<span id="cb61-330"><a href="#cb61-330" aria-hidden="true" tabindex="-1"></a><span class="in">                  qvalueCutoff = 0.05)</span></span>
<span id="cb61-331"><a href="#cb61-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-332"><a href="#cb61-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-333"><a href="#cb61-333" aria-hidden="true" tabindex="-1"></a><span class="fu"># Enrichment analysis</span></span>
<span id="cb61-334"><a href="#cb61-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-335"><a href="#cb61-335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Visualize GO SEA enrichment results}</span></span>
<span id="cb61-336"><a href="#cb61-336" aria-hidden="true" tabindex="-1"></a><span class="in">cnetplot(goSEA, colorEdge = TRUE, cex_label_gene = 0.5)</span></span>
<span id="cb61-337"><a href="#cb61-337" aria-hidden="true" tabindex="-1"></a><span class="in">dotplot(goSEA)</span></span>
<span id="cb61-338"><a href="#cb61-338" aria-hidden="true" tabindex="-1"></a><span class="in">goSEA &lt;- pairwise_termsim(goSEA)</span></span>
<span id="cb61-339"><a href="#cb61-339" aria-hidden="true" tabindex="-1"></a><span class="in">treeplot(goSEA)</span></span>
<span id="cb61-340"><a href="#cb61-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-341"><a href="#cb61-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-342"><a href="#cb61-342" aria-hidden="true" tabindex="-1"></a><span class="fu"># Random forest classification</span></span>
<span id="cb61-343"><a href="#cb61-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-344"><a href="#cb61-344" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prepare data</span></span>
<span id="cb61-345"><a href="#cb61-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-348"><a href="#cb61-348" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-349"><a href="#cb61-349" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and prepare the data</span></span>
<span id="cb61-350"><a href="#cb61-350" aria-hidden="true" tabindex="-1"></a>data_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb61-351"><a href="#cb61-351" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data_raw <span class="sc">%&gt;%</span></span>
<span id="cb61-352"><a href="#cb61-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneSymbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-353"><a href="#cb61-353" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span></span>
<span id="cb61-354"><a href="#cb61-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GeneSymbol)) <span class="sc">%&gt;%</span></span>
<span id="cb61-355"><a href="#cb61-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"GeneSymbol"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-356"><a href="#cb61-356" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="dv">0</span>, .)) <span class="sc">%&gt;%</span></span>
<span id="cb61-357"><a href="#cb61-357" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(as.integer)</span>
<span id="cb61-358"><a href="#cb61-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-359"><a href="#cb61-359" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare metadata</span></span>
<span id="cb61-360"><a href="#cb61-360" aria-hidden="true" tabindex="-1"></a>metadata_raw <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb61-361"><a href="#cb61-361" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata_raw <span class="sc">%&gt;%</span></span>
<span id="cb61-362"><a href="#cb61-362" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Index =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb61-363"><a href="#cb61-363" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"Index"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb61-364"><a href="#cb61-364" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sample =</span> ID, <span class="at">disease =</span> Disease) <span class="sc">%&gt;%</span></span>
<span id="cb61-365"><a href="#cb61-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">gsub</span>(<span class="st">"_raw"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb61-366"><a href="#cb61-366" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, individual, disease) <span class="sc">%&gt;%</span></span>
<span id="cb61-367"><a href="#cb61-367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">factor</span>(individual), <span class="at">disease =</span> <span class="fu">factor</span>(disease))</span>
<span id="cb61-368"><a href="#cb61-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-369"><a href="#cb61-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure the data is in a matrix form with gene symbols as rownames</span></span>
<span id="cb61-370"><a href="#cb61-370" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data)</span>
<span id="cb61-371"><a href="#cb61-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-372"><a href="#cb61-372" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DESeqDataSet</span></span>
<span id="cb61-373"><a href="#cb61-373" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb61-374"><a href="#cb61-374" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> countData,</span>
<span id="cb61-375"><a href="#cb61-375" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata,</span>
<span id="cb61-376"><a href="#cb61-376" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> disease</span>
<span id="cb61-377"><a href="#cb61-377" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb61-378"><a href="#cb61-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-379"><a href="#cb61-379" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate size factors</span></span>
<span id="cb61-380"><a href="#cb61-380" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb61-381"><a href="#cb61-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-382"><a href="#cb61-382" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate gene-wise dispersions</span></span>
<span id="cb61-383"><a href="#cb61-383" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateDispersionsGeneEst</span>(dds)</span>
<span id="cb61-384"><a href="#cb61-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-385"><a href="#cb61-385" aria-hidden="true" tabindex="-1"></a><span class="co"># Directly use the gene-wise estimates</span></span>
<span id="cb61-386"><a href="#cb61-386" aria-hidden="true" tabindex="-1"></a><span class="fu">dispersions</span>(dds) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(dds)<span class="sc">$</span>dispGeneEst</span>
<span id="cb61-387"><a href="#cb61-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-388"><a href="#cb61-388" aria-hidden="true" tabindex="-1"></a>normCounts <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb61-389"><a href="#cb61-389" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-390"><a href="#cb61-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-393"><a href="#cb61-393" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-394"><a href="#cb61-394" aria-hidden="true" tabindex="-1"></a><span class="co">#Define model fitting procedure</span></span>
<span id="cb61-395"><a href="#cb61-395" aria-hidden="true" tabindex="-1"></a><span class="co">#NB: you will also need the packages statmod and e1071 for this analysis</span></span>
<span id="cb61-396"><a href="#cb61-396" aria-hidden="true" tabindex="-1"></a><span class="co">#New score function</span></span>
<span id="cb61-397"><a href="#cb61-397" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb61-398"><a href="#cb61-398" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">mean</span>(x)</span>
<span id="cb61-399"><a href="#cb61-399" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-400"><a href="#cb61-400" aria-hidden="true" tabindex="-1"></a><span class="co">#New filter function</span></span>
<span id="cb61-401"><a href="#cb61-401" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>filter <span class="ot">&lt;-</span> <span class="cf">function</span>(score, x, y){</span>
<span id="cb61-402"><a href="#cb61-402" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(data) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb61-403"><a href="#cb61-403" aria-hidden="true" tabindex="-1"></a>selection <span class="ot">&lt;-</span> score <span class="sc">&gt;</span> <span class="fu">quantile</span>(score, <span class="fl">0.9</span>) <span class="sc">&amp;</span> meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb61-404"><a href="#cb61-404" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(score) <span class="sc">%in%</span> <span class="fu">names</span>(score)[selection]</span>
<span id="cb61-405"><a href="#cb61-405" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-406"><a href="#cb61-406" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-407"><a href="#cb61-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-410"><a href="#cb61-410" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-411"><a href="#cb61-411" aria-hidden="true" tabindex="-1"></a><span class="co">#Create training set</span></span>
<span id="cb61-412"><a href="#cb61-412" aria-hidden="true" tabindex="-1"></a>trainingSet <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">assay</span>(normCounts)))</span>
<span id="cb61-413"><a href="#cb61-413" aria-hidden="true" tabindex="-1"></a><span class="co">#Train and validate decision tree (C5.0)</span></span>
<span id="cb61-414"><a href="#cb61-414" aria-hidden="true" tabindex="-1"></a>dtControl <span class="ot">&lt;-</span> <span class="fu">sbfControl</span>(</span>
<span id="cb61-415"><a href="#cb61-415" aria-hidden="true" tabindex="-1"></a>  <span class="at">functions =</span> rfSBF,</span>
<span id="cb61-416"><a href="#cb61-416" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"loocv"</span>,</span>
<span id="cb61-417"><a href="#cb61-417" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveDetails =</span> <span class="cn">TRUE</span>,</span>
<span id="cb61-418"><a href="#cb61-418" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb61-419"><a href="#cb61-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-420"><a href="#cb61-420" aria-hidden="true" tabindex="-1"></a>dtModel <span class="ot">&lt;-</span> <span class="fu">sbf</span>(</span>
<span id="cb61-421"><a href="#cb61-421" aria-hidden="true" tabindex="-1"></a>  trainingSet,</span>
<span id="cb61-422"><a href="#cb61-422" aria-hidden="true" tabindex="-1"></a>  metadata<span class="sc">$</span>disease,</span>
<span id="cb61-423"><a href="#cb61-423" aria-hidden="true" tabindex="-1"></a>  <span class="at">sbfControl =</span> dtControl)</span>
<span id="cb61-424"><a href="#cb61-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-425"><a href="#cb61-425" aria-hidden="true" tabindex="-1"></a><span class="co">#Show model</span></span>
<span id="cb61-426"><a href="#cb61-426" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(dtModel$fit)</span></span>
<span id="cb61-427"><a href="#cb61-427" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-428"><a href="#cb61-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-429"><a href="#cb61-429" aria-hidden="true" tabindex="-1"></a><span class="fu">## Performance evaluation</span></span>
<span id="cb61-430"><a href="#cb61-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-433"><a href="#cb61-433" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-434"><a href="#cb61-434" aria-hidden="true" tabindex="-1"></a><span class="co">#Print confusion matrix from validation</span></span>
<span id="cb61-435"><a href="#cb61-435" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(dtModel<span class="sc">$</span>pred<span class="sc">$</span>predictions<span class="sc">$</span>pred, metadata<span class="sc">$</span>disease)</span>
<span id="cb61-436"><a href="#cb61-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-437"><a href="#cb61-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-438"><a href="#cb61-438" aria-hidden="true" tabindex="-1"></a><span class="fu"># Spearman correlation analysis</span></span>
<span id="cb61-439"><a href="#cb61-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-440"><a href="#cb61-440" aria-hidden="true" tabindex="-1"></a>Loading the data for the correlaton analysis</span>
<span id="cb61-441"><a href="#cb61-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-444"><a href="#cb61-444" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-445"><a href="#cb61-445" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb61-446"><a href="#cb61-446" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/significant_genes.tsv"</span>)</span>
<span id="cb61-447"><a href="#cb61-447" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb61-448"><a href="#cb61-448" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-449"><a href="#cb61-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-450"><a href="#cb61-450" aria-hidden="true" tabindex="-1"></a>Performing Spearman correlation of the 63 upregulated genes on selected clinical variables of interest and comparing the correlation between Alzheimer's disease and control  samples.</span>
<span id="cb61-451"><a href="#cb61-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-452"><a href="#cb61-452" aria-hidden="true" tabindex="-1"></a><span class="in">```{r , fig.width=20, fig.height=10}</span></span>
<span id="cb61-453"><a href="#cb61-453" aria-hidden="true" tabindex="-1"></a><span class="in"># Function to create correlation data and plot</span></span>
<span id="cb61-454"><a href="#cb61-454" aria-hidden="true" tabindex="-1"></a><span class="in">create_correlation_plot &lt;- function(data, metadata, gene_symbols, variable, title) {</span></span>
<span id="cb61-455"><a href="#cb61-455" aria-hidden="true" tabindex="-1"></a><span class="in">  correlation_data &lt;- data %&gt;%</span></span>
<span id="cb61-456"><a href="#cb61-456" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(GeneSymbol %in% gene_symbols) %&gt;%</span></span>
<span id="cb61-457"><a href="#cb61-457" aria-hidden="true" tabindex="-1"></a><span class="in">    pivot_longer(cols = -GeneSymbol, names_to = "ID", values_to = "value") %&gt;%</span></span>
<span id="cb61-458"><a href="#cb61-458" aria-hidden="true" tabindex="-1"></a><span class="in">    right_join(metadata, by = "ID") %&gt;%</span></span>
<span id="cb61-459"><a href="#cb61-459" aria-hidden="true" tabindex="-1"></a><span class="in">    group_by(GeneSymbol, Disease) %&gt;%</span></span>
<span id="cb61-460"><a href="#cb61-460" aria-hidden="true" tabindex="-1"></a><span class="in">    summarise(correlation = cor(value, !!rlang::sym(variable), method = "spearman"), .groups = "drop") %&gt;%</span></span>
<span id="cb61-461"><a href="#cb61-461" aria-hidden="true" tabindex="-1"></a><span class="in">    ungroup()</span></span>
<span id="cb61-462"><a href="#cb61-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-463"><a href="#cb61-463" aria-hidden="true" tabindex="-1"></a><span class="in">  # Extract the order of genes based on their correlation in AD</span></span>
<span id="cb61-464"><a href="#cb61-464" aria-hidden="true" tabindex="-1"></a><span class="in">  ad_correlations &lt;- correlation_data %&gt;%</span></span>
<span id="cb61-465"><a href="#cb61-465" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(Disease == "AD") %&gt;%</span></span>
<span id="cb61-466"><a href="#cb61-466" aria-hidden="true" tabindex="-1"></a><span class="in">    arrange(desc(correlation)) %&gt;%</span></span>
<span id="cb61-467"><a href="#cb61-467" aria-hidden="true" tabindex="-1"></a><span class="in">    pull(GeneSymbol)</span></span>
<span id="cb61-468"><a href="#cb61-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-469"><a href="#cb61-469" aria-hidden="true" tabindex="-1"></a><span class="in">  # Ensure that all GeneSymbols are included, even those not present in AD data</span></span>
<span id="cb61-470"><a href="#cb61-470" aria-hidden="true" tabindex="-1"></a><span class="in">  full_gene_list &lt;- unique(correlation_data$GeneSymbol)</span></span>
<span id="cb61-471"><a href="#cb61-471" aria-hidden="true" tabindex="-1"></a><span class="in">  ad_order &lt;- unique(c(ad_correlations, full_gene_list)) # This ensures all genes are included and in order</span></span>
<span id="cb61-472"><a href="#cb61-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-473"><a href="#cb61-473" aria-hidden="true" tabindex="-1"></a><span class="in">  # Set the levels of GeneSymbol according to their order in AD</span></span>
<span id="cb61-474"><a href="#cb61-474" aria-hidden="true" tabindex="-1"></a><span class="in">  correlation_data$GeneSymbol &lt;- factor(correlation_data$GeneSymbol, levels = ad_order)</span></span>
<span id="cb61-475"><a href="#cb61-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-476"><a href="#cb61-476" aria-hidden="true" tabindex="-1"></a><span class="in">  # Plot the heatmap with genes ordered by their correlation in AD</span></span>
<span id="cb61-477"><a href="#cb61-477" aria-hidden="true" tabindex="-1"></a><span class="in">  heatmap_plot &lt;- ggplot(correlation_data, aes(x = Disease, y = GeneSymbol, fill = correlation)) +</span></span>
<span id="cb61-478"><a href="#cb61-478" aria-hidden="true" tabindex="-1"></a><span class="in">    geom_tile() +</span></span>
<span id="cb61-479"><a href="#cb61-479" aria-hidden="true" tabindex="-1"></a><span class="in">    scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) +</span></span>
<span id="cb61-480"><a href="#cb61-480" aria-hidden="true" tabindex="-1"></a><span class="in">    theme_minimal() +</span></span>
<span id="cb61-481"><a href="#cb61-481" aria-hidden="true" tabindex="-1"></a><span class="in">    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +</span></span>
<span id="cb61-482"><a href="#cb61-482" aria-hidden="true" tabindex="-1"></a><span class="in">    labs(fill = "Correlation", y = "Gene (ordered by AD correlation)", x = "Disease", title = title)</span></span>
<span id="cb61-483"><a href="#cb61-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-484"><a href="#cb61-484" aria-hidden="true" tabindex="-1"></a><span class="in">  return(heatmap_plot)</span></span>
<span id="cb61-485"><a href="#cb61-485" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb61-486"><a href="#cb61-486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-487"><a href="#cb61-487" aria-hidden="true" tabindex="-1"></a><span class="in"># Variables of interest</span></span>
<span id="cb61-488"><a href="#cb61-488" aria-hidden="true" tabindex="-1"></a><span class="in">variables_of_interest &lt;- c("Age", "PMI", "Mean_plaque", "CDR")</span></span>
<span id="cb61-489"><a href="#cb61-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-490"><a href="#cb61-490" aria-hidden="true" tabindex="-1"></a><span class="in"># Plot list</span></span>
<span id="cb61-491"><a href="#cb61-491" aria-hidden="true" tabindex="-1"></a><span class="in">plot_list &lt;- list()</span></span>
<span id="cb61-492"><a href="#cb61-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-493"><a href="#cb61-493" aria-hidden="true" tabindex="-1"></a><span class="in"># Loop over variables and create plots</span></span>
<span id="cb61-494"><a href="#cb61-494" aria-hidden="true" tabindex="-1"></a><span class="in">for (variable in variables_of_interest) {</span></span>
<span id="cb61-495"><a href="#cb61-495" aria-hidden="true" tabindex="-1"></a><span class="in">  plot_list[[variable]] &lt;- create_correlation_plot(data, metadata, de_results$GeneSymbol, variable, paste("Correlation with", variable))</span></span>
<span id="cb61-496"><a href="#cb61-496" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb61-497"><a href="#cb61-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-498"><a href="#cb61-498" aria-hidden="true" tabindex="-1"></a><span class="in"># Combine plots</span></span>
<span id="cb61-499"><a href="#cb61-499" aria-hidden="true" tabindex="-1"></a><span class="in">combined_plot &lt;- wrap_plots(plot_list, nrow = 1)</span></span>
<span id="cb61-500"><a href="#cb61-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-501"><a href="#cb61-501" aria-hidden="true" tabindex="-1"></a><span class="in"># Print the combined plot</span></span>
<span id="cb61-502"><a href="#cb61-502" aria-hidden="true" tabindex="-1"></a><span class="in">print(combined_plot)</span></span>
<span id="cb61-503"><a href="#cb61-503" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb61-504"><a href="#cb61-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-505"><a href="#cb61-505" aria-hidden="true" tabindex="-1"></a><span class="fu"># Session info</span></span>
<span id="cb61-506"><a href="#cb61-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-509"><a href="#cb61-509" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb61-510"><a href="#cb61-510" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb61-511"><a href="#cb61-511" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




<script src="group2_final_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>