<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.537">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Emil Johansson, Inika Prasad, Vaishnavi Divya Shridar &amp; Yuqi Zheng">

<title>Group 2 report: Characterization of Alzheimer’s disease</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="group2_final_files/libs/clipboard/clipboard.min.js"></script>
<script src="group2_final_files/libs/quarto-html/quarto.js"></script>
<script src="group2_final_files/libs/quarto-html/popper.min.js"></script>
<script src="group2_final_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="group2_final_files/libs/quarto-html/anchor.min.js"></script>
<link href="group2_final_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="group2_final_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="group2_final_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="group2_final_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="group2_final_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <div class="quarto-title-block"><div><h1 class="title">Group 2 report: Characterization of Alzheimer’s disease</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
            <p class="subtitle lead">A project for the course Statistics in Bioinformatics</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Author</div>
      <div class="quarto-title-meta-contents">
               <p>Emil Johansson, Inika Prasad, Vaishnavi Divya Shridar &amp; Yuqi Zheng </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">Saturday the 30th of November 2024</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a></li>
  <li><a href="#qc" id="toc-qc" class="nav-link" data-scroll-target="#qc"><span class="header-section-number">2</span> QC</a></li>
  <li><a href="#differential-expression-analysis" id="toc-differential-expression-analysis" class="nav-link" data-scroll-target="#differential-expression-analysis"><span class="header-section-number">3</span> Differential expression analysis</a>
  <ul>
  <li><a href="#design-matrix" id="toc-design-matrix" class="nav-link" data-scroll-target="#design-matrix"><span class="header-section-number">3.1</span> Design matrix</a></li>
  <li><a href="#hypothesis-correction" id="toc-hypothesis-correction" class="nav-link" data-scroll-target="#hypothesis-correction"><span class="header-section-number">3.2</span> Hypothesis correction</a></li>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot"><span class="header-section-number">3.3</span> Volcano plot</a></li>
  </ul></li>
  <li><a href="#protein-protein-interaction-analysis" id="toc-protein-protein-interaction-analysis" class="nav-link" data-scroll-target="#protein-protein-interaction-analysis"><span class="header-section-number">4</span> Protein-protein interaction analysis</a></li>
  <li><a href="#enrichment-analysis" id="toc-enrichment-analysis" class="nav-link" data-scroll-target="#enrichment-analysis"><span class="header-section-number">5</span> Enrichment analysis</a></li>
  <li><a href="#gsea-analysis" id="toc-gsea-analysis" class="nav-link" data-scroll-target="#gsea-analysis"><span class="header-section-number">6</span> GSEA Analysis</a></li>
  <li><a href="#incorrect-gsea-analysis" id="toc-incorrect-gsea-analysis" class="nav-link" data-scroll-target="#incorrect-gsea-analysis"><span class="header-section-number">7</span> Incorrect GSEA Analysis</a>
  <ul>
  <li><a href="#prepare-data" id="toc-prepare-data" class="nav-link" data-scroll-target="#prepare-data"><span class="header-section-number">7.1</span> Prepare data</a></li>
  <li><a href="#run-gsego" id="toc-run-gsego" class="nav-link" data-scroll-target="#run-gsego"><span class="header-section-number">7.2</span> Run gseGO</a></li>
  <li><a href="#plots" id="toc-plots" class="nav-link" data-scroll-target="#plots"><span class="header-section-number">7.3</span> Plots</a></li>
  <li><a href="#gsea-on-specific-clusters" id="toc-gsea-on-specific-clusters" class="nav-link" data-scroll-target="#gsea-on-specific-clusters"><span class="header-section-number">7.4</span> GSEA on specific clusters</a></li>
  </ul></li>
  <li><a href="#cytoscape" id="toc-cytoscape" class="nav-link" data-scroll-target="#cytoscape"><span class="header-section-number">8</span> CytoScape</a>
  <ul>
  <li><a href="#load-protein-list" id="toc-load-protein-list" class="nav-link" data-scroll-target="#load-protein-list"><span class="header-section-number">8.1</span> Load protein list</a></li>
  <li><a href="#initialize-string-db" id="toc-initialize-string-db" class="nav-link" data-scroll-target="#initialize-string-db"><span class="header-section-number">8.2</span> Initialize STRING db</a></li>
  <li><a href="#map-gene-names-to-string-aliases" id="toc-map-gene-names-to-string-aliases" class="nav-link" data-scroll-target="#map-gene-names-to-string-aliases"><span class="header-section-number">8.3</span> Map gene names to STRING aliases</a></li>
  <li><a href="#fetch-interaction-data" id="toc-fetch-interaction-data" class="nav-link" data-scroll-target="#fetch-interaction-data"><span class="header-section-number">8.4</span> Fetch interaction data</a></li>
  <li><a href="#create-dataframe-for-ppi-network" id="toc-create-dataframe-for-ppi-network" class="nav-link" data-scroll-target="#create-dataframe-for-ppi-network"><span class="header-section-number">8.5</span> Create dataframe for PPI network</a></li>
  <li><a href="#make-the-network-in-cytoscape" id="toc-make-the-network-in-cytoscape" class="nav-link" data-scroll-target="#make-the-network-in-cytoscape"><span class="header-section-number">8.6</span> Make the network in Cytoscape</a></li>
  <li><a href="#save-and-view-the-network" id="toc-save-and-view-the-network" class="nav-link" data-scroll-target="#save-and-view-the-network"><span class="header-section-number">8.7</span> Save and View the network</a></li>
  <li><a href="#cluster-ppi-network" id="toc-cluster-ppi-network" class="nav-link" data-scroll-target="#cluster-ppi-network"><span class="header-section-number">8.8</span> Cluster PPI network</a>
  <ul class="collapse">
  <li><a href="#glay-clustering" id="toc-glay-clustering" class="nav-link" data-scroll-target="#glay-clustering"><span class="header-section-number">8.8.1</span> GLay clustering</a></li>
  </ul></li>
  <li><a href="#save-network-info" id="toc-save-network-info" class="nav-link" data-scroll-target="#save-network-info"><span class="header-section-number">8.9</span> Save network info</a></li>
  <li><a href="#members-of-each-big-cluster" id="toc-members-of-each-big-cluster" class="nav-link" data-scroll-target="#members-of-each-big-cluster"><span class="header-section-number">8.10</span> Members of each big cluster</a></li>
  </ul></li>
  <li><a href="#enrichment-analysis-for-each-cluster-go-biological-process" id="toc-enrichment-analysis-for-each-cluster-go-biological-process" class="nav-link" data-scroll-target="#enrichment-analysis-for-each-cluster-go-biological-process"><span class="header-section-number">9</span> Enrichment analysis for each cluster (GO Biological Process)</a>
  <ul>
  <li><a href="#enrichment-analysis-for-each-cluster-go-molecular-function" id="toc-enrichment-analysis-for-each-cluster-go-molecular-function" class="nav-link" data-scroll-target="#enrichment-analysis-for-each-cluster-go-molecular-function"><span class="header-section-number">9.1</span> Enrichment analysis for each cluster (GO Molecular Function)</a></li>
  <li><a href="#enrichment-analysis-for-each-cluster-go-cellular-component" id="toc-enrichment-analysis-for-each-cluster-go-cellular-component" class="nav-link" data-scroll-target="#enrichment-analysis-for-each-cluster-go-cellular-component"><span class="header-section-number">9.2</span> Enrichment analysis for each cluster (GO Cellular Component)</a></li>
  </ul></li>
  <li><a href="#random-forest-classification" id="toc-random-forest-classification" class="nav-link" data-scroll-target="#random-forest-classification"><span class="header-section-number">10</span> Random forest classification</a>
  <ul>
  <li><a href="#prepare-data-1" id="toc-prepare-data-1" class="nav-link" data-scroll-target="#prepare-data-1"><span class="header-section-number">10.1</span> Prepare data</a></li>
  <li><a href="#performance-evaluation" id="toc-performance-evaluation" class="nav-link" data-scroll-target="#performance-evaluation"><span class="header-section-number">10.2</span> Performance evaluation</a></li>
  </ul></li>
  <li><a href="#spearman-correlation-analysis" id="toc-spearman-correlation-analysis" class="nav-link" data-scroll-target="#spearman-correlation-analysis"><span class="header-section-number">11</span> Spearman correlation analysis</a></li>
  <li><a href="#final-reflection-12-a4" id="toc-final-reflection-12-a4" class="nav-link" data-scroll-target="#final-reflection-12-a4"><span class="header-section-number">12</span> Final reflection (1/2 A4)</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info"><span class="header-section-number">13</span> Session info</a></li>
  </ul>
</nav>
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#    install.packages("BiocManager")</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("RCy3")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Libraries we need</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>libs <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tidyverse"</span>, <span class="st">"readxl"</span>, <span class="st">"limma"</span>, <span class="st">"ggrepel"</span>, <span class="st">"magrittr"</span>, <span class="st">"kableExtra"</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"patchwork"</span>, <span class="st">"DT"</span>, <span class="st">"tidymodels"</span>, <span class="st">"ggbeeswarm"</span>, <span class="st">"gt"</span>, <span class="st">"skimr"</span>, <span class="st">"GGally"</span>,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"visdat"</span>, <span class="st">"corrr"</span>, <span class="st">"ggsignif"</span>, <span class="st">"vip"</span>, <span class="st">"themis"</span>, <span class="st">"keras"</span>, <span class="st">"xgboost"</span>, <span class="st">"kknn"</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tensorflow"</span>, <span class="st">"xlsx"</span>, <span class="st">"HDAnalyzeR"</span>, <span class="st">"plotly"</span>, <span class="st">"umap"</span>, <span class="st">"uwot"</span>, <span class="st">"ggplotify"</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cowplot"</span>, <span class="st">"ggvenn"</span>, <span class="st">"UpSetR"</span>, <span class="st">"ComplexUpset"</span>, <span class="st">"DESeq2"</span>, <span class="st">"pheatmap"</span>, <span class="st">"randomForest"</span>,</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"caret"</span>, <span class="st">"edgeR"</span>, <span class="st">"STRINGdb"</span>, <span class="st">"igraph"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"enrichplot"</span>,</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"egg"</span>, <span class="st">"RCy3"</span>, <span class="st">"RColorBrewer"</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Install missing libraries</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>installed_libs <span class="ot">&lt;-</span> libs <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(installed_libs <span class="sc">==</span> <span class="cn">FALSE</span>)) {</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(libs[<span class="sc">!</span>installed_libs])</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(libs, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="qc" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> QC</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene expression count data</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>count_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">matches</span>(<span class="st">"_raw$"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Round counts to integers</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"C"</span>), <span class="fu">starts_with</span>(<span class="st">"A"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span><span class="fu">round</span>(.))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>genes_before_filtering <span class="ot">&lt;-</span> <span class="fu">nrow</span>(count_data_filtered)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of genes before filtering:"</span>, genes_before_filtering, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of genes before filtering: 19185 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> count_data_filtered)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(dge) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(meanLog2CPM, <span class="at">main =</span> <span class="st">"Histogram of Mean Log2 CPM"</span>, <span class="at">xlab =</span> <span class="st">"Mean Log2 CPM"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/filter%20low%20counts%20using%20mean%20log2%20CPM-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data_filtered[meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span>, ]</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>genes_after_filtering <span class="ot">&lt;-</span> <span class="fu">nrow</span>(count_data_filtered)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of genes after filtering:"</span>, genes_after_filtering, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of genes after filtering: 14950 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(count_data_filtered),</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^C"</span>, <span class="fu">colnames</span>(count_data_filtered)), <span class="st">"HC"</span>, <span class="st">"AD"</span>),</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>)  <span class="co"># Manually adding batch information</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(<span class="at">counts =</span> count_data_filtered)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># design &lt;- model.matrix(~ condition + batch, data = sample_conditions)</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition <span class="sc">+</span> <span class="fu">factor</span>(batch), <span class="at">data =</span> sample_conditions)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">voom</span>(dge, design, <span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/normalization%20with%20Voom-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(v<span class="sc">$</span>E)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     C1_raw   C2_raw   C3_raw   C4_raw   C5_raw   C6_raw   C7_raw   C8_raw
1  1.666028 3.622681 2.402353 2.242940 2.175301 1.945160 2.220551 2.047241
2  8.171797 7.319737 8.634353 8.026213 9.276020 8.897405 7.848728 7.295732
12 6.987940 2.022997 3.397751 3.657259 6.019456 7.878962 2.560113 3.316176
14 6.421959 6.029584 6.000448 6.510198 5.319996 6.775631 6.584768 6.530001
16 8.762372 8.293814 8.435144 8.724345 7.865274 8.882593 8.895612 8.728100
18 8.144870 7.275008 8.455469 8.478941 8.341384 8.366975 8.453630 8.680521
     A9_raw  A10_raw  A11_raw  A12_raw  A13_raw  A14_raw  A15_raw  A16_raw
1  1.906348 1.627370 2.073706 2.429696 2.914332 2.243341 2.639876 2.290996
2  8.228932 8.609914 7.917652 8.579893 8.510049 8.863482 8.725027 8.051962
12 4.499693 5.975793 3.873325 6.067364 7.633415 7.161695 5.411179 8.052767
14 6.392968 6.333542 6.492932 6.165991 5.714636 6.341330 6.336129 6.563178
16 8.703734 8.418269 8.892265 8.288686 8.648926 8.520732 8.270540 8.844795
18 8.301662 8.294551 8.288104 8.189206 8.289339 8.255371 8.281214 8.574186
    A17_raw
1  2.772543
2  8.536880
12 5.083795
14 6.130582
16 8.484317
18 8.402370</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(v<span class="sc">$</span>E, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">main =</span> <span class="st">"Distribution of Voom Normalized Counts"</span>, <span class="at">xlab =</span> <span class="st">"Voom Normalized Counts"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/normalization%20with%20Voom-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dds &lt;- DESeqDataSetFromMatrix(countData = count_data_filtered, </span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                               colData = sample_conditions, </span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                               design = ~ condition + factor(batch))</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Normalize counts using variance stabilizing transformation (vst)</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># normCounts &lt;- vst(dds, blind = TRUE)</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co"># assay(normCounts)[1:5, 1:5]</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(assay(normCounts), breaks = 50, main = "Distribution of VST Normalized Counts", xlab = "Normalized Counts")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(v<span class="sc">$</span>E)  <span class="co"># Convert v$E to a data frame</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>voom_normalized<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(voom_normalized)  <span class="co"># Add GeneID as a column</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>gene_info_voom <span class="ot">&lt;-</span> <span class="fu">merge</span>(voom_normalized, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>gene_info_voom <span class="ot">&lt;-</span> gene_info_voom[, <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>, <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_info_voom), <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>)))]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(gene_info_voom, file = "voom_normalized.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># sampleDist &lt;- cor(assay(normCounts), method = "spearman")</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">cor</span>(v<span class="sc">$</span>E, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>sampleColor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Accent"</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sampleColor) <span class="ot">&lt;-</span> <span class="fu">unique</span>(sample_conditions<span class="sc">$</span>condition)</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDist,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist),</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist),</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Condition =</span> sample_conditions<span class="sc">$</span>condition, <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample),</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> <span class="fu">list</span>(<span class="at">Storage =</span> sampleColor),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Sample Correlation Heatmap by Condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20condition%20heatmap-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pcaRes &lt;- prcomp(t(assay(normCounts)))</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>pcaRes <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(v<span class="sc">$</span>E))</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>varExp <span class="ot">&lt;-</span> <span class="fu">round</span>(pcaRes<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pcaRes<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>pcaDF <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pcaRes<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pcaRes<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">Condition =</span> sample_conditions<span class="sc">$</span>condition,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sample =</span> sample_conditions<span class="sc">$</span>sample</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>pcaPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> pcaDF,</span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Condition, <span class="at">label =</span> Sample)</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, varExp[<span class="dv">1</span>], <span class="st">" %)"</span>) ,</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, varExp[<span class="dv">2</span>], <span class="st">" %)"</span>) ,</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Plot of Samples"</span>) <span class="sc">+</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Accent"</span>)) <span class="sc">+</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>)  <span class="co"># Added fixed ratio to ensure equal scaling between PC1 and PC2</span></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pcaPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20PCA%20by%20condition-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a heatmap showing Batch 1 and Batch 2 samples separately</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>sampleDist_batch <span class="ot">&lt;-</span> <span class="fu">cor</span>(v<span class="sc">$</span>E, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>batchColor <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Set2"</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(batchColor) <span class="ot">&lt;-</span> <span class="fu">unique</span>(sample_conditions<span class="sc">$</span>batch)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(sampleDist_batch,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_rows =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist_batch),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">clustering_distance_cols =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> sampleDist_batch),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Batch =</span> <span class="fu">factor</span>(sample_conditions<span class="sc">$</span>batch), <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample),</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> <span class="fu">list</span>(<span class="at">Batch =</span> batchColor),</span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"Sample Correlation Heatmap by Batch"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20batch%20heatmap-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA Plot by Batch</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>pcaRes_batch <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(v<span class="sc">$</span>E))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>varExp_batch <span class="ot">&lt;-</span> <span class="fu">round</span>(pcaRes_batch<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span> <span class="sc">/</span> <span class="fu">sum</span>(pcaRes_batch<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>pcaDF_batch <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC1 =</span> pcaRes_batch<span class="sc">$</span>x[, <span class="dv">1</span>],</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">PC2 =</span> pcaRes_batch<span class="sc">$</span>x[, <span class="dv">2</span>],</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">Batch =</span> <span class="fu">factor</span>(sample_conditions<span class="sc">$</span>batch),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">Sample =</span> sample_conditions<span class="sc">$</span>sample</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>pcaPlot_batch <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> pcaDF_batch,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Batch, <span class="at">label =</span> Sample)</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">paste0</span>(<span class="st">"PC1 ("</span>, varExp_batch[<span class="dv">1</span>], <span class="st">" %)"</span>),</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">paste0</span>(<span class="st">"PC2 ("</span>, varExp_batch[<span class="dv">2</span>], <span class="st">" %)"</span>),</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">"PCA Plot of Samples by Batch"</span>) <span class="sc">+</span></span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>), <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">brewer.pal</span>(<span class="dv">3</span>, <span class="st">"Set2"</span>)) <span class="sc">+</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pcaPlot_batch)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/QC-%20PCA%20by%20Batch-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove identified outliers</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co"># outliers &lt;- c("C1_raw", "C2_raw", "C4_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>outliers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"C1_raw"</span>, <span class="st">"C5_raw"</span>, <span class="st">"C6_raw"</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>remaining_samples <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">colnames</span>(count_data_filtered), outliers)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>count_data_filtered <span class="ot">&lt;-</span> count_data_filtered[, remaining_samples]</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> sample_conditions[<span class="sc">!</span>sample_conditions<span class="sc">$</span>sample <span class="sc">%in%</span> outliers, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="differential-expression-analysis" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Differential expression analysis</h1>
<section id="design-matrix" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="design-matrix"><span class="header-section-number">3.1</span> Design matrix</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define design matrix including batch effect</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co"># designMatrix &lt;- model.matrix(~ 0 + condition + batch, data = sample_conditions)</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>designMatrix <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> condition <span class="sc">+</span> <span class="fu">factor</span>(batch), <span class="at">data =</span> sample_conditions)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(designMatrix) <span class="ot">&lt;-</span> <span class="fu">make.names</span>(<span class="fu">colnames</span>(designMatrix))  <span class="co"># Ensure valid column names for contrast matrix</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Design Matrix with Batch Effect:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Design Matrix with Batch Effect:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(designMatrix))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>  conditionAD conditionHC factor.batch.1
2           0           1              0
3           0           1              0
4           0           1              1
7           0           1              1
8           0           1              1
9           1           0              0</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define contrast matrix</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>contrastMatrix <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(<span class="at">AD_vs_HC =</span> conditionAD <span class="sc">-</span> conditionHC, <span class="at">levels =</span> designMatrix)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Contrast Matrix:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Contrast Matrix:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(contrastMatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                Contrasts
Levels           AD_vs_HC
  conditionAD           1
  conditionHC          -1
  factor.batch.1        0</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DGEList and estimate dispersion</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">DGEList</span>(count_data_filtered)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, designMatrix, <span class="at">robust =</span> <span class="cn">TRUE</span>)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Fit the model using likelihood ratio test (LRT), now including batch in the design</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmFit</span>(dge, designMatrix)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">glmLRT</span>(fit, <span class="at">contrast =</span> contrastMatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="hypothesis-correction" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="hypothesis-correction"><span class="header-section-number">3.2</span> Hypothesis correction</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">topTags</span>(lrt, <span class="at">n =</span> <span class="fu">nrow</span>(count_data_filtered))</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>table<span class="sc">$</span>FDR <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res<span class="sc">$</span>table<span class="sc">$</span>PValue, <span class="at">method =</span> <span class="st">"BH"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Summary of p-values:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Summary of p-values:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(res<span class="sc">$</span>table<span class="sc">$</span>PValue))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  0.1672  0.4128  0.4423  0.7062  1.0000 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Summary of FDRs:</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Summary of FDRs:</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">summary</span>(res<span class="sc">$</span>table<span class="sc">$</span>FDR))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>     Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
0.0000056 0.6683192 0.8253958 0.7729789 0.9415187 1.0000000 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>sigRes <span class="ot">&lt;-</span> <span class="fu">subset</span>(res<span class="sc">$</span>table, FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Number of significant genes:"</span>, <span class="fu">nrow</span>(sigRes), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Number of significant genes: 63 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(sigRes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">logFC</th>
<th style="text-align: right;">logCPM</th>
<th style="text-align: right;">LR</th>
<th style="text-align: right;">PValue</th>
<th style="text-align: right;">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1356</td>
<td style="text-align: right;">2.029851</td>
<td style="text-align: right;">5.348685</td>
<td style="text-align: right;">39.25040</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000056</td>
</tr>
<tr class="even">
<td style="text-align: left;">960</td>
<td style="text-align: right;">2.024781</td>
<td style="text-align: right;">5.760368</td>
<td style="text-align: right;">32.46153</td>
<td style="text-align: right;">0e+00</td>
<td style="text-align: right;">0.0000909</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5920</td>
<td style="text-align: right;">1.133116</td>
<td style="text-align: right;">3.334786</td>
<td style="text-align: right;">28.76463</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="even">
<td style="text-align: left;">10437</td>
<td style="text-align: right;">1.655241</td>
<td style="text-align: right;">2.208464</td>
<td style="text-align: right;">28.46395</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7098</td>
<td style="text-align: right;">1.032912</td>
<td style="text-align: right;">1.682106</td>
<td style="text-align: right;">27.89196</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
<tr class="even">
<td style="text-align: left;">929</td>
<td style="text-align: right;">1.585105</td>
<td style="text-align: right;">3.429787</td>
<td style="text-align: right;">27.75458</td>
<td style="text-align: right;">1e-07</td>
<td style="text-align: right;">0.0003432</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>sigRes<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sigRes)  <span class="co"># Add GeneID as a column to the results</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">merge</span>(sigRes, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> gene_info[, <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>, <span class="fu">setdiff</span>(<span class="fu">names</span>(gene_info), <span class="fu">c</span>(<span class="st">"GeneID"</span>, <span class="st">"GeneSymbol"</span>)))]</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(gene_info, file = "sig_genes.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="volcano-plot" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="volcano-plot"><span class="header-section-number">3.3</span> Volcano plot</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add gene symbols to the results</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>count_data_with_symbols <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> count_data_with_symbols[, <span class="st">"GeneSymbol"</span>, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>res<span class="sc">$</span>table<span class="sc">$</span>GeneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res<span class="sc">$</span>table)</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>gene_info_all <span class="ot">&lt;-</span> <span class="fu">merge</span>(res<span class="sc">$</span>table, gene_symbols, <span class="at">by.x =</span> <span class="st">"GeneID"</span>, <span class="at">by.y =</span> <span class="st">"row.names"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Volcano plot using gene symbols for all genes</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>volcanoPlot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gene_info_all, <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(FDR),</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">color =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>, <span class="st">"darkred"</span>, <span class="st">"grey"</span>),</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">label =</span> <span class="fu">ifelse</span>(FDR <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="dv">1</span>, GeneSymbol, <span class="st">""</span>))) <span class="sc">+</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="st">"Fold Change, Log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">""</span>)) <span class="sc">+</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">expression</span>(<span class="st">"Adjusted P value, Log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">""</span>)) <span class="sc">+</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">1</span>), <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">linewidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>) <span class="sc">+</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkred"</span>, <span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">max.overlaps =</span> <span class="dv">10</span>)</span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcanoPlot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/visualize%20results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="protein-protein-interaction-analysis" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Protein-protein interaction analysis</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize STRINGdb</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>(<span class="at">version =</span> <span class="st">"11"</span>, <span class="at">species =</span> <span class="dv">9606</span>, <span class="at">score_threshold =</span> <span class="dv">400</span>, <span class="at">input_directory =</span> <span class="st">""</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> gene_info<span class="sc">$</span>GeneSymbol</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">GeneSymbol =</span> gene_symbols, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>mapped_genes <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(gene_symbols, <span class="st">"GeneSymbol"</span>, <span class="at">removeUnmappedRows =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Warning:  we couldn't map to STRING 1% of your identifiers</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get PPI Network for the mapped genes</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>ppi_network <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_interactions</span>(mapped_genes<span class="sc">$</span>STRING_id)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>string_db<span class="sc">$</span><span class="fu">plot_network</span>(mapped_genes<span class="sc">$</span>STRING_id)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Parameter add_link not available in version 11.0 (please use 11.0b or later)"</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/protein-protein%20interation%20analysis-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>gene_symbols_vector <span class="ot">&lt;-</span> gene_symbols<span class="sc">$</span>GeneSymbol  <span class="co"># Extract the gene symbols as a vector</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>mapped_genes_entrez <span class="ot">&lt;-</span> <span class="fu">bitr</span>(gene_symbols_vector,</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fromType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">toType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">OrgDb =</span> org.Hs.eg.db)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a>goSEA <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> mapped_genes_entrez<span class="sc">$</span>ENTREZID,</span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">keyType =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a>                  <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="enrichment-analysis" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Enrichment analysis</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(goSEA, <span class="at">colorEdge =</span> <span class="cn">TRUE</span>, <span class="at">cex_label_gene =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(goSEA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>goSEA <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(goSEA)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">treeplot</span>(goSEA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/Visualize%20GO%20SEA%20enrichment%20results-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import data</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/significant_genes.tsv'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="gsea-analysis" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> GSEA Analysis</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(all_genes<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(all_genes)) <span class="co"># gene ids</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co">#processed_genes_for_gse</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-22"><a href="#cb53-22" aria-hidden="true" tabindex="-1"></a>ego</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>#
# Gene Set Enrichment Analysis
#
#...@organism    Homo sapiens 
#...@setType     BP 
#...@keytype     ENTREZID 
#...@geneList    Named num [1:14947] 3.11 2.67 2.63 2.6 2.42 ...
 - attr(*, "names")= chr [1:14947] "12" "4057" "1117" "80832" ...
#...nPerm    1000 
#...pvalues adjusted by 'none' with cutoff &lt;0.05 
#...2343 enriched terms found
'data.frame':   2343 obs. of  11 variables:
 $ ID             : chr  "GO:0044283" "GO:0048729" "GO:0030855" "GO:0050865" ...
 $ Description    : chr  "small molecule biosynthetic process" "tissue morphogenesis" "epithelial cell differentiation" "regulation of cell activation" ...
 $ setSize        : int  499 485 479 489 481 467 473 420 467 474 ...
 $ enrichmentScore: num  0.377 0.325 0.341 0.472 0.332 ...
 $ NES            : num  1.77 1.53 1.6 2.22 1.56 ...
 $ pvalue         : num  0.00136 0.00136 0.00136 0.00136 0.00136 ...
 $ p.adjust       : num  0.00136 0.00136 0.00136 0.00136 0.00136 ...
 $ qvalue         : num  0.0261 0.0261 0.0261 0.0261 0.0261 ...
 $ rank           : num  3114 3112 2089 2021 3066 ...
 $ leading_edge   : chr  "tags=32%, list=21%, signal=26%" "tags=29%, list=21%, signal=24%" "tags=24%, list=14%, signal=21%" "tags=28%, list=14%, signal=25%" ...
 $ core_enrichment: chr  "5724/834/51302/241/972/344/27306/5354/301/341/8942/1109/653/2247/8644/6916/1066/79993/55304/3988/10170/51816/17"| __truncated__ "960/6781/187/25932/388389/7046/9021/57664/255743/26508/7855/653/3371/9953/7052/4313/1435/2247/4804/8324/6423/52"| __truncated__ "6781/3383/5920/1026/27284/25932/2302/7132/4615/10855/135228/301/7855/653/2247/8644/8324/1066/2828/5241/7431/126"| __truncated__ "6347/3127/2214/7412/57817/5724/170692/2207/54209/11326/3600/3123/3113/10875/199/3122/8673/1026/972/3111/2302/46"| __truncated__ ...
#...Citation
 T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan, X Fu, S Liu, X Bo, and G Yu.
 clusterProfiler 4.0: A universal enrichment tool for interpreting omics data.
 The Innovation. 2021, 2(3):100141 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/gsea_on_all-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"GSEA For Biological Processes in Alzheimer's Disease"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/gsea_on_all-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>count_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">matches</span>(<span class="st">"_raw$"</span>))</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>subset_count_data <span class="ot">&lt;-</span> count_data[<span class="fu">rownames</span>(count_data) <span class="sc">%in%</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>GeneID),]</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">'data/ad_metadata.xlsx'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">cor</span>(subset_count_data, <span class="at">method =</span> <span class="st">"spearman"</span>) </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(subset_count_data),</span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^C"</span>, <span class="fu">colnames</span>(subset_count_data)), <span class="st">"HC"</span>, <span class="st">"AD"</span>),</span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>),  <span class="do">## Manually adding batch information; batch effect is a bit serious as otherwise no gene will be identified as significant</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">plaque =</span> metadata<span class="sc">$</span>BM9_plaque</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>sample_conditions</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">sample</th>
<th style="text-align: left;">condition</th>
<th style="text-align: right;">batch</th>
<th style="text-align: right;">plaque</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">C1_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">C2_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C3_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">C4_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C5_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">C6_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">C7_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="even">
<td style="text-align: left;">C8_raw</td>
<td style="text-align: left;">HC</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A9_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">A10_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">27.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A11_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">17.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">A12_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">11.2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A13_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">7.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">A14_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">5.6</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A15_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">13.6</td>
</tr>
<tr class="even">
<td style="text-align: left;">A16_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">16.0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">A17_raw</td>
<td style="text-align: left;">AD</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">13.7</td>
</tr>
</tbody>
</table>
</div>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>heat_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">condition =</span> <span class="fu">c</span>(<span class="st">"HC"</span> <span class="ot">=</span> <span class="st">"#01bfc5"</span>, <span class="st">"AD"</span> <span class="ot">=</span> <span class="st">"salmon"</span>))</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="at">mat =</span> sampleDist,</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Plaque =</span> sample_conditions<span class="sc">$</span>plaque, <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="incorrect-gsea-analysis" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Incorrect GSEA Analysis</h1>
<p>The below analysis does not follow the expected steps! TO AVOID.</p>
<section id="prepare-data" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="prepare-data"><span class="header-section-number">7.1</span> Prepare data</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create input vector for gene enrichment</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>significant_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(significant_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>GeneID) <span class="co"># gene ids</span></span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>significant_genes_for_gse <span class="ot">&lt;-</span> significant_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>significant_genes_for_gse)] <span class="co"># order ranked</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-gsego" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="run-gsego"><span class="header-section-number">7.2</span> Run gseGO</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology ALL</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>ego_all <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList   =</span> significant_genes_for_gse,</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"ALL"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology MF</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>ego1 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"MF"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="do">## gave 0 enriched genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology CC</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>ego2 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"CC"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology BP</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>ego3 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) <span class="co"># “holm”, “hochberg”, “hommel”, “bonferroni”, “BH”, “BY”, “fdr”, “none”</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># compare the processes and components identified as signifcant between the ontologies</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="co">#ego3$Description</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="co">#ego2$Description</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a><span class="co">#ego_all$Description</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a><span class="co">#setdiff(ego2$Description, ego_all$Description)</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a><span class="co">#setdiff(ego3$Description, ego_all$Description) # "cellular response to stress" disappears from list</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="plots" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="plots"><span class="header-section-number">7.3</span> Plots</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego2, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"GSEA For Cellular Components in Alzheimer's Disease"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego3, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"GSEA For Biological Processes in Alzheimer's Disease"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">emapplot</span>(<span class="fu">pairwise_termsim</span>(ego3), <span class="at">showCategory =</span> <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="do">## compare the different plots</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>d_all <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(ego_all, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>)) </span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>d2 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(ego2, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign)<span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>)) </span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>d3 <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(ego3, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign)<span class="sc">+</span>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>)) </span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(d_all, d2, d3, </span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"ALL"</span>, <span class="st">"CC"</span>, <span class="st">"BP"</span>),</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">nrow =</span> <span class="dv">3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ridgeplot</span>(ego3) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"enrichment distribution"</span>) <span class="sc">+</span>   <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot</span>(ego3, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> ego3<span class="sc">$</span>Description[<span class="dv">1</span>], <span class="at">geneSetID =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot</span>(ego3, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> ego3<span class="sc">$</span>Description[<span class="dv">2</span>], <span class="at">geneSetID =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="gsea-on-specific-clusters" class="level2" data-number="7.4">
<h2 data-number="7.4" class="anchored" data-anchor-id="gsea-on-specific-clusters"><span class="header-section-number">7.4</span> GSEA on specific clusters</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster1.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb74-10"><a href="#cb74-10" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb74-11"><a href="#cb74-11" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb74-12"><a href="#cb74-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb74-13"><a href="#cb74-13" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb74-14"><a href="#cb74-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-15"><a href="#cb74-15" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb74-16"><a href="#cb74-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb74-17"><a href="#cb74-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb74-18"><a href="#cb74-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb74-19"><a href="#cb74-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb74-20"><a href="#cb74-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb74-21"><a href="#cb74-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb74-22"><a href="#cb74-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb74-23"><a href="#cb74-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb74-24"><a href="#cb74-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb74-25"><a href="#cb74-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-26"><a href="#cb74-26" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster2.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb75-12"><a href="#cb75-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb75-13"><a href="#cb75-13" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb75-14"><a href="#cb75-14" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>   51338    10437     2214    58475      929     4069      713     5724 
1.662620 1.655218 1.646171 1.612562 1.585090 1.469628 1.439121 1.434389 
   64231     1230     2207    10581     3113    10875     1536     5920 
1.407308 1.404221 1.345758 1.242926 1.234730 1.215954 1.156830 1.133084 
    4688      972 
1.104455 1.098473 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb77-10"><a href="#cb77-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb77-11"><a href="#cb77-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-12"><a href="#cb77-12" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster5.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="cytoscape" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> CytoScape</h1>
<p>We will investigate protein-protein interactions (PPIs) for a given list of proteins. To do so, we will retreive PPI data from the STRING database and visualize networks in Cytoscape using the R package RCy3. We will further cluster the network to investigate whether subsets of proteins share a particular molecular function.</p>
<p>Requirements: Cytoscape running, with the clusterMaker app installed.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm that Cytoscape is running</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cytoscapeVersionInfo</span> ()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>      apiVersion cytoscapeVersion 
            "v1"         "3.10.3" </code></pre>
</div>
</div>
<section id="load-protein-list" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="load-protein-list"><span class="header-section-number">8.1</span> Load protein list</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>proteins <span class="ot">&lt;-</span> <span class="fu">readLines</span>(<span class="st">"data/gene_symbols_list.txt"</span>)</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(proteins)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "IFI30"    "IFITM2"   "FGL2"     "STON1"    "FAM129A"  "SERPINA3"</code></pre>
</div>
</div>
</section>
<section id="initialize-string-db" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="initialize-string-db"><span class="header-section-number">8.2</span> Initialize STRING db</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize STRINGdb object </span></span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>(<span class="at">version=</span><span class="st">"12.0"</span>, </span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">species=</span><span class="dv">9606</span>, <span class="co"># Homo sapiens</span></span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">score_threshold=</span><span class="dv">400</span>, </span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">input_directory=</span><span class="st">""</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="map-gene-names-to-string-aliases" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="map-gene-names-to-string-aliases"><span class="header-section-number">8.3</span> Map gene names to STRING aliases</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dataframe with uppercase letters (to match with STRING alias mapping)</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">query.term =</span> <span class="fu">toupper</span>(proteins)) </span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="co"># check if file with mapping to STRING aliases exists</span></span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a><span class="co"># and create it if it doesn't</span></span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="st">"data/mapped2STRING.csv"</span>)) {</span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>  mapped <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/mapped2STRING.csv"</span>)</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>mapped <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(<span class="at">my_data_frame =</span> data, </span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">my_data_frame_id_col_names =</span> <span class="st">"query.term"</span>, </span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>                        <span class="at">takeFirst=</span><span class="cn">TRUE</span>, </span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>                        <span class="at">removeUnmappedRows=</span><span class="cn">TRUE</span>, </span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quiet=</span><span class="cn">FALSE</span>)</span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="at">x =</span> mapped, <span class="at">file =</span> <span class="st">"data/mapped2STRING.csv"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mapped)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">query.term</th>
<th style="text-align: left;">STRING_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">IFI30</td>
<td style="text-align: left;">9606.ENSP00000384886</td>
</tr>
<tr class="even">
<td style="text-align: left;">IFITM2</td>
<td style="text-align: left;">9606.ENSP00000484689</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FGL2</td>
<td style="text-align: left;">9606.ENSP00000248598</td>
</tr>
<tr class="even">
<td style="text-align: left;">STON1</td>
<td style="text-align: left;">9606.ENSP00000384615</td>
</tr>
<tr class="odd">
<td style="text-align: left;">FAM129A</td>
<td style="text-align: left;">9606.ENSP00000356481</td>
</tr>
<tr class="even">
<td style="text-align: left;">SERPINA3</td>
<td style="text-align: left;">9606.ENSP00000450540</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how many proteins were not mapped</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Since some gene IDs can map to several STRING identifiers, we account for duplicates </span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(data<span class="sc">$</span>query.term)) <span class="sc">-</span> <span class="fu">length</span>(<span class="fu">unique</span>(mapped<span class="sc">$</span>query.term))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1</code></pre>
</div>
</div>
</section>
<section id="fetch-interaction-data" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="fetch-interaction-data"><span class="header-section-number">8.4</span> Fetch interaction data</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get interactions for the mapped proteins if the file isn't already there</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a><span class="co"># check if file with interaction data exists </span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="co"># and create it if it doesn't</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="st">"data/interactions.csv"</span>)) {</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  interactions <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/interactions.csv"</span>)</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>interactions <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_interactions</span>(mapped<span class="sc">$</span>STRING_id)</span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="at">x =</span> interactions, <span class="at">file =</span> <span class="st">"data/interactions.csv"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a><span class="co"># View the first few interactions </span></span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(interactions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">from</th>
<th style="text-align: left;">to</th>
<th style="text-align: right;">combined_score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000216117</td>
<td style="text-align: right;">435</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000216117</td>
<td style="text-align: right;">435</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000009530</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: right;">497</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000009530</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: right;">497</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: right;">768</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: right;">768</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="create-dataframe-for-ppi-network" class="level2" data-number="8.5">
<h2 data-number="8.5" class="anchored" data-anchor-id="create-dataframe-for-ppi-network"><span class="header-section-number">8.5</span> Create dataframe for PPI network</h2>
<p>Define nodes for the network</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get node columns for the Cytoscape network</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">c</span>(interactions<span class="sc">$</span>from, interactions<span class="sc">$</span>to)))</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with the mapped protein names to include original protein names as labels</span></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">merge</span>(nodes, mapped[, <span class="fu">c</span>(<span class="st">"STRING_id"</span>, <span class="st">"query.term"</span>)], </span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">by.x =</span> <span class="st">"id"</span>, <span class="at">by.y =</span> <span class="st">"STRING_id"</span>, </span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">all.x =</span> <span class="cn">TRUE</span>, </span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nodes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">id</th>
<th style="text-align: left;">query.term</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000009530</td>
<td style="text-align: left;">CD74</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">TNFRSF1A</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000206423</td>
<td style="text-align: left;">CCDC80</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000209929</td>
<td style="text-align: left;">FMO2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000216117</td>
<td style="text-align: left;">HMOX1</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: left;">CCL2</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Define edges for the network</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">source =</span> interactions<span class="sc">$</span>from, <span class="at">target =</span> interactions<span class="sc">$</span>to)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove directionality of edges (necessary for various clusterings later)</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and sort the columns, then get unique rows</span></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>unique_edges <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">t</span>(<span class="fu">apply</span>(edges, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">sort</span>(x))))</span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert back to a dataframe</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(unique_edges, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the columns if needed</span></span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(edges)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">source</th>
<th style="text-align: left;">target</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000216117</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000009530</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000162749</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000216117</td>
<td style="text-align: left;">9606.ENSP00000225831</td>
</tr>
<tr class="odd">
<td style="text-align: left;">9606.ENSP00000225831</td>
<td style="text-align: left;">9606.ENSP00000245907</td>
</tr>
<tr class="even">
<td style="text-align: left;">9606.ENSP00000245907</td>
<td style="text-align: left;">9606.ENSP00000246006</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="make-the-network-in-cytoscape" class="level2" data-number="8.6">
<h2 data-number="8.6" class="anchored" data-anchor-id="make-the-network-in-cytoscape"><span class="header-section-number">8.6</span> Make the network in Cytoscape</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Connect to Cytoscape</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cytoscapePing</span>()</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new Cytoscape network from your data</span></span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a><span class="fu">createNetworkFromDataFrames</span>(<span class="at">nodes =</span> nodes, </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">edges =</span> edges, </span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># title = mytitle, </span></span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">collection =</span> <span class="st">"My Collection"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>networkSUID 
       2228 </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set node labels to the original protein names, and other visual tweaks</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeLabelMapping</span>(<span class="st">'query.term'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>NULL</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeShapeDefault</span>(<span class="st">'ELLIPSE'</span>)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeColorDefault</span>(<span class="st">'#9fbcda'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Now we modify the network layout in the app to reduce overlap of nodes and make it overall more aesthetically pleasing. Parameters will depend on the network</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># using force-directed layout. Higher coefficients means nodes are closer together. </span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">layoutNetwork</span>(<span class="st">'force-directed defaultSpringCoefficient=0.000006 defaultSpringLength=1'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="save-and-view-the-network" class="level2" data-number="8.7">
<h2 data-number="8.7" class="anchored" data-anchor-id="save-and-view-the-network"><span class="header-section-number">8.7</span> Save and View the network</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure network fits in the frame to be saved </span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">fitContent</span>()</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co">#save network image</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a><span class="fu">exportImage</span>(<span class="at">filename =</span> <span class="st">"currentnetwork.png"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                                                             file 
"/Users/emiljohansson/R/MedBioinfo/func-omics/currentnetwork.png" </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># view network image</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"currentnetwork.png"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="currentnetwork.png" class="img-fluid figure-img" width="398"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="cluster-ppi-network" class="level2" data-number="8.8">
<h2 data-number="8.8" class="anchored" data-anchor-id="cluster-ppi-network"><span class="header-section-number">8.8</span> Cluster PPI network</h2>
<p>commands specify: restoreEdges: restores edges after clustering, showUI: displays the new network, and undirectedEdges: assumes edges are undirected</p>
<section id="glay-clustering" class="level3" data-number="8.8.1">
<h3 data-number="8.8.1" class="anchored" data-anchor-id="glay-clustering"><span class="header-section-number">8.8.1</span> GLay clustering</h3>
<p>Run GLay community clustering</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run GLay community clustering</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>RCy3<span class="sc">::</span><span class="fu">commandsRun</span>(<span class="st">"cluster glay restoreEdges= true showUI = true undirectedEdges = true"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Clusters: 14"        "Average size: 4,429" "Maximum size: 18"   
[4] "Minimum size: 1"     "Modularity: 0,589"  </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co"># reduce overlaps by making the nodes less inclined to be close to each other</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">layoutNetwork</span>(<span class="st">'force-directed defaultSpringCoefficient=0.000006 defaultSpringLength=1'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Visualize network after clustering</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure network fits in the frame to be saved </span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a><span class="fu">fitContent</span>()</span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a><span class="co">#save network image</span></span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a><span class="fu">exportImage</span>(<span class="at">filename =</span> <span class="st">"currentnetwork_clustered.png"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                       file 
"/Users/emiljohansson/R/MedBioinfo/func-omics/currentnetwork_clustered.png" </code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="co"># view network image</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"currentnetwork_clustered.png"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="currentnetwork_clustered.png" class="img-fluid figure-img" width="398"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="save-network-info" class="level2" data-number="8.9">
<h2 data-number="8.9" class="anchored" data-anchor-id="save-network-info"><span class="header-section-number">8.9</span> Save network info</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the table with clustering results </span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>network_table <span class="ot">&lt;-</span> <span class="fu">getTableColumns</span>()</span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>clusterinfo <span class="ot">&lt;-</span> <span class="fu">table</span>(network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span>)</span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>clusterinfo</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
 1  2  3  4  5  6  7  8  9 10 11 12 13 14 
18 18 14  1  1  2  1  1  1  1  1  1  1  1 </code></pre>
</div>
</div>
</section>
<section id="members-of-each-big-cluster" class="level2" data-number="8.10">
<h2 data-number="8.10" class="anchored" data-anchor-id="members-of-each-big-cluster"><span class="header-section-number">8.10</span> Members of each big cluster</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the members of clusters with &gt;5 nodes for subsequent analysis </span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>bigclusters <span class="ot">&lt;-</span> <span class="fu">names</span>(clusterinfo)[(<span class="fu">table</span>(network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span>) <span class="sc">&gt;</span> <span class="dv">5</span>)]</span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>bigclusternames <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster"</span>, bigclusters)</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a><span class="co"># empty list of clusters to cycle through</span></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a>clusterlist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bigclusters)){</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create given clustername</span></span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>  given_clustername <span class="ot">&lt;-</span> bigclusternames[i]</span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb107-13"><a href="#cb107-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a list of these lists </span></span>
<span id="cb107-14"><a href="#cb107-14" aria-hidden="true" tabindex="-1"></a>  clusterlist[[i]] <span class="ot">&lt;-</span> network_table<span class="sc">$</span>query.term[network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span> <span class="sc">==</span> bigclusters[i]]</span>
<span id="cb107-15"><a href="#cb107-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb107-16"><a href="#cb107-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-17"><a href="#cb107-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(clusterlist) <span class="ot">&lt;-</span> bigclusternames</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Members of each cluster</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>clusterlist</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>$cluster1
 [1] "ICAM1"     "FCGR2A"    "S100A11"   "VCAM1"     "TLR3"      "TNFRSF11B"
 [7] "OLR1"      "ANXA2"     "FCGR1A"    "IL13RA1"   "IL1R1"     "CD44"     
[13] "MSR1"      "ANGPT1"    "MYD88"     "TNFRSF1A"  "CD93"      "EMP1"     

$cluster2
 [1] "FCER1G"   "CCR1"     "MS4A7"    "C1QB"     "MS4A4A"   "NCF2"    
 [7] "FCGR3A"   "CYBB"     "IFI30"    "CD14"     "HLA-DPA1" "MS4A6A"  
[13] "PTAFR"    "IFITM2"   "CD74"     "FGL2"     "RARRES3"  "LYZ"     

$cluster3
 [1] "SCIN"     "C7"       "TREM2"    "STEAP3"   "C4A"      "C4B"     
 [7] "SERPINA1" "SERPINA3" "APLNR"    "HMOX1"    "CCL2"     "C3"      
[13] "GFAP"     "CP"      </code></pre>
</div>
</div>
</section>
</section>
<section id="enrichment-analysis-for-each-cluster-go-biological-process" class="level1" data-number="9">
<h1 data-number="9"><span class="header-section-number">9</span> Enrichment analysis for each cluster (GO Biological Process)</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb110-2"><a href="#cb110-2" aria-hidden="true" tabindex="-1"></a>GO_BP_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb110-3"><a href="#cb110-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb110-4"><a href="#cb110-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb110-5"><a href="#cb110-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb110-6"><a href="#cb110-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb110-7"><a href="#cb110-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb110-8"><a href="#cb110-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb110-9"><a href="#cb110-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb110-10"><a href="#cb110-10" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Create dotplots for results</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_BP_enrichments)){</span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_BP_enrichments[[x]]</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA BP for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-40-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-40-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<section id="enrichment-analysis-for-each-cluster-go-molecular-function" class="level2" data-number="9.1">
<h2 data-number="9.1" class="anchored" data-anchor-id="enrichment-analysis-for-each-cluster-go-molecular-function"><span class="header-section-number">9.1</span> Enrichment analysis for each cluster (GO Molecular Function)</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>GO_MF_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb112-3"><a href="#cb112-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-4"><a href="#cb112-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb112-5"><a href="#cb112-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb112-6"><a href="#cb112-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb112-7"><a href="#cb112-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"MF"</span>,</span>
<span id="cb112-8"><a href="#cb112-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb112-9"><a href="#cb112-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb112-10"><a href="#cb112-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb112-11"><a href="#cb112-11" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Create dotplots for results</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_MF_enrichments)){</span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_MF_enrichments[[x]]</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA MF for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-42-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-42-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-42-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="enrichment-analysis-for-each-cluster-go-cellular-component" class="level2" data-number="9.2">
<h2 data-number="9.2" class="anchored" data-anchor-id="enrichment-analysis-for-each-cluster-go-cellular-component"><span class="header-section-number">9.2</span> Enrichment analysis for each cluster (GO Cellular Component)</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>GO_CC_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"CC"</span>,</span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb114-9"><a href="#cb114-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb114-10"><a href="#cb114-10" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Create dotplots for results</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_CC_enrichments)){</span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_CC_enrichments[[x]]</span>
<span id="cb115-5"><a href="#cb115-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-6"><a href="#cb115-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb115-7"><a href="#cb115-7" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb115-8"><a href="#cb115-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA CC for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb115-9"><a href="#cb115-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb115-10"><a href="#cb115-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb115-11"><a href="#cb115-11" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-44-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-44-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="random-forest-classification" class="level1" data-number="10">
<h1 data-number="10"><span class="header-section-number">10</span> Random forest classification</h1>
<section id="prepare-data-1" class="level2" data-number="10.1">
<h2 data-number="10.1" class="anchored" data-anchor-id="prepare-data-1"><span class="header-section-number">10.1</span> Prepare data</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and prepare the data</span></span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>data_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data_raw <span class="sc">%&gt;%</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneSymbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span></span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GeneSymbol)) <span class="sc">%&gt;%</span></span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"GeneSymbol"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="dv">0</span>, .)) <span class="sc">%&gt;%</span></span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(as.integer)</span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare metadata</span></span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>metadata_raw <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata_raw <span class="sc">%&gt;%</span></span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Index =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"Index"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sample =</span> ID, <span class="at">disease =</span> Disease) <span class="sc">%&gt;%</span></span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">gsub</span>(<span class="st">"_raw"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, individual, disease) <span class="sc">%&gt;%</span></span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">factor</span>(individual), <span class="at">disease =</span> <span class="fu">factor</span>(disease))</span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure the data is in a matrix form with gene symbols as rownames</span></span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data)</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DESeqDataSet</span></span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> countData,</span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata,</span>
<span id="cb116-28"><a href="#cb116-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> disease</span>
<span id="cb116-29"><a href="#cb116-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb116-30"><a href="#cb116-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-31"><a href="#cb116-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate size factors</span></span>
<span id="cb116-32"><a href="#cb116-32" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb116-33"><a href="#cb116-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-34"><a href="#cb116-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate gene-wise dispersions</span></span>
<span id="cb116-35"><a href="#cb116-35" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateDispersionsGeneEst</span>(dds)</span>
<span id="cb116-36"><a href="#cb116-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-37"><a href="#cb116-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Directly use the gene-wise estimates</span></span>
<span id="cb116-38"><a href="#cb116-38" aria-hidden="true" tabindex="-1"></a><span class="fu">dispersions</span>(dds) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(dds)<span class="sc">$</span>dispGeneEst</span>
<span id="cb116-39"><a href="#cb116-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-40"><a href="#cb116-40" aria-hidden="true" tabindex="-1"></a>normCounts <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Define model fitting procedure</span></span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a><span class="co">#NB: you will also need the packages statmod and e1071 for this analysis</span></span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a><span class="co">#New score function</span></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">mean</span>(x)</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a><span class="co">#New filter function</span></span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>filter <span class="ot">&lt;-</span> <span class="cf">function</span>(score, x, y){</span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(data) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb117-10"><a href="#cb117-10" aria-hidden="true" tabindex="-1"></a>selection <span class="ot">&lt;-</span> score <span class="sc">&gt;</span> <span class="fu">quantile</span>(score, <span class="fl">0.9</span>) <span class="sc">&amp;</span> meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb117-11"><a href="#cb117-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(score) <span class="sc">%in%</span> <span class="fu">names</span>(score)[selection]</span>
<span id="cb117-12"><a href="#cb117-12" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create training set</span></span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>trainingSet <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">assay</span>(normCounts)))</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Train and validate decision tree (C5.0)</span></span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>dtControl <span class="ot">&lt;-</span> <span class="fu">sbfControl</span>(</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">functions =</span> rfSBF,</span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"loocv"</span>,</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveDetails =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a>dtModel <span class="ot">&lt;-</span> <span class="fu">sbf</span>(</span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a>  trainingSet,</span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>  metadata<span class="sc">$</span>disease,</span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">sbfControl =</span> dtControl)</span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Show model</span></span>
<span id="cb118-16"><a href="#cb118-16" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(dtModel$fit)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="performance-evaluation" class="level2" data-number="10.2">
<h2 data-number="10.2" class="anchored" data-anchor-id="performance-evaluation"><span class="header-section-number">10.2</span> Performance evaluation</h2>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Print confusion matrix from validation</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(dtModel<span class="sc">$</span>pred<span class="sc">$</span>predictions<span class="sc">$</span>pred, metadata<span class="sc">$</span>disease)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>Confusion Matrix and Statistics

          Reference
Prediction AD Control
   AD       5       6
   Control  4       2
                                          
               Accuracy : 0.4118          
                 95% CI : (0.1844, 0.6708)
    No Information Rate : 0.5294          
    P-Value [Acc &gt; NIR] : 0.8878          
                                          
                  Kappa : -0.1972         
                                          
 Mcnemar's Test P-Value : 0.7518          
                                          
            Sensitivity : 0.5556          
            Specificity : 0.2500          
         Pos Pred Value : 0.4545          
         Neg Pred Value : 0.3333          
             Prevalence : 0.5294          
         Detection Rate : 0.2941          
   Detection Prevalence : 0.6471          
      Balanced Accuracy : 0.4028          
                                          
       'Positive' Class : AD              
                                          </code></pre>
</div>
</div>
</section>
</section>
<section id="spearman-correlation-analysis" class="level1" data-number="11">
<h1 data-number="11"><span class="header-section-number">11</span> Spearman correlation analysis</h1>
<p>Loading the data for the correlaton analysis</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/significant_genes.tsv"</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Performing Spearman correlation of the 63 upregulated genes on selected clinical variables of interest and comparing the correlation between Alzheimer’s disease and control samples.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to create correlation data and plot</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>create_correlation_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(data, metadata, gene_symbols, variable, title) {</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>  correlation_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(GeneSymbol <span class="sc">%in%</span> gene_symbols) <span class="sc">%&gt;%</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>GeneSymbol, <span class="at">names_to =</span> <span class="st">"ID"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">right_join</span>(metadata, <span class="at">by =</span> <span class="st">"ID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(GeneSymbol, Disease) <span class="sc">%&gt;%</span></span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">correlation =</span> <span class="fu">cor</span>(value, <span class="sc">!!</span>rlang<span class="sc">::</span><span class="fu">sym</span>(variable), <span class="at">method =</span> <span class="st">"spearman"</span>), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ungroup</span>()</span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract the order of genes based on their correlation in AD</span></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>  ad_correlations <span class="ot">&lt;-</span> correlation_data <span class="sc">%&gt;%</span></span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Disease <span class="sc">==</span> <span class="st">"AD"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(correlation)) <span class="sc">%&gt;%</span></span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(GeneSymbol)</span>
<span id="cb122-16"><a href="#cb122-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-17"><a href="#cb122-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Ensure that all GeneSymbols are included, even those not present in AD data</span></span>
<span id="cb122-18"><a href="#cb122-18" aria-hidden="true" tabindex="-1"></a>  full_gene_list <span class="ot">&lt;-</span> <span class="fu">unique</span>(correlation_data<span class="sc">$</span>GeneSymbol)</span>
<span id="cb122-19"><a href="#cb122-19" aria-hidden="true" tabindex="-1"></a>  ad_order <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(ad_correlations, full_gene_list)) <span class="co"># This ensures all genes are included and in order</span></span>
<span id="cb122-20"><a href="#cb122-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-21"><a href="#cb122-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set the levels of GeneSymbol according to their order in AD</span></span>
<span id="cb122-22"><a href="#cb122-22" aria-hidden="true" tabindex="-1"></a>  correlation_data<span class="sc">$</span>GeneSymbol <span class="ot">&lt;-</span> <span class="fu">factor</span>(correlation_data<span class="sc">$</span>GeneSymbol, <span class="at">levels =</span> ad_order)</span>
<span id="cb122-23"><a href="#cb122-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-24"><a href="#cb122-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot the heatmap with genes ordered by their correlation in AD</span></span>
<span id="cb122-25"><a href="#cb122-25" aria-hidden="true" tabindex="-1"></a>  heatmap_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(correlation_data, <span class="fu">aes</span>(<span class="at">x =</span> Disease, <span class="at">y =</span> GeneSymbol, <span class="at">fill =</span> correlation)) <span class="sc">+</span></span>
<span id="cb122-26"><a href="#cb122-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb122-27"><a href="#cb122-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"blue"</span>, <span class="at">high =</span> <span class="st">"red"</span>, <span class="at">mid =</span> <span class="st">"white"</span>, <span class="at">midpoint =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb122-28"><a href="#cb122-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb122-29"><a href="#cb122-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb122-30"><a href="#cb122-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Correlation"</span>, <span class="at">y =</span> <span class="st">"Gene (ordered by AD correlation)"</span>, <span class="at">x =</span> <span class="st">"Disease"</span>, <span class="at">title =</span> title)</span>
<span id="cb122-31"><a href="#cb122-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-32"><a href="#cb122-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(heatmap_plot)</span>
<span id="cb122-33"><a href="#cb122-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb122-34"><a href="#cb122-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-35"><a href="#cb122-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Variables of interest</span></span>
<span id="cb122-36"><a href="#cb122-36" aria-hidden="true" tabindex="-1"></a>variables_of_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Age"</span>, <span class="st">"PMI"</span>, <span class="st">"Mean_plaque"</span>, <span class="st">"CDR"</span>)</span>
<span id="cb122-37"><a href="#cb122-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-38"><a href="#cb122-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot list</span></span>
<span id="cb122-39"><a href="#cb122-39" aria-hidden="true" tabindex="-1"></a>plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb122-40"><a href="#cb122-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-41"><a href="#cb122-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop over variables and create plots</span></span>
<span id="cb122-42"><a href="#cb122-42" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (variable <span class="cf">in</span> variables_of_interest) {</span>
<span id="cb122-43"><a href="#cb122-43" aria-hidden="true" tabindex="-1"></a>  plot_list[[variable]] <span class="ot">&lt;-</span> <span class="fu">create_correlation_plot</span>(data, metadata, de_results<span class="sc">$</span>GeneSymbol, variable, <span class="fu">paste</span>(<span class="st">"Correlation with"</span>, variable))</span>
<span id="cb122-44"><a href="#cb122-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb122-45"><a href="#cb122-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-46"><a href="#cb122-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine plots</span></span>
<span id="cb122-47"><a href="#cb122-47" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plot_list, <span class="at">nrow =</span> <span class="dv">1</span>)</span>
<span id="cb122-48"><a href="#cb122-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-49"><a href="#cb122-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the combined plot</span></span>
<span id="cb122-50"><a href="#cb122-50" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="group2_final_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="1920"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="final-reflection-12-a4" class="level1" data-number="12">
<h1 data-number="12"><span class="header-section-number">12</span> Final reflection (1/2 A4)</h1>
<p>Statistical challenges: I think for our project there was a few challanges that we tried to overcome. The first one was when picking a completely new dataset unfamiliar to us. Even though some of the group members had worked on RNA-seq data before, it was a challenge for me to understand all the important steps during the pre-processing and considerations when analyzing. Since my background is in plasma proteomics, I found some similarities in the dataframe and how I could analyze the data. For instance, I was not used to the idea of count data because I normally work with either normalized protein expression (NPX) or medium florecence intensity (MFI).</p>
<p>My largest contributions to this work was the attempt to classify Alzheimer’s Disease (AD) using a leave-one-out (LOO) cross-validation (CV). In the short time limit that we had, I chose to apply a similar pipeline that was presented during the course. There was an issue with the vst() transformation funciton, which led me to perform some extra steps before I could apply the spelled out version of the function, which gave me a functional object from which I could proceed with this pipeline. In the end, the random forest classification gave an AUC of 0.41, which is worse than a random guess. There could have been several reasons behind this. On the one hand, since we may have picked the wrong data set or performed unlegitimate pre-processing, the data could have been more noisy.</p>
<p>The other contribution was to perform Spearman correlation analysis of the significant upregulated genes from the differential expression analysis on the AD patients and the controls. Some biological relevancies was found like the higher correlation to CDR. However, it didn’t make any sense to me that the top markers would have a negative correlation with age since age is a big factor behind AD, but this could be because of the noisy data.</p>
<p>Final reflection: In light of my own PhD project and what I have learnt from this course, I now feel more confortable applying these concept to my plasma proteomics pipelines and I now have more thoughts on how I could improve my current projects with methods such as canonical correlation analysis (CCA). My goal is to characterize the cardiovascular proteome and I think it will include some more multivariate analysis, therefore many of these concept I learnt during the course will be of great help.</p>
</section>
<section id="session-info" class="level1" data-number="13">
<h1 data-number="13"><span class="header-section-number">13</span> Session info</h1>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Stockholm
tzcode source: internal

attached base packages:
[1] stats4    grid      stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-3          RCy3_2.22.1                
 [3] egg_0.4.5                   gridExtra_2.3              
 [5] enrichplot_1.22.0           org.Hs.eg.db_3.18.0        
 [7] AnnotationDbi_1.64.1        clusterProfiler_4.10.1     
 [9] igraph_2.0.3                STRINGdb_2.14.3            
[11] edgeR_4.0.16                caret_6.0-94               
[13] lattice_0.22-6              randomForest_4.7-1.1       
[15] pheatmap_1.0.12             DESeq2_1.42.1              
[17] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[19] MatrixGenerics_1.14.0       matrixStats_1.4.1          
[21] GenomicRanges_1.54.1        GenomeInfoDb_1.38.8        
[23] IRanges_2.36.0              S4Vectors_0.40.2           
[25] BiocGenerics_0.48.1         ComplexUpset_1.3.3         
[27] UpSetR_1.4.0                ggvenn_0.1.10              
[29] cowplot_1.1.3               ggplotify_0.1.2            
[31] uwot_0.2.2                  Matrix_1.6-4               
[33] umap_0.2.10.0               plotly_4.10.4              
[35] HDAnalyzeR_1.0.0            xlsx_0.6.5                 
[37] tensorflow_2.16.0           kknn_1.3.1                 
[39] xgboost_1.7.8.1             keras_2.15.0               
[41] themis_1.0.2                vip_0.4.1                  
[43] ggsignif_0.6.4              corrr_0.4.4                
[45] visdat_0.6.0                GGally_2.2.1               
[47] skimr_2.1.5                 gt_0.11.0                  
[49] ggbeeswarm_0.7.2            yardstick_1.3.1            
[51] workflowsets_1.1.0          workflows_1.1.4            
[53] tune_1.2.1                  rsample_1.2.1              
[55] recipes_1.1.0               parsnip_1.2.1              
[57] modeldata_1.4.0             infer_1.0.7                
[59] dials_1.3.0                 scales_1.3.0               
[61] broom_1.0.6                 tidymodels_1.2.0           
[63] DT_0.33                     patchwork_1.3.0            
[65] kableExtra_1.4.0            magrittr_2.0.3             
[67] ggrepel_0.9.6               limma_3.58.1               
[69] readxl_1.4.3                lubridate_1.9.3            
[71] forcats_1.0.0               stringr_1.5.1              
[73] dplyr_1.1.4                 purrr_1.0.2                
[75] readr_2.1.5                 tidyr_1.3.1                
[77] tibble_3.2.1                ggplot2_3.5.1              
[79] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             nnet_7.3-19             Biostrings_2.70.3      
  [4] vctrs_0.6.5             proxy_0.4-27            digest_0.6.37          
  [7] png_0.1-8               tfruns_1.5.3            IRdisplay_1.1          
 [10] parallelly_1.38.0       MASS_7.3-60             reshape2_1.4.4         
 [13] foreach_1.5.2           qvalue_2.34.0           withr_3.0.1            
 [16] xfun_0.49               ggfun_0.1.6             survival_3.7-0         
 [19] memoise_2.0.1           gson_0.1.0              systemfonts_1.1.0      
 [22] tidytree_0.4.6          gtools_3.9.5            IRkernel_1.3.2         
 [25] KEGGREST_1.42.0         httr_1.4.7              globals_0.16.3         
 [28] hash_2.2.6.3            rstudioapi_0.16.0       archive_1.1.9          
 [31] generics_0.1.3          DOSE_3.28.2             base64enc_0.1-3        
 [34] curl_5.2.2              repr_1.1.7              zlibbioc_1.48.2        
 [37] ggraph_2.2.1            polyclip_1.10-7         GenomeInfoDbData_1.2.11
 [40] SparseArray_1.2.4       evaluate_1.0.0          S4Arrays_1.2.1         
 [43] hms_1.1.3               colorspace_2.1-1        reticulate_1.39.0      
 [46] viridis_0.6.5           ggtree_3.10.1           future.apply_1.11.2    
 [49] lhs_1.2.0               XML_3.99-0.17           shadowtext_0.1.4       
 [52] ROSE_0.0-4              class_7.3-22            pillar_1.9.0           
 [55] nlme_3.1-166            iterators_1.0.14        GPfit_1.0-8            
 [58] caTools_1.18.3          compiler_4.3.2          RSpectra_0.16-2        
 [61] stringi_1.8.4           gower_1.0.1             plyr_1.8.9             
 [64] crayon_1.5.3            abind_1.4-8             gridGraphics_0.5-1     
 [67] chron_2.3-61            locfit_1.5-9.10         graphlayouts_1.1.1     
 [70] bit_4.0.5               fastmatch_1.1-4         whisker_0.4.1          
 [73] codetools_0.2-20        openssl_2.2.1           e1071_1.7-16           
 [76] splines_4.3.2           Rcpp_1.0.13             DiceDesign_1.10        
 [79] HDO.db_0.99.1           cellranger_1.1.0        knitr_1.48             
 [82] blob_1.2.4              utf8_1.2.4              pbdZMQ_0.3-13          
 [85] RJSONIO_1.3-1.9         fs_1.6.4                listenv_0.9.1          
 [88] sqldf_0.4-11            statmod_1.5.0           tzdb_0.4.0             
 [91] svglite_2.1.3           tweenr_2.0.3            pkgconfig_2.0.3        
 [94] tools_4.3.2             cachem_1.1.0            RSQLite_2.3.7          
 [97] viridisLite_0.4.2       DBI_1.2.3               fastmap_1.2.0          
[100] rmarkdown_2.28          ggstats_0.6.0           graph_1.80.0           
[103] rpart_4.1.23            farver_2.1.2            tidygraph_1.3.1        
[106] scatterpie_0.2.4        gsubfn_0.7              yaml_2.3.10            
[109] cli_3.6.3               lifecycle_1.0.4         askpass_1.2.0          
[112] lava_1.8.0              backports_1.5.0         BiocParallel_1.36.0    
[115] timechange_0.3.0        gtable_0.3.5            ggridges_0.5.6         
[118] parallel_4.3.2          pROC_1.18.5             ape_5.8                
[121] jsonlite_1.8.8          bitops_1.0-8            bit64_4.0.5            
[124] yulab.utils_0.1.7       proto_1.0.0             GOSemSim_2.28.1        
[127] zeallot_0.1.0           timeDate_4032.109       lazyeval_0.2.2         
[130] htmltools_0.5.8.1       rJava_1.0-11            GO.db_3.18.0           
[133] glue_1.7.0              XVector_0.42.0          RCurl_1.98-1.16        
[136] treeio_1.26.0           R6_2.5.1                gplots_3.1.3.1         
[139] labeling_0.4.3          xlsxjars_0.6.1          aplot_0.2.3            
[142] ipred_0.9-15            DelayedArray_0.28.0     tidyselect_1.2.1       
[145] vipor_0.4.7             plotrix_3.8-4           ggforce_0.4.2          
[148] xml2_1.3.6              future_1.34.0           ModelMetrics_1.2.2.2   
[151] munsell_0.5.1           KernSmooth_2.23-24      furrr_0.3.1            
[154] data.table_1.16.0       htmlwidgets_1.6.4       fgsea_1.28.0           
[157] rlang_1.1.4             uuid_1.2-1              ggnewscale_0.5.0       
[160] fansi_1.0.6             base64url_1.4           hardhat_1.4.0          
[163] beeswarm_0.4.0          prodlim_2024.06.25     </code></pre>
</div>
</div>
<!-- -->

</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb125" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Group 2 report: Characterization of Alzheimer's disease"</span></span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "A project for the course Statistics in Bioinformatics"</span></span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Emil Johansson, Inika Prasad, Vaishnavi Divya Shridar &amp; Yuqi Zheng"</span></span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> last-modified</span></span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a><span class="an">date-format:</span><span class="co"> "dddd [the] Do [of] MMMM YYYY"</span></span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: lumen</span></span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a><span class="co">    title-block-banner: true</span></span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a><span class="co">    smooth-scroll: true</span></span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 4</span></span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: right</span></span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a><span class="co">    number-sections: true</span></span>
<span id="cb125-16"><a href="#cb125-16" aria-hidden="true" tabindex="-1"></a><span class="co">    number-depth: 4</span></span>
<span id="cb125-17"><a href="#cb125-17" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: true</span></span>
<span id="cb125-18"><a href="#cb125-18" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb125-19"><a href="#cb125-19" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb125-20"><a href="#cb125-20" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb125-21"><a href="#cb125-21" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: kable</span></span>
<span id="cb125-22"><a href="#cb125-22" aria-hidden="true" tabindex="-1"></a><span class="co">    standalone: true</span></span>
<span id="cb125-23"><a href="#cb125-23" aria-hidden="true" tabindex="-1"></a><span class="co">    fig-align: left</span></span>
<span id="cb125-24"><a href="#cb125-24" aria-hidden="true" tabindex="-1"></a><span class="co">    figure:</span></span>
<span id="cb125-25"><a href="#cb125-25" aria-hidden="true" tabindex="-1"></a><span class="co">      caption: "Figure {number}: {label}"</span></span>
<span id="cb125-26"><a href="#cb125-26" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb125-27"><a href="#cb125-27" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: inline</span></span>
<span id="cb125-28"><a href="#cb125-28" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb125-29"><a href="#cb125-29" aria-hidden="true" tabindex="-1"></a><span class="co">  echo: true</span></span>
<span id="cb125-30"><a href="#cb125-30" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb125-31"><a href="#cb125-31" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb125-32"><a href="#cb125-32" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> </span></span>
<span id="cb125-33"><a href="#cb125-33" aria-hidden="true" tabindex="-1"></a><span class="co">  markdown: </span></span>
<span id="cb125-34"><a href="#cb125-34" aria-hidden="true" tabindex="-1"></a><span class="co">    wrap: 72</span></span>
<span id="cb125-35"><a href="#cb125-35" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb125-36"><a href="#cb125-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-37"><a href="#cb125-37" aria-hidden="true" tabindex="-1"></a><span class="fu"># Setup</span></span>
<span id="cb125-38"><a href="#cb125-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-41"><a href="#cb125-41" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-42"><a href="#cb125-42" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb125-43"><a href="#cb125-43" aria-hidden="true" tabindex="-1"></a><span class="co">#    install.packages("BiocManager")</span></span>
<span id="cb125-44"><a href="#cb125-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-45"><a href="#cb125-45" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("RCy3")</span></span>
<span id="cb125-46"><a href="#cb125-46" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-47"><a href="#cb125-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-50"><a href="#cb125-50" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-51"><a href="#cb125-51" aria-hidden="true" tabindex="-1"></a><span class="co"># Libraries we need</span></span>
<span id="cb125-52"><a href="#cb125-52" aria-hidden="true" tabindex="-1"></a>libs <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb125-53"><a href="#cb125-53" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tidyverse"</span>, <span class="st">"readxl"</span>, <span class="st">"limma"</span>, <span class="st">"ggrepel"</span>, <span class="st">"magrittr"</span>, <span class="st">"kableExtra"</span>,</span>
<span id="cb125-54"><a href="#cb125-54" aria-hidden="true" tabindex="-1"></a>  <span class="st">"patchwork"</span>, <span class="st">"DT"</span>, <span class="st">"tidymodels"</span>, <span class="st">"ggbeeswarm"</span>, <span class="st">"gt"</span>, <span class="st">"skimr"</span>, <span class="st">"GGally"</span>,</span>
<span id="cb125-55"><a href="#cb125-55" aria-hidden="true" tabindex="-1"></a>  <span class="st">"visdat"</span>, <span class="st">"corrr"</span>, <span class="st">"ggsignif"</span>, <span class="st">"vip"</span>, <span class="st">"themis"</span>, <span class="st">"keras"</span>, <span class="st">"xgboost"</span>, <span class="st">"kknn"</span>,</span>
<span id="cb125-56"><a href="#cb125-56" aria-hidden="true" tabindex="-1"></a>  <span class="st">"tensorflow"</span>, <span class="st">"xlsx"</span>, <span class="st">"HDAnalyzeR"</span>, <span class="st">"plotly"</span>, <span class="st">"umap"</span>, <span class="st">"uwot"</span>, <span class="st">"ggplotify"</span>,</span>
<span id="cb125-57"><a href="#cb125-57" aria-hidden="true" tabindex="-1"></a>  <span class="st">"cowplot"</span>, <span class="st">"ggvenn"</span>, <span class="st">"UpSetR"</span>, <span class="st">"ComplexUpset"</span>, <span class="st">"DESeq2"</span>, <span class="st">"pheatmap"</span>, <span class="st">"randomForest"</span>,</span>
<span id="cb125-58"><a href="#cb125-58" aria-hidden="true" tabindex="-1"></a>  <span class="st">"caret"</span>, <span class="st">"edgeR"</span>, <span class="st">"STRINGdb"</span>, <span class="st">"igraph"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"org.Hs.eg.db"</span>, <span class="st">"enrichplot"</span>,</span>
<span id="cb125-59"><a href="#cb125-59" aria-hidden="true" tabindex="-1"></a>  <span class="st">"egg"</span>, <span class="st">"RCy3"</span>, <span class="st">"RColorBrewer"</span></span>
<span id="cb125-60"><a href="#cb125-60" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb125-61"><a href="#cb125-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-62"><a href="#cb125-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Install missing libraries</span></span>
<span id="cb125-63"><a href="#cb125-63" aria-hidden="true" tabindex="-1"></a>installed_libs <span class="ot">&lt;-</span> libs <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>())</span>
<span id="cb125-64"><a href="#cb125-64" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">any</span>(installed_libs <span class="sc">==</span> <span class="cn">FALSE</span>)) {</span>
<span id="cb125-65"><a href="#cb125-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(libs[<span class="sc">!</span>installed_libs])</span>
<span id="cb125-66"><a href="#cb125-66" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-67"><a href="#cb125-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-68"><a href="#cb125-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb125-69"><a href="#cb125-69" aria-hidden="true" tabindex="-1"></a><span class="fu">invisible</span>(<span class="fu">lapply</span>(libs, library, <span class="at">character.only =</span> <span class="cn">TRUE</span>))</span>
<span id="cb125-70"><a href="#cb125-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-71"><a href="#cb125-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-72"><a href="#cb125-72" aria-hidden="true" tabindex="-1"></a><span class="fu"># QC</span></span>
<span id="cb125-73"><a href="#cb125-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-74"><a href="#cb125-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{r load data}</span></span>
<span id="cb125-75"><a href="#cb125-75" aria-hidden="true" tabindex="-1"></a><span class="in"># Load gene expression count data</span></span>
<span id="cb125-76"><a href="#cb125-76" aria-hidden="true" tabindex="-1"></a><span class="in">count_data &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1) %&gt;%</span></span>
<span id="cb125-77"><a href="#cb125-77" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(matches("_raw$"))</span></span>
<span id="cb125-78"><a href="#cb125-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-79"><a href="#cb125-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-80"><a href="#cb125-80" aria-hidden="true" tabindex="-1"></a><span class="in">```{r preprocess data}</span></span>
<span id="cb125-81"><a href="#cb125-81" aria-hidden="true" tabindex="-1"></a><span class="in"># Round counts to integers</span></span>
<span id="cb125-82"><a href="#cb125-82" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data %&gt;% </span></span>
<span id="cb125-83"><a href="#cb125-83" aria-hidden="true" tabindex="-1"></a><span class="in">  dplyr::select(starts_with("C"), starts_with("A")) %&gt;%</span></span>
<span id="cb125-84"><a href="#cb125-84" aria-hidden="true" tabindex="-1"></a><span class="in">  mutate_all(~round(.))</span></span>
<span id="cb125-85"><a href="#cb125-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-86"><a href="#cb125-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-87"><a href="#cb125-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r filter low counts using mean log2 CPM}</span></span>
<span id="cb125-88"><a href="#cb125-88" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data</span></span>
<span id="cb125-89"><a href="#cb125-89" aria-hidden="true" tabindex="-1"></a><span class="in">genes_before_filtering &lt;- nrow(count_data_filtered)</span></span>
<span id="cb125-90"><a href="#cb125-90" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of genes before filtering:", genes_before_filtering, "\n")</span></span>
<span id="cb125-91"><a href="#cb125-91" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(counts = count_data_filtered)</span></span>
<span id="cb125-92"><a href="#cb125-92" aria-hidden="true" tabindex="-1"></a><span class="in"># dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb125-93"><a href="#cb125-93" aria-hidden="true" tabindex="-1"></a><span class="in">meanLog2CPM &lt;- rowMeans(log2(cpm(dge) + 1))</span></span>
<span id="cb125-94"><a href="#cb125-94" aria-hidden="true" tabindex="-1"></a><span class="in">hist(meanLog2CPM, main = "Histogram of Mean Log2 CPM", xlab = "Mean Log2 CPM")</span></span>
<span id="cb125-95"><a href="#cb125-95" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data_filtered[meanLog2CPM &gt; 1, ]</span></span>
<span id="cb125-96"><a href="#cb125-96" aria-hidden="true" tabindex="-1"></a><span class="in">genes_after_filtering &lt;- nrow(count_data_filtered)</span></span>
<span id="cb125-97"><a href="#cb125-97" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of genes after filtering:", genes_after_filtering, "\n")</span></span>
<span id="cb125-98"><a href="#cb125-98" aria-hidden="true" tabindex="-1"></a><span class="in">sample_conditions &lt;- data.frame(</span></span>
<span id="cb125-99"><a href="#cb125-99" aria-hidden="true" tabindex="-1"></a><span class="in">  sample = colnames(count_data_filtered),</span></span>
<span id="cb125-100"><a href="#cb125-100" aria-hidden="true" tabindex="-1"></a><span class="in">  condition = ifelse(grepl("^C", colnames(count_data_filtered)), "HC", "AD"),</span></span>
<span id="cb125-101"><a href="#cb125-101" aria-hidden="true" tabindex="-1"></a><span class="in">  batch = c(0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1)  # Manually adding batch information</span></span>
<span id="cb125-102"><a href="#cb125-102" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb125-103"><a href="#cb125-103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-104"><a href="#cb125-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-105"><a href="#cb125-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{r normalization with Voom}</span></span>
<span id="cb125-106"><a href="#cb125-106" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(counts = count_data_filtered)</span></span>
<span id="cb125-107"><a href="#cb125-107" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb125-108"><a href="#cb125-108" aria-hidden="true" tabindex="-1"></a><span class="in"># design &lt;- model.matrix(~ condition + batch, data = sample_conditions)</span></span>
<span id="cb125-109"><a href="#cb125-109" aria-hidden="true" tabindex="-1"></a><span class="in">design &lt;- model.matrix(~ condition + factor(batch), data = sample_conditions)</span></span>
<span id="cb125-110"><a href="#cb125-110" aria-hidden="true" tabindex="-1"></a><span class="in">v &lt;- voom(dge, design, plot = TRUE)</span></span>
<span id="cb125-111"><a href="#cb125-111" aria-hidden="true" tabindex="-1"></a><span class="in">head(v$E)</span></span>
<span id="cb125-112"><a href="#cb125-112" aria-hidden="true" tabindex="-1"></a><span class="in">hist(v$E, breaks = 50, main = "Distribution of Voom Normalized Counts", xlab = "Voom Normalized Counts")</span></span>
<span id="cb125-113"><a href="#cb125-113" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-114"><a href="#cb125-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-115"><a href="#cb125-115" aria-hidden="true" tabindex="-1"></a><span class="in">```{r normalization with DESeq2}</span></span>
<span id="cb125-116"><a href="#cb125-116" aria-hidden="true" tabindex="-1"></a><span class="in"># dds &lt;- DESeqDataSetFromMatrix(countData = count_data_filtered, </span></span>
<span id="cb125-117"><a href="#cb125-117" aria-hidden="true" tabindex="-1"></a><span class="in">#                               colData = sample_conditions, </span></span>
<span id="cb125-118"><a href="#cb125-118" aria-hidden="true" tabindex="-1"></a><span class="in">#                               design = ~ condition + factor(batch))</span></span>
<span id="cb125-119"><a href="#cb125-119" aria-hidden="true" tabindex="-1"></a><span class="in"># Normalize counts using variance stabilizing transformation (vst)</span></span>
<span id="cb125-120"><a href="#cb125-120" aria-hidden="true" tabindex="-1"></a><span class="in"># normCounts &lt;- vst(dds, blind = TRUE)</span></span>
<span id="cb125-121"><a href="#cb125-121" aria-hidden="true" tabindex="-1"></a><span class="in"># assay(normCounts)[1:5, 1:5]</span></span>
<span id="cb125-122"><a href="#cb125-122" aria-hidden="true" tabindex="-1"></a><span class="in"># hist(assay(normCounts), breaks = 50, main = "Distribution of VST Normalized Counts", xlab = "Normalized Counts")</span></span>
<span id="cb125-123"><a href="#cb125-123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-124"><a href="#cb125-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-125"><a href="#cb125-125" aria-hidden="true" tabindex="-1"></a><span class="in">```{r save normalized results}</span></span>
<span id="cb125-126"><a href="#cb125-126" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb125-127"><a href="#cb125-127" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb125-128"><a href="#cb125-128" aria-hidden="true" tabindex="-1"></a><span class="in">voom_normalized &lt;- as.data.frame(v$E)  # Convert v$E to a data frame</span></span>
<span id="cb125-129"><a href="#cb125-129" aria-hidden="true" tabindex="-1"></a><span class="in">voom_normalized$GeneID &lt;- rownames(voom_normalized)  # Add GeneID as a column</span></span>
<span id="cb125-130"><a href="#cb125-130" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_voom &lt;- merge(voom_normalized, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb125-131"><a href="#cb125-131" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_voom &lt;- gene_info_voom[, c("GeneID", "GeneSymbol", setdiff(names(gene_info_voom), c("GeneID", "GeneSymbol")))]</span></span>
<span id="cb125-132"><a href="#cb125-132" aria-hidden="true" tabindex="-1"></a><span class="in"># write.table(gene_info_voom, file = "voom_normalized.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span>
<span id="cb125-133"><a href="#cb125-133" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-134"><a href="#cb125-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-135"><a href="#cb125-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: condition heatmap}</span></span>
<span id="cb125-136"><a href="#cb125-136" aria-hidden="true" tabindex="-1"></a><span class="in"># sampleDist &lt;- cor(assay(normCounts), method = "spearman")</span></span>
<span id="cb125-137"><a href="#cb125-137" aria-hidden="true" tabindex="-1"></a><span class="in">sampleDist &lt;- cor(v$E, method = "spearman")</span></span>
<span id="cb125-138"><a href="#cb125-138" aria-hidden="true" tabindex="-1"></a><span class="in">sampleColor &lt;- brewer.pal(3, "Accent")[1:2]</span></span>
<span id="cb125-139"><a href="#cb125-139" aria-hidden="true" tabindex="-1"></a><span class="in">names(sampleColor) &lt;- unique(sample_conditions$condition)</span></span>
<span id="cb125-140"><a href="#cb125-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-141"><a href="#cb125-141" aria-hidden="true" tabindex="-1"></a><span class="in">pheatmap(sampleDist,</span></span>
<span id="cb125-142"><a href="#cb125-142" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_rows = as.dist(1 - sampleDist),</span></span>
<span id="cb125-143"><a href="#cb125-143" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_cols = as.dist(1 - sampleDist),</span></span>
<span id="cb125-144"><a href="#cb125-144" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_col = data.frame(Condition = sample_conditions$condition, row.names = sample_conditions$sample),</span></span>
<span id="cb125-145"><a href="#cb125-145" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_colors = list(Storage = sampleColor),</span></span>
<span id="cb125-146"><a href="#cb125-146" aria-hidden="true" tabindex="-1"></a><span class="in">  main = "Sample Correlation Heatmap by Condition")</span></span>
<span id="cb125-147"><a href="#cb125-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-148"><a href="#cb125-148" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: PCA by condition}</span></span>
<span id="cb125-149"><a href="#cb125-149" aria-hidden="true" tabindex="-1"></a><span class="in"># pcaRes &lt;- prcomp(t(assay(normCounts)))</span></span>
<span id="cb125-150"><a href="#cb125-150" aria-hidden="true" tabindex="-1"></a><span class="in">pcaRes &lt;- prcomp(t(v$E))</span></span>
<span id="cb125-151"><a href="#cb125-151" aria-hidden="true" tabindex="-1"></a><span class="in">varExp &lt;- round(pcaRes$sdev^2 / sum(pcaRes$sdev^2) * 100)</span></span>
<span id="cb125-152"><a href="#cb125-152" aria-hidden="true" tabindex="-1"></a><span class="in">pcaDF &lt;- data.frame(</span></span>
<span id="cb125-153"><a href="#cb125-153" aria-hidden="true" tabindex="-1"></a><span class="in">  PC1 = pcaRes$x[, 1],</span></span>
<span id="cb125-154"><a href="#cb125-154" aria-hidden="true" tabindex="-1"></a><span class="in">  PC2 = pcaRes$x[, 2],</span></span>
<span id="cb125-155"><a href="#cb125-155" aria-hidden="true" tabindex="-1"></a><span class="in">  Condition = sample_conditions$condition,</span></span>
<span id="cb125-156"><a href="#cb125-156" aria-hidden="true" tabindex="-1"></a><span class="in">  Sample = sample_conditions$sample</span></span>
<span id="cb125-157"><a href="#cb125-157" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb125-158"><a href="#cb125-158" aria-hidden="true" tabindex="-1"></a><span class="in">pcaPlot &lt;- ggplot(</span></span>
<span id="cb125-159"><a href="#cb125-159" aria-hidden="true" tabindex="-1"></a><span class="in">  data = pcaDF,</span></span>
<span id="cb125-160"><a href="#cb125-160" aria-hidden="true" tabindex="-1"></a><span class="in">  mapping = aes(x = PC1, y = PC2, color = Condition, label = Sample)</span></span>
<span id="cb125-161"><a href="#cb125-161" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb125-162"><a href="#cb125-162" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(size = 3) +</span></span>
<span id="cb125-163"><a href="#cb125-163" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 4) +</span></span>
<span id="cb125-164"><a href="#cb125-164" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x = paste0("PC1 (", varExp[1], " %)") ,</span></span>
<span id="cb125-165"><a href="#cb125-165" aria-hidden="true" tabindex="-1"></a><span class="in">       y = paste0("PC2 (", varExp[2], " %)") ,</span></span>
<span id="cb125-166"><a href="#cb125-166" aria-hidden="true" tabindex="-1"></a><span class="in">       title = "PCA Plot of Samples") +</span></span>
<span id="cb125-167"><a href="#cb125-167" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb125-168"><a href="#cb125-168" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(axis.text = element_text(size = 12), legend.text = element_text(size = 10)) +</span></span>
<span id="cb125-169"><a href="#cb125-169" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_color_manual(values = brewer.pal(3, "Accent")) +</span></span>
<span id="cb125-170"><a href="#cb125-170" aria-hidden="true" tabindex="-1"></a><span class="in">  coord_fixed(ratio = 1)  # Added fixed ratio to ensure equal scaling between PC1 and PC2</span></span>
<span id="cb125-171"><a href="#cb125-171" aria-hidden="true" tabindex="-1"></a><span class="in">print(pcaPlot)</span></span>
<span id="cb125-172"><a href="#cb125-172" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-173"><a href="#cb125-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-174"><a href="#cb125-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: batch heatmap}</span></span>
<span id="cb125-175"><a href="#cb125-175" aria-hidden="true" tabindex="-1"></a><span class="in"># Create a heatmap showing Batch 1 and Batch 2 samples separately</span></span>
<span id="cb125-176"><a href="#cb125-176" aria-hidden="true" tabindex="-1"></a><span class="in">sampleDist_batch &lt;- cor(v$E, method = "spearman")</span></span>
<span id="cb125-177"><a href="#cb125-177" aria-hidden="true" tabindex="-1"></a><span class="in">batchColor &lt;- brewer.pal(3, "Set2")[1:2]</span></span>
<span id="cb125-178"><a href="#cb125-178" aria-hidden="true" tabindex="-1"></a><span class="in">names(batchColor) &lt;- unique(sample_conditions$batch)</span></span>
<span id="cb125-179"><a href="#cb125-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-180"><a href="#cb125-180" aria-hidden="true" tabindex="-1"></a><span class="in">pheatmap(sampleDist_batch,</span></span>
<span id="cb125-181"><a href="#cb125-181" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_rows = as.dist(1 - sampleDist_batch),</span></span>
<span id="cb125-182"><a href="#cb125-182" aria-hidden="true" tabindex="-1"></a><span class="in">  clustering_distance_cols = as.dist(1 - sampleDist_batch),</span></span>
<span id="cb125-183"><a href="#cb125-183" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_col = data.frame(Batch = factor(sample_conditions$batch), row.names = sample_conditions$sample),</span></span>
<span id="cb125-184"><a href="#cb125-184" aria-hidden="true" tabindex="-1"></a><span class="in">  annotation_colors = list(Batch = batchColor),</span></span>
<span id="cb125-185"><a href="#cb125-185" aria-hidden="true" tabindex="-1"></a><span class="in">  main = "Sample Correlation Heatmap by Batch")</span></span>
<span id="cb125-186"><a href="#cb125-186" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-187"><a href="#cb125-187" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: PCA by Batch}</span></span>
<span id="cb125-188"><a href="#cb125-188" aria-hidden="true" tabindex="-1"></a><span class="in"># PCA Plot by Batch</span></span>
<span id="cb125-189"><a href="#cb125-189" aria-hidden="true" tabindex="-1"></a><span class="in">pcaRes_batch &lt;- prcomp(t(v$E))</span></span>
<span id="cb125-190"><a href="#cb125-190" aria-hidden="true" tabindex="-1"></a><span class="in">varExp_batch &lt;- round(pcaRes_batch$sdev^2 / sum(pcaRes_batch$sdev^2) * 100)</span></span>
<span id="cb125-191"><a href="#cb125-191" aria-hidden="true" tabindex="-1"></a><span class="in">pcaDF_batch &lt;- data.frame(</span></span>
<span id="cb125-192"><a href="#cb125-192" aria-hidden="true" tabindex="-1"></a><span class="in">  PC1 = pcaRes_batch$x[, 1],</span></span>
<span id="cb125-193"><a href="#cb125-193" aria-hidden="true" tabindex="-1"></a><span class="in">  PC2 = pcaRes_batch$x[, 2],</span></span>
<span id="cb125-194"><a href="#cb125-194" aria-hidden="true" tabindex="-1"></a><span class="in">  Batch = factor(sample_conditions$batch),</span></span>
<span id="cb125-195"><a href="#cb125-195" aria-hidden="true" tabindex="-1"></a><span class="in">  Sample = sample_conditions$sample</span></span>
<span id="cb125-196"><a href="#cb125-196" aria-hidden="true" tabindex="-1"></a><span class="in">)</span></span>
<span id="cb125-197"><a href="#cb125-197" aria-hidden="true" tabindex="-1"></a><span class="in">pcaPlot_batch &lt;- ggplot(</span></span>
<span id="cb125-198"><a href="#cb125-198" aria-hidden="true" tabindex="-1"></a><span class="in">  data = pcaDF_batch,</span></span>
<span id="cb125-199"><a href="#cb125-199" aria-hidden="true" tabindex="-1"></a><span class="in">  mapping = aes(x = PC1, y = PC2, color = Batch, label = Sample)</span></span>
<span id="cb125-200"><a href="#cb125-200" aria-hidden="true" tabindex="-1"></a><span class="in">) +</span></span>
<span id="cb125-201"><a href="#cb125-201" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(size = 3) +</span></span>
<span id="cb125-202"><a href="#cb125-202" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 4) +</span></span>
<span id="cb125-203"><a href="#cb125-203" aria-hidden="true" tabindex="-1"></a><span class="in">  labs(x = paste0("PC1 (", varExp_batch[1], " %)"),</span></span>
<span id="cb125-204"><a href="#cb125-204" aria-hidden="true" tabindex="-1"></a><span class="in">       y = paste0("PC2 (", varExp_batch[2], " %)"),</span></span>
<span id="cb125-205"><a href="#cb125-205" aria-hidden="true" tabindex="-1"></a><span class="in">       title = "PCA Plot of Samples by Batch") +</span></span>
<span id="cb125-206"><a href="#cb125-206" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb125-207"><a href="#cb125-207" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(axis.text = element_text(size = 12), legend.text = element_text(size = 10)) +</span></span>
<span id="cb125-208"><a href="#cb125-208" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_color_manual(values = brewer.pal(3, "Set2")) +</span></span>
<span id="cb125-209"><a href="#cb125-209" aria-hidden="true" tabindex="-1"></a><span class="in">  coord_fixed(ratio = 1)</span></span>
<span id="cb125-210"><a href="#cb125-210" aria-hidden="true" tabindex="-1"></a><span class="in">print(pcaPlot_batch)</span></span>
<span id="cb125-211"><a href="#cb125-211" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-212"><a href="#cb125-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-213"><a href="#cb125-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{r QC: outlier removal}</span></span>
<span id="cb125-214"><a href="#cb125-214" aria-hidden="true" tabindex="-1"></a><span class="in"># Remove identified outliers</span></span>
<span id="cb125-215"><a href="#cb125-215" aria-hidden="true" tabindex="-1"></a><span class="in"># outliers &lt;- c("C1_raw", "C2_raw", "C4_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb125-216"><a href="#cb125-216" aria-hidden="true" tabindex="-1"></a><span class="in">outliers &lt;- c("C1_raw", "C5_raw", "C6_raw")</span></span>
<span id="cb125-217"><a href="#cb125-217" aria-hidden="true" tabindex="-1"></a><span class="in">remaining_samples &lt;- setdiff(colnames(count_data_filtered), outliers)</span></span>
<span id="cb125-218"><a href="#cb125-218" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_filtered &lt;- count_data_filtered[, remaining_samples]</span></span>
<span id="cb125-219"><a href="#cb125-219" aria-hidden="true" tabindex="-1"></a><span class="in">sample_conditions &lt;- sample_conditions[!sample_conditions$sample %in% outliers, ]</span></span>
<span id="cb125-220"><a href="#cb125-220" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-221"><a href="#cb125-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-222"><a href="#cb125-222" aria-hidden="true" tabindex="-1"></a><span class="fu"># Differential expression analysis</span></span>
<span id="cb125-223"><a href="#cb125-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-224"><a href="#cb125-224" aria-hidden="true" tabindex="-1"></a><span class="fu">## Design matrix</span></span>
<span id="cb125-225"><a href="#cb125-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-226"><a href="#cb125-226" aria-hidden="true" tabindex="-1"></a><span class="in">```{r statistical analyses}</span></span>
<span id="cb125-227"><a href="#cb125-227" aria-hidden="true" tabindex="-1"></a><span class="in"># Define design matrix including batch effect</span></span>
<span id="cb125-228"><a href="#cb125-228" aria-hidden="true" tabindex="-1"></a><span class="in"># designMatrix &lt;- model.matrix(~ 0 + condition + batch, data = sample_conditions)</span></span>
<span id="cb125-229"><a href="#cb125-229" aria-hidden="true" tabindex="-1"></a><span class="in">designMatrix &lt;- model.matrix(~ 0 + condition + factor(batch), data = sample_conditions)</span></span>
<span id="cb125-230"><a href="#cb125-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-231"><a href="#cb125-231" aria-hidden="true" tabindex="-1"></a><span class="in">colnames(designMatrix) &lt;- make.names(colnames(designMatrix))  # Ensure valid column names for contrast matrix</span></span>
<span id="cb125-232"><a href="#cb125-232" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Design Matrix with Batch Effect:\n")</span></span>
<span id="cb125-233"><a href="#cb125-233" aria-hidden="true" tabindex="-1"></a><span class="in">print(head(designMatrix))</span></span>
<span id="cb125-234"><a href="#cb125-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-235"><a href="#cb125-235" aria-hidden="true" tabindex="-1"></a><span class="in"># Define contrast matrix</span></span>
<span id="cb125-236"><a href="#cb125-236" aria-hidden="true" tabindex="-1"></a><span class="in">contrastMatrix &lt;- makeContrasts(AD_vs_HC = conditionAD - conditionHC, levels = designMatrix)</span></span>
<span id="cb125-237"><a href="#cb125-237" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Contrast Matrix:\n")</span></span>
<span id="cb125-238"><a href="#cb125-238" aria-hidden="true" tabindex="-1"></a><span class="in">print(contrastMatrix)</span></span>
<span id="cb125-239"><a href="#cb125-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-240"><a href="#cb125-240" aria-hidden="true" tabindex="-1"></a><span class="in"># Prepare DGEList and estimate dispersion</span></span>
<span id="cb125-241"><a href="#cb125-241" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- DGEList(count_data_filtered)</span></span>
<span id="cb125-242"><a href="#cb125-242" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- calcNormFactors(dge)</span></span>
<span id="cb125-243"><a href="#cb125-243" aria-hidden="true" tabindex="-1"></a><span class="in">dge &lt;- estimateDisp(dge, designMatrix, robust = TRUE)</span></span>
<span id="cb125-244"><a href="#cb125-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-245"><a href="#cb125-245" aria-hidden="true" tabindex="-1"></a><span class="in"># Fit the model using likelihood ratio test (LRT), now including batch in the design</span></span>
<span id="cb125-246"><a href="#cb125-246" aria-hidden="true" tabindex="-1"></a><span class="in">fit &lt;- glmFit(dge, designMatrix)</span></span>
<span id="cb125-247"><a href="#cb125-247" aria-hidden="true" tabindex="-1"></a><span class="in">lrt &lt;- glmLRT(fit, contrast = contrastMatrix)</span></span>
<span id="cb125-248"><a href="#cb125-248" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-249"><a href="#cb125-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-250"><a href="#cb125-250" aria-hidden="true" tabindex="-1"></a><span class="fu">## Hypothesis correction</span></span>
<span id="cb125-251"><a href="#cb125-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-252"><a href="#cb125-252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r hypothesis testing}</span></span>
<span id="cb125-253"><a href="#cb125-253" aria-hidden="true" tabindex="-1"></a><span class="in">res &lt;- topTags(lrt, n = nrow(count_data_filtered))</span></span>
<span id="cb125-254"><a href="#cb125-254" aria-hidden="true" tabindex="-1"></a><span class="in">res$table$FDR &lt;- p.adjust(res$table$PValue, method = "BH")</span></span>
<span id="cb125-255"><a href="#cb125-255" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Summary of p-values:\n")</span></span>
<span id="cb125-256"><a href="#cb125-256" aria-hidden="true" tabindex="-1"></a><span class="in">print(summary(res$table$PValue))</span></span>
<span id="cb125-257"><a href="#cb125-257" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Summary of FDRs:\n")</span></span>
<span id="cb125-258"><a href="#cb125-258" aria-hidden="true" tabindex="-1"></a><span class="in">print(summary(res$table$FDR))</span></span>
<span id="cb125-259"><a href="#cb125-259" aria-hidden="true" tabindex="-1"></a><span class="in">sigRes &lt;- subset(res$table, FDR &lt; 0.05 &amp; abs(logFC) &gt; 1)</span></span>
<span id="cb125-260"><a href="#cb125-260" aria-hidden="true" tabindex="-1"></a><span class="in">cat("Number of significant genes:", nrow(sigRes), "\n")</span></span>
<span id="cb125-261"><a href="#cb125-261" aria-hidden="true" tabindex="-1"></a><span class="in">knitr::kable(head(sigRes))</span></span>
<span id="cb125-262"><a href="#cb125-262" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-263"><a href="#cb125-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-264"><a href="#cb125-264" aria-hidden="true" tabindex="-1"></a><span class="in">```{r add gene symbols}</span></span>
<span id="cb125-265"><a href="#cb125-265" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb125-266"><a href="#cb125-266" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb125-267"><a href="#cb125-267" aria-hidden="true" tabindex="-1"></a><span class="in">sigRes$GeneID &lt;- rownames(sigRes)  # Add GeneID as a column to the results</span></span>
<span id="cb125-268"><a href="#cb125-268" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- merge(sigRes, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb125-269"><a href="#cb125-269" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info &lt;- gene_info[, c("GeneID", "GeneSymbol", setdiff(names(gene_info), c("GeneID", "GeneSymbol")))]</span></span>
<span id="cb125-270"><a href="#cb125-270" aria-hidden="true" tabindex="-1"></a><span class="in"># write.table(gene_info, file = "sig_genes.tsv", sep = "\t", quote = FALSE, row.names = FALSE)</span></span>
<span id="cb125-271"><a href="#cb125-271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-272"><a href="#cb125-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-273"><a href="#cb125-273" aria-hidden="true" tabindex="-1"></a><span class="fu">## Volcano plot</span></span>
<span id="cb125-274"><a href="#cb125-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-275"><a href="#cb125-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{r visualize results}</span></span>
<span id="cb125-276"><a href="#cb125-276" aria-hidden="true" tabindex="-1"></a><span class="in"># Add gene symbols to the results</span></span>
<span id="cb125-277"><a href="#cb125-277" aria-hidden="true" tabindex="-1"></a><span class="in">count_data_with_symbols &lt;- read.table("data/GSE53697_RNAseq_AD.txt", header = TRUE, sep = "\t", row.names = 1)</span></span>
<span id="cb125-278"><a href="#cb125-278" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- count_data_with_symbols[, "GeneSymbol", drop = FALSE]</span></span>
<span id="cb125-279"><a href="#cb125-279" aria-hidden="true" tabindex="-1"></a><span class="in">res$table$GeneID &lt;- rownames(res$table)</span></span>
<span id="cb125-280"><a href="#cb125-280" aria-hidden="true" tabindex="-1"></a><span class="in">gene_info_all &lt;- merge(res$table, gene_symbols, by.x = "GeneID", by.y = "row.names", all.x = TRUE)</span></span>
<span id="cb125-281"><a href="#cb125-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-282"><a href="#cb125-282" aria-hidden="true" tabindex="-1"></a><span class="in"># Volcano plot using gene symbols for all genes</span></span>
<span id="cb125-283"><a href="#cb125-283" aria-hidden="true" tabindex="-1"></a><span class="in">volcanoPlot &lt;- ggplot(gene_info_all, aes(x = logFC, y = -log10(FDR),</span></span>
<span id="cb125-284"><a href="#cb125-284" aria-hidden="true" tabindex="-1"></a><span class="in">                                         color = ifelse(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, "darkred", "grey"),</span></span>
<span id="cb125-285"><a href="#cb125-285" aria-hidden="true" tabindex="-1"></a><span class="in">                                         label = ifelse(FDR &lt; 0.05 &amp; abs(logFC) &gt; 1, GeneSymbol, ""))) +</span></span>
<span id="cb125-286"><a href="#cb125-286" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_point(alpha = 0.5, size = 2) +</span></span>
<span id="cb125-287"><a href="#cb125-287" aria-hidden="true" tabindex="-1"></a><span class="in">  xlab(expression("Fold Change, Log"[2]*"")) +</span></span>
<span id="cb125-288"><a href="#cb125-288" aria-hidden="true" tabindex="-1"></a><span class="in">  ylab(expression("Adjusted P value, Log"[10]*"")) +</span></span>
<span id="cb125-289"><a href="#cb125-289" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_vline(xintercept = c(-1, 1), linetype = "dotted", linewidth = 1) +</span></span>
<span id="cb125-290"><a href="#cb125-290" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_hline(yintercept = -log10(0.05), linetype = "dotted", linewidth = 1) +</span></span>
<span id="cb125-291"><a href="#cb125-291" aria-hidden="true" tabindex="-1"></a><span class="in">  theme_minimal() +</span></span>
<span id="cb125-292"><a href="#cb125-292" aria-hidden="true" tabindex="-1"></a><span class="in">  theme(legend.position = "none") +</span></span>
<span id="cb125-293"><a href="#cb125-293" aria-hidden="true" tabindex="-1"></a><span class="in">  scale_colour_manual(values = c("darkred", "grey")) +</span></span>
<span id="cb125-294"><a href="#cb125-294" aria-hidden="true" tabindex="-1"></a><span class="in">  geom_text_repel(size = 3, max.overlaps = 10)</span></span>
<span id="cb125-295"><a href="#cb125-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-296"><a href="#cb125-296" aria-hidden="true" tabindex="-1"></a><span class="in">print(volcanoPlot)</span></span>
<span id="cb125-297"><a href="#cb125-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-298"><a href="#cb125-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-299"><a href="#cb125-299" aria-hidden="true" tabindex="-1"></a><span class="fu"># Protein-protein interaction analysis</span></span>
<span id="cb125-300"><a href="#cb125-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-301"><a href="#cb125-301" aria-hidden="true" tabindex="-1"></a><span class="in">```{r protein-protein interation analysis}</span></span>
<span id="cb125-302"><a href="#cb125-302" aria-hidden="true" tabindex="-1"></a><span class="in"># Initialize STRINGdb</span></span>
<span id="cb125-303"><a href="#cb125-303" aria-hidden="true" tabindex="-1"></a><span class="in">string_db &lt;- STRINGdb$new(version = "11", species = 9606, score_threshold = 400, input_directory = "")</span></span>
<span id="cb125-304"><a href="#cb125-304" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- gene_info$GeneSymbol</span></span>
<span id="cb125-305"><a href="#cb125-305" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols &lt;- data.frame(GeneSymbol = gene_symbols, stringsAsFactors = FALSE)</span></span>
<span id="cb125-306"><a href="#cb125-306" aria-hidden="true" tabindex="-1"></a><span class="in">mapped_genes &lt;- string_db$map(gene_symbols, "GeneSymbol", removeUnmappedRows = TRUE)</span></span>
<span id="cb125-307"><a href="#cb125-307" aria-hidden="true" tabindex="-1"></a><span class="in"># Get PPI Network for the mapped genes</span></span>
<span id="cb125-308"><a href="#cb125-308" aria-hidden="true" tabindex="-1"></a><span class="in">ppi_network &lt;- string_db$get_interactions(mapped_genes$STRING_id)</span></span>
<span id="cb125-309"><a href="#cb125-309" aria-hidden="true" tabindex="-1"></a><span class="in">string_db$plot_network(mapped_genes$STRING_id)</span></span>
<span id="cb125-310"><a href="#cb125-310" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-311"><a href="#cb125-311" aria-hidden="true" tabindex="-1"></a><span class="in">```{r enrichGO SEA}</span></span>
<span id="cb125-312"><a href="#cb125-312" aria-hidden="true" tabindex="-1"></a><span class="in">gene_symbols_vector &lt;- gene_symbols$GeneSymbol  # Extract the gene symbols as a vector</span></span>
<span id="cb125-313"><a href="#cb125-313" aria-hidden="true" tabindex="-1"></a><span class="in">mapped_genes_entrez &lt;- bitr(gene_symbols_vector,</span></span>
<span id="cb125-314"><a href="#cb125-314" aria-hidden="true" tabindex="-1"></a><span class="in">                            fromType = "SYMBOL",</span></span>
<span id="cb125-315"><a href="#cb125-315" aria-hidden="true" tabindex="-1"></a><span class="in">                            toType = "ENTREZID",</span></span>
<span id="cb125-316"><a href="#cb125-316" aria-hidden="true" tabindex="-1"></a><span class="in">                            OrgDb = org.Hs.eg.db)</span></span>
<span id="cb125-317"><a href="#cb125-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-318"><a href="#cb125-318" aria-hidden="true" tabindex="-1"></a><span class="in">goSEA &lt;- enrichGO(gene = mapped_genes_entrez$ENTREZID,</span></span>
<span id="cb125-319"><a href="#cb125-319" aria-hidden="true" tabindex="-1"></a><span class="in">                  OrgDb = org.Hs.eg.db,</span></span>
<span id="cb125-320"><a href="#cb125-320" aria-hidden="true" tabindex="-1"></a><span class="in">                  keyType = "ENTREZID",</span></span>
<span id="cb125-321"><a href="#cb125-321" aria-hidden="true" tabindex="-1"></a><span class="in">                  ont = "BP",</span></span>
<span id="cb125-322"><a href="#cb125-322" aria-hidden="true" tabindex="-1"></a><span class="in">                  pAdjustMethod = "BH",</span></span>
<span id="cb125-323"><a href="#cb125-323" aria-hidden="true" tabindex="-1"></a><span class="in">                  pvalueCutoff = 0.05,</span></span>
<span id="cb125-324"><a href="#cb125-324" aria-hidden="true" tabindex="-1"></a><span class="in">                  qvalueCutoff = 0.05)</span></span>
<span id="cb125-325"><a href="#cb125-325" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-326"><a href="#cb125-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-327"><a href="#cb125-327" aria-hidden="true" tabindex="-1"></a><span class="fu"># Enrichment analysis</span></span>
<span id="cb125-328"><a href="#cb125-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-329"><a href="#cb125-329" aria-hidden="true" tabindex="-1"></a><span class="in">```{r Visualize GO SEA enrichment results}</span></span>
<span id="cb125-330"><a href="#cb125-330" aria-hidden="true" tabindex="-1"></a><span class="in">cnetplot(goSEA, colorEdge = TRUE, cex_label_gene = 0.5)</span></span>
<span id="cb125-331"><a href="#cb125-331" aria-hidden="true" tabindex="-1"></a><span class="in">dotplot(goSEA)</span></span>
<span id="cb125-332"><a href="#cb125-332" aria-hidden="true" tabindex="-1"></a><span class="in">goSEA &lt;- pairwise_termsim(goSEA)</span></span>
<span id="cb125-333"><a href="#cb125-333" aria-hidden="true" tabindex="-1"></a><span class="in">treeplot(goSEA)</span></span>
<span id="cb125-334"><a href="#cb125-334" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-335"><a href="#cb125-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-338"><a href="#cb125-338" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-339"><a href="#cb125-339" aria-hidden="true" tabindex="-1"></a><span class="co"># import data</span></span>
<span id="cb125-340"><a href="#cb125-340" aria-hidden="true" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/significant_genes.tsv'</span>) </span>
<span id="cb125-341"><a href="#cb125-341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-342"><a href="#cb125-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-343"><a href="#cb125-343" aria-hidden="true" tabindex="-1"></a><span class="fu"># GSEA Analysis</span></span>
<span id="cb125-344"><a href="#cb125-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-345"><a href="#cb125-345" aria-hidden="true" tabindex="-1"></a><span class="in">```{r gsea_on_all}</span></span>
<span id="cb125-346"><a href="#cb125-346" aria-hidden="true" tabindex="-1"></a><span class="in">all_genes &lt;- as.data.frame(read_csv('data/all_gene_processed.csv'))</span></span>
<span id="cb125-347"><a href="#cb125-347" aria-hidden="true" tabindex="-1"></a><span class="in">rownames(all_genes) &lt;- all_genes$`...1`</span></span>
<span id="cb125-348"><a href="#cb125-348" aria-hidden="true" tabindex="-1"></a><span class="in">all_genes &lt;- all_genes[,2:6]</span></span>
<span id="cb125-349"><a href="#cb125-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-350"><a href="#cb125-350" aria-hidden="true" tabindex="-1"></a><span class="in">processed_genes_for_gse &lt;- as.vector(all_genes$logFC)  # p value vector</span></span>
<span id="cb125-351"><a href="#cb125-351" aria-hidden="true" tabindex="-1"></a><span class="in">names(processed_genes_for_gse) &lt;- as.vector(rownames(all_genes)) # gene ids</span></span>
<span id="cb125-352"><a href="#cb125-352" aria-hidden="true" tabindex="-1"></a><span class="in">processed_genes_for_gse &lt;- processed_genes_for_gse[order(-processed_genes_for_gse)] # order ranked</span></span>
<span id="cb125-353"><a href="#cb125-353" aria-hidden="true" tabindex="-1"></a><span class="in">#processed_genes_for_gse</span></span>
<span id="cb125-354"><a href="#cb125-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-355"><a href="#cb125-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-356"><a href="#cb125-356" aria-hidden="true" tabindex="-1"></a><span class="in">ego &lt;- gseGO(geneList     = processed_genes_for_gse,</span></span>
<span id="cb125-357"><a href="#cb125-357" aria-hidden="true" tabindex="-1"></a><span class="in">              OrgDb        = org.Hs.eg.db,</span></span>
<span id="cb125-358"><a href="#cb125-358" aria-hidden="true" tabindex="-1"></a><span class="in">              ont          = "BP", ## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-359"><a href="#cb125-359" aria-hidden="true" tabindex="-1"></a><span class="in">              nPerm        = 1000,</span></span>
<span id="cb125-360"><a href="#cb125-360" aria-hidden="true" tabindex="-1"></a><span class="in">              minGSSize    = 3,</span></span>
<span id="cb125-361"><a href="#cb125-361" aria-hidden="true" tabindex="-1"></a><span class="in">              maxGSSize    = 500,</span></span>
<span id="cb125-362"><a href="#cb125-362" aria-hidden="true" tabindex="-1"></a><span class="in">              pvalueCutoff = 0.05,</span></span>
<span id="cb125-363"><a href="#cb125-363" aria-hidden="true" tabindex="-1"></a><span class="in">              verbose      = TRUE,</span></span>
<span id="cb125-364"><a href="#cb125-364" aria-hidden="true" tabindex="-1"></a><span class="in">              by = "fgsea",</span></span>
<span id="cb125-365"><a href="#cb125-365" aria-hidden="true" tabindex="-1"></a><span class="in">              pAdjustMethod = "none")</span></span>
<span id="cb125-366"><a href="#cb125-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-367"><a href="#cb125-367" aria-hidden="true" tabindex="-1"></a><span class="in">ego</span></span>
<span id="cb125-368"><a href="#cb125-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-369"><a href="#cb125-369" aria-hidden="true" tabindex="-1"></a><span class="in">goplot(ego)</span></span>
<span id="cb125-370"><a href="#cb125-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-371"><a href="#cb125-371" aria-hidden="true" tabindex="-1"></a><span class="in">dotplot(ego, showCategory=10, split=".sign") + facet_grid(.~.sign) + labs(title = "GSEA For Biological Processes in Alzheimer's Disease")</span></span>
<span id="cb125-372"><a href="#cb125-372" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-373"><a href="#cb125-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-376"><a href="#cb125-376" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-377"><a href="#cb125-377" aria-hidden="true" tabindex="-1"></a>count_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"data/GSE53697_RNAseq_AD.txt"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-378"><a href="#cb125-378" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">matches</span>(<span class="st">"_raw$"</span>))</span>
<span id="cb125-379"><a href="#cb125-379" aria-hidden="true" tabindex="-1"></a>subset_count_data <span class="ot">&lt;-</span> count_data[<span class="fu">rownames</span>(count_data) <span class="sc">%in%</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>GeneID),]</span>
<span id="cb125-380"><a href="#cb125-380" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">'data/ad_metadata.xlsx'</span>)</span>
<span id="cb125-381"><a href="#cb125-381" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-382"><a href="#cb125-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-385"><a href="#cb125-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-386"><a href="#cb125-386" aria-hidden="true" tabindex="-1"></a>sampleDist <span class="ot">&lt;-</span> <span class="fu">cor</span>(subset_count_data, <span class="at">method =</span> <span class="st">"spearman"</span>) </span>
<span id="cb125-387"><a href="#cb125-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-388"><a href="#cb125-388" aria-hidden="true" tabindex="-1"></a>sample_conditions <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb125-389"><a href="#cb125-389" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(subset_count_data),</span>
<span id="cb125-390"><a href="#cb125-390" aria-hidden="true" tabindex="-1"></a>  <span class="at">condition =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">"^C"</span>, <span class="fu">colnames</span>(subset_count_data)), <span class="st">"HC"</span>, <span class="st">"AD"</span>),</span>
<span id="cb125-391"><a href="#cb125-391" aria-hidden="true" tabindex="-1"></a>  <span class="at">batch =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>),  <span class="do">## Manually adding batch information; batch effect is a bit serious as otherwise no gene will be identified as significant</span></span>
<span id="cb125-392"><a href="#cb125-392" aria-hidden="true" tabindex="-1"></a>  <span class="at">plaque =</span> metadata<span class="sc">$</span>BM9_plaque</span>
<span id="cb125-393"><a href="#cb125-393" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb125-394"><a href="#cb125-394" aria-hidden="true" tabindex="-1"></a>sample_conditions</span>
<span id="cb125-395"><a href="#cb125-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-396"><a href="#cb125-396" aria-hidden="true" tabindex="-1"></a>heat_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">condition =</span> <span class="fu">c</span>(<span class="st">"HC"</span> <span class="ot">=</span> <span class="st">"#01bfc5"</span>, <span class="st">"AD"</span> <span class="ot">=</span> <span class="st">"salmon"</span>))</span>
<span id="cb125-397"><a href="#cb125-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-398"><a href="#cb125-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-399"><a href="#cb125-399" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="at">mat =</span> sampleDist,</span>
<span id="cb125-400"><a href="#cb125-400" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>,</span>
<span id="cb125-401"><a href="#cb125-401" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb125-402"><a href="#cb125-402" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> <span class="fu">data.frame</span>(<span class="at">Plaque =</span> sample_conditions<span class="sc">$</span>plaque, <span class="at">row.names =</span> sample_conditions<span class="sc">$</span>sample))</span>
<span id="cb125-403"><a href="#cb125-403" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-404"><a href="#cb125-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-405"><a href="#cb125-405" aria-hidden="true" tabindex="-1"></a><span class="fu"># Incorrect GSEA Analysis</span></span>
<span id="cb125-406"><a href="#cb125-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-407"><a href="#cb125-407" aria-hidden="true" tabindex="-1"></a>The below analysis does not follow the expected steps! TO AVOID.</span>
<span id="cb125-408"><a href="#cb125-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-409"><a href="#cb125-409" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prepare data</span></span>
<span id="cb125-410"><a href="#cb125-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-413"><a href="#cb125-413" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-414"><a href="#cb125-414" aria-hidden="true" tabindex="-1"></a><span class="co"># create input vector for gene enrichment</span></span>
<span id="cb125-415"><a href="#cb125-415" aria-hidden="true" tabindex="-1"></a>significant_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb125-416"><a href="#cb125-416" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(significant_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(significant_genes<span class="sc">$</span>GeneID) <span class="co"># gene ids</span></span>
<span id="cb125-417"><a href="#cb125-417" aria-hidden="true" tabindex="-1"></a>significant_genes_for_gse <span class="ot">&lt;-</span> significant_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>significant_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb125-418"><a href="#cb125-418" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-419"><a href="#cb125-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-420"><a href="#cb125-420" aria-hidden="true" tabindex="-1"></a><span class="fu">## Run gseGO</span></span>
<span id="cb125-421"><a href="#cb125-421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-424"><a href="#cb125-424" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-425"><a href="#cb125-425" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology ALL</span></span>
<span id="cb125-426"><a href="#cb125-426" aria-hidden="true" tabindex="-1"></a>ego_all <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList   =</span> significant_genes_for_gse,</span>
<span id="cb125-427"><a href="#cb125-427" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-428"><a href="#cb125-428" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"ALL"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-429"><a href="#cb125-429" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-430"><a href="#cb125-430" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb125-431"><a href="#cb125-431" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-432"><a href="#cb125-432" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-433"><a href="#cb125-433" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-434"><a href="#cb125-434" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-435"><a href="#cb125-435" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span>
<span id="cb125-436"><a href="#cb125-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-437"><a href="#cb125-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-438"><a href="#cb125-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-441"><a href="#cb125-441" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-442"><a href="#cb125-442" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology MF</span></span>
<span id="cb125-443"><a href="#cb125-443" aria-hidden="true" tabindex="-1"></a>ego1 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb125-444"><a href="#cb125-444" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-445"><a href="#cb125-445" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"MF"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-446"><a href="#cb125-446" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-447"><a href="#cb125-447" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb125-448"><a href="#cb125-448" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-449"><a href="#cb125-449" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-450"><a href="#cb125-450" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-451"><a href="#cb125-451" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-452"><a href="#cb125-452" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span>
<span id="cb125-453"><a href="#cb125-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-454"><a href="#cb125-454" aria-hidden="true" tabindex="-1"></a><span class="do">## gave 0 enriched genes</span></span>
<span id="cb125-455"><a href="#cb125-455" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-456"><a href="#cb125-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-457"><a href="#cb125-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-460"><a href="#cb125-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-461"><a href="#cb125-461" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology CC</span></span>
<span id="cb125-462"><a href="#cb125-462" aria-hidden="true" tabindex="-1"></a>ego2 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb125-463"><a href="#cb125-463" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-464"><a href="#cb125-464" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"CC"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-465"><a href="#cb125-465" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-466"><a href="#cb125-466" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb125-467"><a href="#cb125-467" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-468"><a href="#cb125-468" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-469"><a href="#cb125-469" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-470"><a href="#cb125-470" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-471"><a href="#cb125-471" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) </span>
<span id="cb125-472"><a href="#cb125-472" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-473"><a href="#cb125-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-476"><a href="#cb125-476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-477"><a href="#cb125-477" aria-hidden="true" tabindex="-1"></a><span class="co"># run gseGO for ontology BP</span></span>
<span id="cb125-478"><a href="#cb125-478" aria-hidden="true" tabindex="-1"></a>ego3 <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> significant_genes_for_gse,</span>
<span id="cb125-479"><a href="#cb125-479" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-480"><a href="#cb125-480" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-481"><a href="#cb125-481" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-482"><a href="#cb125-482" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">10</span>,</span>
<span id="cb125-483"><a href="#cb125-483" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-484"><a href="#cb125-484" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-485"><a href="#cb125-485" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-486"><a href="#cb125-486" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-487"><a href="#cb125-487" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>) <span class="co"># “holm”, “hochberg”, “hommel”, “bonferroni”, “BH”, “BY”, “fdr”, “none”</span></span>
<span id="cb125-488"><a href="#cb125-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-489"><a href="#cb125-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-492"><a href="#cb125-492" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-493"><a href="#cb125-493" aria-hidden="true" tabindex="-1"></a><span class="co"># compare the processes and components identified as signifcant between the ontologies</span></span>
<span id="cb125-494"><a href="#cb125-494" aria-hidden="true" tabindex="-1"></a><span class="co">#ego3$Description</span></span>
<span id="cb125-495"><a href="#cb125-495" aria-hidden="true" tabindex="-1"></a><span class="co">#ego2$Description</span></span>
<span id="cb125-496"><a href="#cb125-496" aria-hidden="true" tabindex="-1"></a><span class="co">#ego_all$Description</span></span>
<span id="cb125-497"><a href="#cb125-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-498"><a href="#cb125-498" aria-hidden="true" tabindex="-1"></a><span class="co">#setdiff(ego2$Description, ego_all$Description)</span></span>
<span id="cb125-499"><a href="#cb125-499" aria-hidden="true" tabindex="-1"></a><span class="co">#setdiff(ego3$Description, ego_all$Description) # "cellular response to stress" disappears from list</span></span>
<span id="cb125-500"><a href="#cb125-500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-501"><a href="#cb125-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-502"><a href="#cb125-502" aria-hidden="true" tabindex="-1"></a><span class="fu">## Plots</span></span>
<span id="cb125-503"><a href="#cb125-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-506"><a href="#cb125-506" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-507"><a href="#cb125-507" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego3)</span>
<span id="cb125-508"><a href="#cb125-508" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-509"><a href="#cb125-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-512"><a href="#cb125-512" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-513"><a href="#cb125-513" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego2, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"GSEA For Cellular Components in Alzheimer's Disease"</span>)</span>
<span id="cb125-514"><a href="#cb125-514" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-515"><a href="#cb125-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-518"><a href="#cb125-518" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-519"><a href="#cb125-519" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego3, <span class="at">showCategory=</span><span class="dv">10</span>, <span class="at">split=</span><span class="st">".sign"</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(.<span class="sc">~</span>.sign) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"GSEA For Biological Processes in Alzheimer's Disease"</span>)</span>
<span id="cb125-520"><a href="#cb125-520" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-521"><a href="#cb125-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-524"><a href="#cb125-524" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-525"><a href="#cb125-525" aria-hidden="true" tabindex="-1"></a><span class="fu">emapplot</span>(<span class="fu">pairwise_termsim</span>(ego3), <span class="at">showCategory =</span> <span class="dv">10</span>)</span>
<span id="cb125-526"><a href="#cb125-526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-527"><a href="#cb125-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-528"><a href="#cb125-528" aria-hidden="true" tabindex="-1"></a><span class="in">```{r , fig.width=10, fig.height=20}</span></span>
<span id="cb125-529"><a href="#cb125-529" aria-hidden="true" tabindex="-1"></a><span class="in">## compare the different plots</span></span>
<span id="cb125-530"><a href="#cb125-530" aria-hidden="true" tabindex="-1"></a><span class="in">d_all &lt;- dotplot(ego_all, showCategory=10, split=".sign") + facet_grid(.~.sign) +  theme(axis.text.y = element_text(hjust = 1, size = 7)) </span></span>
<span id="cb125-531"><a href="#cb125-531" aria-hidden="true" tabindex="-1"></a><span class="in">d2 &lt;- dotplot(ego2, showCategory=10, split=".sign") + facet_grid(.~.sign)+  theme(axis.text.y = element_text(hjust = 1, size = 7)) </span></span>
<span id="cb125-532"><a href="#cb125-532" aria-hidden="true" tabindex="-1"></a><span class="in">d3 &lt;- dotplot(ego3, showCategory=10, split=".sign") + facet_grid(.~.sign)+  theme(axis.text.y = element_text(hjust = 1, size = 7)) </span></span>
<span id="cb125-533"><a href="#cb125-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-534"><a href="#cb125-534" aria-hidden="true" tabindex="-1"></a><span class="in">ggarrange(d_all, d2, d3, </span></span>
<span id="cb125-535"><a href="#cb125-535" aria-hidden="true" tabindex="-1"></a><span class="in">          labels = c("ALL", "CC", "BP"),</span></span>
<span id="cb125-536"><a href="#cb125-536" aria-hidden="true" tabindex="-1"></a><span class="in">          ncol = 1, nrow = 3)</span></span>
<span id="cb125-537"><a href="#cb125-537" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-538"><a href="#cb125-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-541"><a href="#cb125-541" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-542"><a href="#cb125-542" aria-hidden="true" tabindex="-1"></a><span class="fu">ridgeplot</span>(ego3) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"enrichment distribution"</span>) <span class="sc">+</span>   <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">7</span>)) </span>
<span id="cb125-543"><a href="#cb125-543" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-544"><a href="#cb125-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-547"><a href="#cb125-547" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-548"><a href="#cb125-548" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot</span>(ego3, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> ego3<span class="sc">$</span>Description[<span class="dv">1</span>], <span class="at">geneSetID =</span> <span class="dv">1</span>)</span>
<span id="cb125-549"><a href="#cb125-549" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-550"><a href="#cb125-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-553"><a href="#cb125-553" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-554"><a href="#cb125-554" aria-hidden="true" tabindex="-1"></a><span class="fu">gseaplot</span>(ego3, <span class="at">by =</span> <span class="st">"all"</span>, <span class="at">title =</span> ego3<span class="sc">$</span>Description[<span class="dv">2</span>], <span class="at">geneSetID =</span> <span class="dv">2</span>)</span>
<span id="cb125-555"><a href="#cb125-555" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-556"><a href="#cb125-556" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-557"><a href="#cb125-557" aria-hidden="true" tabindex="-1"></a><span class="fu">## GSEA on specific clusters</span></span>
<span id="cb125-558"><a href="#cb125-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-561"><a href="#cb125-561" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-562"><a href="#cb125-562" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb125-563"><a href="#cb125-563" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb125-564"><a href="#cb125-564" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb125-565"><a href="#cb125-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-566"><a href="#cb125-566" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb125-567"><a href="#cb125-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-568"><a href="#cb125-568" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster1.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-569"><a href="#cb125-569" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb125-570"><a href="#cb125-570" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb125-571"><a href="#cb125-571" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb125-572"><a href="#cb125-572" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb125-573"><a href="#cb125-573" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb125-574"><a href="#cb125-574" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb125-575"><a href="#cb125-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-576"><a href="#cb125-576" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb125-577"><a href="#cb125-577" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-578"><a href="#cb125-578" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-579"><a href="#cb125-579" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-580"><a href="#cb125-580" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb125-581"><a href="#cb125-581" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-582"><a href="#cb125-582" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-583"><a href="#cb125-583" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-584"><a href="#cb125-584" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-585"><a href="#cb125-585" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb125-586"><a href="#cb125-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-587"><a href="#cb125-587" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span>
<span id="cb125-588"><a href="#cb125-588" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-589"><a href="#cb125-589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-592"><a href="#cb125-592" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-593"><a href="#cb125-593" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb125-594"><a href="#cb125-594" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb125-595"><a href="#cb125-595" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb125-596"><a href="#cb125-596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-597"><a href="#cb125-597" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb125-598"><a href="#cb125-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-599"><a href="#cb125-599" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster2.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-600"><a href="#cb125-600" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb125-601"><a href="#cb125-601" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb125-602"><a href="#cb125-602" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb125-603"><a href="#cb125-603" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb125-604"><a href="#cb125-604" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb125-605"><a href="#cb125-605" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb125-606"><a href="#cb125-606" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse</span>
<span id="cb125-607"><a href="#cb125-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-608"><a href="#cb125-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-609"><a href="#cb125-609" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb125-610"><a href="#cb125-610" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-611"><a href="#cb125-611" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-612"><a href="#cb125-612" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-613"><a href="#cb125-613" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb125-614"><a href="#cb125-614" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-615"><a href="#cb125-615" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-616"><a href="#cb125-616" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-617"><a href="#cb125-617" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-618"><a href="#cb125-618" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb125-619"><a href="#cb125-619" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-620"><a href="#cb125-620" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span>
<span id="cb125-621"><a href="#cb125-621" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-622"><a href="#cb125-622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-623"><a href="#cb125-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-626"><a href="#cb125-626" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-627"><a href="#cb125-627" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">read_csv</span>(<span class="st">'data/all_gene_processed.csv'</span>))</span>
<span id="cb125-628"><a href="#cb125-628" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(all_genes) <span class="ot">&lt;-</span> all_genes<span class="sc">$</span><span class="st">`</span><span class="at">...1</span><span class="st">`</span></span>
<span id="cb125-629"><a href="#cb125-629" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> all_genes[,<span class="dv">2</span><span class="sc">:</span><span class="dv">6</span>]</span>
<span id="cb125-630"><a href="#cb125-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-631"><a href="#cb125-631" aria-hidden="true" tabindex="-1"></a>voom_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">'data/voom_normalized.tsv'</span>) </span>
<span id="cb125-632"><a href="#cb125-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-633"><a href="#cb125-633" aria-hidden="true" tabindex="-1"></a>gene_cluster <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(<span class="st">'data/cluster5.txt'</span>, <span class="at">delim=</span><span class="st">'</span><span class="sc">\n</span><span class="st">'</span>, <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-634"><a href="#cb125-634" aria-hidden="true" tabindex="-1"></a>gene_cluster_data <span class="ot">&lt;-</span> voom_normalized[voom_normalized<span class="sc">$</span>GeneSymbol <span class="sc">%in%</span> <span class="fu">as.vector</span>(gene_cluster<span class="sc">$</span>X1),]</span>
<span id="cb125-635"><a href="#cb125-635" aria-hidden="true" tabindex="-1"></a>genes_for_gse <span class="ot">&lt;-</span> all_genes[<span class="fu">rownames</span>(all_genes) <span class="sc">%in%</span> gene_cluster_data<span class="sc">$</span>GeneID,]</span>
<span id="cb125-636"><a href="#cb125-636" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb125-637"><a href="#cb125-637" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(genes_for_gse<span class="sc">$</span>logFC)  <span class="co"># p value vector</span></span>
<span id="cb125-638"><a href="#cb125-638" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(processed_genes_for_gse) <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">rownames</span>(genes_for_gse)) <span class="co"># gene ids</span></span>
<span id="cb125-639"><a href="#cb125-639" aria-hidden="true" tabindex="-1"></a>processed_genes_for_gse <span class="ot">&lt;-</span> processed_genes_for_gse[<span class="fu">order</span>(<span class="sc">-</span>processed_genes_for_gse)] <span class="co"># order ranked</span></span>
<span id="cb125-640"><a href="#cb125-640" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-641"><a href="#cb125-641" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">gseGO</span>(<span class="at">geneList     =</span> processed_genes_for_gse,</span>
<span id="cb125-642"><a href="#cb125-642" aria-hidden="true" tabindex="-1"></a>              <span class="at">OrgDb        =</span> org.Hs.eg.db,</span>
<span id="cb125-643"><a href="#cb125-643" aria-hidden="true" tabindex="-1"></a>              <span class="at">ont          =</span> <span class="st">"BP"</span>, <span class="do">## Molecular Function (MF), Biological Process (BP), and Cellular Component (CC)</span></span>
<span id="cb125-644"><a href="#cb125-644" aria-hidden="true" tabindex="-1"></a>              <span class="at">nPerm        =</span> <span class="dv">1000</span>,</span>
<span id="cb125-645"><a href="#cb125-645" aria-hidden="true" tabindex="-1"></a>              <span class="at">minGSSize    =</span> <span class="dv">3</span>,</span>
<span id="cb125-646"><a href="#cb125-646" aria-hidden="true" tabindex="-1"></a>              <span class="at">maxGSSize    =</span> <span class="dv">500</span>,</span>
<span id="cb125-647"><a href="#cb125-647" aria-hidden="true" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb125-648"><a href="#cb125-648" aria-hidden="true" tabindex="-1"></a>              <span class="at">verbose      =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-649"><a href="#cb125-649" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="st">"fgsea"</span>,</span>
<span id="cb125-650"><a href="#cb125-650" aria-hidden="true" tabindex="-1"></a>              <span class="at">pAdjustMethod =</span> <span class="st">"none"</span>)</span>
<span id="cb125-651"><a href="#cb125-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-652"><a href="#cb125-652" aria-hidden="true" tabindex="-1"></a><span class="fu">goplot</span>(ego)</span>
<span id="cb125-653"><a href="#cb125-653" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-654"><a href="#cb125-654" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-655"><a href="#cb125-655" aria-hidden="true" tabindex="-1"></a><span class="fu"># CytoScape </span></span>
<span id="cb125-656"><a href="#cb125-656" aria-hidden="true" tabindex="-1"></a>We will investigate protein-protein interactions (PPIs) for a given list of proteins. To do so, we will retreive PPI data from the STRING database and visualize networks in Cytoscape using the R package RCy3. We will further cluster the network to investigate whether subsets of proteins share a particular molecular function. </span>
<span id="cb125-657"><a href="#cb125-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-658"><a href="#cb125-658" aria-hidden="true" tabindex="-1"></a>Requirements: Cytoscape running, with the clusterMaker app installed.</span>
<span id="cb125-659"><a href="#cb125-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-662"><a href="#cb125-662" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-663"><a href="#cb125-663" aria-hidden="true" tabindex="-1"></a><span class="co"># Confirm that Cytoscape is running</span></span>
<span id="cb125-664"><a href="#cb125-664" aria-hidden="true" tabindex="-1"></a><span class="fu">cytoscapeVersionInfo</span> ()</span>
<span id="cb125-665"><a href="#cb125-665" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-666"><a href="#cb125-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-667"><a href="#cb125-667" aria-hidden="true" tabindex="-1"></a><span class="fu">## Load protein list </span></span>
<span id="cb125-668"><a href="#cb125-668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-671"><a href="#cb125-671" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-672"><a href="#cb125-672" aria-hidden="true" tabindex="-1"></a>proteins <span class="ot">&lt;-</span> <span class="fu">readLines</span>(<span class="st">"data/gene_symbols_list.txt"</span>)</span>
<span id="cb125-673"><a href="#cb125-673" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(proteins)</span>
<span id="cb125-674"><a href="#cb125-674" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-675"><a href="#cb125-675" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-676"><a href="#cb125-676" aria-hidden="true" tabindex="-1"></a><span class="fu">## Initialize STRING db</span></span>
<span id="cb125-677"><a href="#cb125-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-680"><a href="#cb125-680" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-681"><a href="#cb125-681" aria-hidden="true" tabindex="-1"></a><span class="co"># Initialize STRINGdb object </span></span>
<span id="cb125-682"><a href="#cb125-682" aria-hidden="true" tabindex="-1"></a>string_db <span class="ot">&lt;-</span> STRINGdb<span class="sc">$</span><span class="fu">new</span>(<span class="at">version=</span><span class="st">"12.0"</span>, </span>
<span id="cb125-683"><a href="#cb125-683" aria-hidden="true" tabindex="-1"></a>                          <span class="at">species=</span><span class="dv">9606</span>, <span class="co"># Homo sapiens</span></span>
<span id="cb125-684"><a href="#cb125-684" aria-hidden="true" tabindex="-1"></a>                          <span class="at">score_threshold=</span><span class="dv">400</span>, </span>
<span id="cb125-685"><a href="#cb125-685" aria-hidden="true" tabindex="-1"></a>                          <span class="at">input_directory=</span><span class="st">""</span>)</span>
<span id="cb125-686"><a href="#cb125-686" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-687"><a href="#cb125-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-688"><a href="#cb125-688" aria-hidden="true" tabindex="-1"></a><span class="fu">## Map gene names to STRING aliases </span></span>
<span id="cb125-689"><a href="#cb125-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-692"><a href="#cb125-692" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-693"><a href="#cb125-693" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a dataframe with uppercase letters (to match with STRING alias mapping)</span></span>
<span id="cb125-694"><a href="#cb125-694" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">query.term =</span> <span class="fu">toupper</span>(proteins)) </span>
<span id="cb125-695"><a href="#cb125-695" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-696"><a href="#cb125-696" aria-hidden="true" tabindex="-1"></a><span class="co"># check if file with mapping to STRING aliases exists</span></span>
<span id="cb125-697"><a href="#cb125-697" aria-hidden="true" tabindex="-1"></a><span class="co"># and create it if it doesn't</span></span>
<span id="cb125-698"><a href="#cb125-698" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="st">"data/mapped2STRING.csv"</span>)) {</span>
<span id="cb125-699"><a href="#cb125-699" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-700"><a href="#cb125-700" aria-hidden="true" tabindex="-1"></a>  mapped <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/mapped2STRING.csv"</span>)</span>
<span id="cb125-701"><a href="#cb125-701" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-702"><a href="#cb125-702" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb125-703"><a href="#cb125-703" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-704"><a href="#cb125-704" aria-hidden="true" tabindex="-1"></a>mapped <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">map</span>(<span class="at">my_data_frame =</span> data, </span>
<span id="cb125-705"><a href="#cb125-705" aria-hidden="true" tabindex="-1"></a>                        <span class="at">my_data_frame_id_col_names =</span> <span class="st">"query.term"</span>, </span>
<span id="cb125-706"><a href="#cb125-706" aria-hidden="true" tabindex="-1"></a>                        <span class="at">takeFirst=</span><span class="cn">TRUE</span>, </span>
<span id="cb125-707"><a href="#cb125-707" aria-hidden="true" tabindex="-1"></a>                        <span class="at">removeUnmappedRows=</span><span class="cn">TRUE</span>, </span>
<span id="cb125-708"><a href="#cb125-708" aria-hidden="true" tabindex="-1"></a>                        <span class="at">quiet=</span><span class="cn">FALSE</span>)</span>
<span id="cb125-709"><a href="#cb125-709" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="at">x =</span> mapped, <span class="at">file =</span> <span class="st">"data/mapped2STRING.csv"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-710"><a href="#cb125-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-711"><a href="#cb125-711" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-712"><a href="#cb125-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-713"><a href="#cb125-713" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mapped)</span>
<span id="cb125-714"><a href="#cb125-714" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-715"><a href="#cb125-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-718"><a href="#cb125-718" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-719"><a href="#cb125-719" aria-hidden="true" tabindex="-1"></a><span class="co"># Check how many proteins were not mapped</span></span>
<span id="cb125-720"><a href="#cb125-720" aria-hidden="true" tabindex="-1"></a><span class="co"># Since some gene IDs can map to several STRING identifiers, we account for duplicates </span></span>
<span id="cb125-721"><a href="#cb125-721" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(data<span class="sc">$</span>query.term)) <span class="sc">-</span> <span class="fu">length</span>(<span class="fu">unique</span>(mapped<span class="sc">$</span>query.term))</span>
<span id="cb125-722"><a href="#cb125-722" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-723"><a href="#cb125-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-724"><a href="#cb125-724" aria-hidden="true" tabindex="-1"></a><span class="fu">## Fetch interaction data</span></span>
<span id="cb125-725"><a href="#cb125-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-728"><a href="#cb125-728" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-729"><a href="#cb125-729" aria-hidden="true" tabindex="-1"></a><span class="co"># Get interactions for the mapped proteins if the file isn't already there</span></span>
<span id="cb125-730"><a href="#cb125-730" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-731"><a href="#cb125-731" aria-hidden="true" tabindex="-1"></a><span class="co"># check if file with interaction data exists </span></span>
<span id="cb125-732"><a href="#cb125-732" aria-hidden="true" tabindex="-1"></a><span class="co"># and create it if it doesn't</span></span>
<span id="cb125-733"><a href="#cb125-733" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">file.exists</span>(<span class="st">"data/interactions.csv"</span>)) {</span>
<span id="cb125-734"><a href="#cb125-734" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-735"><a href="#cb125-735" aria-hidden="true" tabindex="-1"></a>  interactions <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/interactions.csv"</span>)</span>
<span id="cb125-736"><a href="#cb125-736" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-737"><a href="#cb125-737" aria-hidden="true" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb125-738"><a href="#cb125-738" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-739"><a href="#cb125-739" aria-hidden="true" tabindex="-1"></a>interactions <span class="ot">&lt;-</span> string_db<span class="sc">$</span><span class="fu">get_interactions</span>(mapped<span class="sc">$</span>STRING_id)</span>
<span id="cb125-740"><a href="#cb125-740" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="at">x =</span> interactions, <span class="at">file =</span> <span class="st">"data/interactions.csv"</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-741"><a href="#cb125-741" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-742"><a href="#cb125-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-743"><a href="#cb125-743" aria-hidden="true" tabindex="-1"></a><span class="co"># View the first few interactions </span></span>
<span id="cb125-744"><a href="#cb125-744" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(interactions)</span>
<span id="cb125-745"><a href="#cb125-745" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-746"><a href="#cb125-746" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-747"><a href="#cb125-747" aria-hidden="true" tabindex="-1"></a><span class="fu">## Create dataframe for PPI network</span></span>
<span id="cb125-748"><a href="#cb125-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-749"><a href="#cb125-749" aria-hidden="true" tabindex="-1"></a>Define nodes for the network</span>
<span id="cb125-750"><a href="#cb125-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-753"><a href="#cb125-753" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-754"><a href="#cb125-754" aria-hidden="true" tabindex="-1"></a><span class="co"># Get node columns for the Cytoscape network</span></span>
<span id="cb125-755"><a href="#cb125-755" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">c</span>(interactions<span class="sc">$</span>from, interactions<span class="sc">$</span>to)))</span>
<span id="cb125-756"><a href="#cb125-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-757"><a href="#cb125-757" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with the mapped protein names to include original protein names as labels</span></span>
<span id="cb125-758"><a href="#cb125-758" aria-hidden="true" tabindex="-1"></a>nodes <span class="ot">&lt;-</span> <span class="fu">merge</span>(nodes, mapped[, <span class="fu">c</span>(<span class="st">"STRING_id"</span>, <span class="st">"query.term"</span>)], </span>
<span id="cb125-759"><a href="#cb125-759" aria-hidden="true" tabindex="-1"></a>               <span class="at">by.x =</span> <span class="st">"id"</span>, <span class="at">by.y =</span> <span class="st">"STRING_id"</span>, </span>
<span id="cb125-760"><a href="#cb125-760" aria-hidden="true" tabindex="-1"></a>               <span class="at">all.x =</span> <span class="cn">TRUE</span>, </span>
<span id="cb125-761"><a href="#cb125-761" aria-hidden="true" tabindex="-1"></a>               <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb125-762"><a href="#cb125-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-763"><a href="#cb125-763" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nodes)</span>
<span id="cb125-764"><a href="#cb125-764" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-765"><a href="#cb125-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-766"><a href="#cb125-766" aria-hidden="true" tabindex="-1"></a>Define edges for the network</span>
<span id="cb125-767"><a href="#cb125-767" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-770"><a href="#cb125-770" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-771"><a href="#cb125-771" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">source =</span> interactions<span class="sc">$</span>from, <span class="at">target =</span> interactions<span class="sc">$</span>to)</span>
<span id="cb125-772"><a href="#cb125-772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-773"><a href="#cb125-773" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove directionality of edges (necessary for various clusterings later)</span></span>
<span id="cb125-774"><a href="#cb125-774" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine and sort the columns, then get unique rows</span></span>
<span id="cb125-775"><a href="#cb125-775" aria-hidden="true" tabindex="-1"></a>unique_edges <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">t</span>(<span class="fu">apply</span>(edges, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">sort</span>(x))))</span>
<span id="cb125-776"><a href="#cb125-776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-777"><a href="#cb125-777" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert back to a dataframe</span></span>
<span id="cb125-778"><a href="#cb125-778" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(unique_edges, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-779"><a href="#cb125-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-780"><a href="#cb125-780" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the columns if needed</span></span>
<span id="cb125-781"><a href="#cb125-781" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"source"</span>, <span class="st">"target"</span>)</span>
<span id="cb125-782"><a href="#cb125-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-783"><a href="#cb125-783" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(edges)</span>
<span id="cb125-784"><a href="#cb125-784" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-785"><a href="#cb125-785" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-786"><a href="#cb125-786" aria-hidden="true" tabindex="-1"></a><span class="fu">## Make the network in Cytoscape</span></span>
<span id="cb125-787"><a href="#cb125-787" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-790"><a href="#cb125-790" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-791"><a href="#cb125-791" aria-hidden="true" tabindex="-1"></a><span class="co"># Connect to Cytoscape</span></span>
<span id="cb125-792"><a href="#cb125-792" aria-hidden="true" tabindex="-1"></a><span class="fu">cytoscapePing</span>()</span>
<span id="cb125-793"><a href="#cb125-793" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-794"><a href="#cb125-794" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a new Cytoscape network from your data</span></span>
<span id="cb125-795"><a href="#cb125-795" aria-hidden="true" tabindex="-1"></a><span class="fu">createNetworkFromDataFrames</span>(<span class="at">nodes =</span> nodes, </span>
<span id="cb125-796"><a href="#cb125-796" aria-hidden="true" tabindex="-1"></a>                            <span class="at">edges =</span> edges, </span>
<span id="cb125-797"><a href="#cb125-797" aria-hidden="true" tabindex="-1"></a>                            <span class="co"># title = mytitle, </span></span>
<span id="cb125-798"><a href="#cb125-798" aria-hidden="true" tabindex="-1"></a>                            <span class="at">collection =</span> <span class="st">"My Collection"</span>)</span>
<span id="cb125-799"><a href="#cb125-799" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-800"><a href="#cb125-800" aria-hidden="true" tabindex="-1"></a><span class="co"># Set node labels to the original protein names, and other visual tweaks</span></span>
<span id="cb125-801"><a href="#cb125-801" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeLabelMapping</span>(<span class="st">'query.term'</span>)</span>
<span id="cb125-802"><a href="#cb125-802" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeShapeDefault</span>(<span class="st">'ELLIPSE'</span>)</span>
<span id="cb125-803"><a href="#cb125-803" aria-hidden="true" tabindex="-1"></a><span class="fu">setNodeColorDefault</span>(<span class="st">'#9fbcda'</span>)</span>
<span id="cb125-804"><a href="#cb125-804" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-805"><a href="#cb125-805" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-806"><a href="#cb125-806" aria-hidden="true" tabindex="-1"></a>Now we modify the network layout in the app to reduce overlap of nodes and make it overall more aesthetically pleasing. Parameters will depend on the network </span>
<span id="cb125-807"><a href="#cb125-807" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-810"><a href="#cb125-810" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-811"><a href="#cb125-811" aria-hidden="true" tabindex="-1"></a><span class="co"># using force-directed layout. Higher coefficients means nodes are closer together. </span></span>
<span id="cb125-812"><a href="#cb125-812" aria-hidden="true" tabindex="-1"></a><span class="fu">layoutNetwork</span>(<span class="st">'force-directed defaultSpringCoefficient=0.000006 defaultSpringLength=1'</span>)</span>
<span id="cb125-813"><a href="#cb125-813" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-814"><a href="#cb125-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-815"><a href="#cb125-815" aria-hidden="true" tabindex="-1"></a><span class="fu">## Save and View the network </span></span>
<span id="cb125-816"><a href="#cb125-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-819"><a href="#cb125-819" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-820"><a href="#cb125-820" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure network fits in the frame to be saved </span></span>
<span id="cb125-821"><a href="#cb125-821" aria-hidden="true" tabindex="-1"></a><span class="fu">fitContent</span>()</span>
<span id="cb125-822"><a href="#cb125-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-823"><a href="#cb125-823" aria-hidden="true" tabindex="-1"></a><span class="co">#save network image</span></span>
<span id="cb125-824"><a href="#cb125-824" aria-hidden="true" tabindex="-1"></a><span class="fu">exportImage</span>(<span class="at">filename =</span> <span class="st">"currentnetwork.png"</span>) </span>
<span id="cb125-825"><a href="#cb125-825" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-826"><a href="#cb125-826" aria-hidden="true" tabindex="-1"></a><span class="co"># view network image</span></span>
<span id="cb125-827"><a href="#cb125-827" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"currentnetwork.png"</span>) </span>
<span id="cb125-828"><a href="#cb125-828" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-829"><a href="#cb125-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-830"><a href="#cb125-830" aria-hidden="true" tabindex="-1"></a><span class="fu">## Cluster PPI network </span></span>
<span id="cb125-831"><a href="#cb125-831" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-832"><a href="#cb125-832" aria-hidden="true" tabindex="-1"></a>commands specify: restoreEdges: restores edges after clustering, showUI: displays the new network, and undirectedEdges: assumes edges are undirected</span>
<span id="cb125-833"><a href="#cb125-833" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-834"><a href="#cb125-834" aria-hidden="true" tabindex="-1"></a><span class="fu">### GLay clustering </span></span>
<span id="cb125-835"><a href="#cb125-835" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-836"><a href="#cb125-836" aria-hidden="true" tabindex="-1"></a>Run GLay community clustering </span>
<span id="cb125-837"><a href="#cb125-837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-840"><a href="#cb125-840" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-841"><a href="#cb125-841" aria-hidden="true" tabindex="-1"></a><span class="co"># Run GLay community clustering</span></span>
<span id="cb125-842"><a href="#cb125-842" aria-hidden="true" tabindex="-1"></a>RCy3<span class="sc">::</span><span class="fu">commandsRun</span>(<span class="st">"cluster glay restoreEdges= true showUI = true undirectedEdges = true"</span>)</span>
<span id="cb125-843"><a href="#cb125-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-844"><a href="#cb125-844" aria-hidden="true" tabindex="-1"></a><span class="co"># reduce overlaps by making the nodes less inclined to be close to each other</span></span>
<span id="cb125-845"><a href="#cb125-845" aria-hidden="true" tabindex="-1"></a><span class="fu">layoutNetwork</span>(<span class="st">'force-directed defaultSpringCoefficient=0.000006 defaultSpringLength=1'</span>)</span>
<span id="cb125-846"><a href="#cb125-846" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-847"><a href="#cb125-847" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-848"><a href="#cb125-848" aria-hidden="true" tabindex="-1"></a>Visualize network after clustering </span>
<span id="cb125-849"><a href="#cb125-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-852"><a href="#cb125-852" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-853"><a href="#cb125-853" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure network fits in the frame to be saved </span></span>
<span id="cb125-854"><a href="#cb125-854" aria-hidden="true" tabindex="-1"></a><span class="fu">fitContent</span>()</span>
<span id="cb125-855"><a href="#cb125-855" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-856"><a href="#cb125-856" aria-hidden="true" tabindex="-1"></a><span class="co">#save network image</span></span>
<span id="cb125-857"><a href="#cb125-857" aria-hidden="true" tabindex="-1"></a><span class="fu">exportImage</span>(<span class="at">filename =</span> <span class="st">"currentnetwork_clustered.png"</span>) </span>
<span id="cb125-858"><a href="#cb125-858" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-859"><a href="#cb125-859" aria-hidden="true" tabindex="-1"></a><span class="co"># view network image</span></span>
<span id="cb125-860"><a href="#cb125-860" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"currentnetwork_clustered.png"</span>) </span>
<span id="cb125-861"><a href="#cb125-861" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-862"><a href="#cb125-862" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-863"><a href="#cb125-863" aria-hidden="true" tabindex="-1"></a><span class="fu">## Save network info </span></span>
<span id="cb125-864"><a href="#cb125-864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-867"><a href="#cb125-867" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-868"><a href="#cb125-868" aria-hidden="true" tabindex="-1"></a><span class="co"># Get the table with clustering results </span></span>
<span id="cb125-869"><a href="#cb125-869" aria-hidden="true" tabindex="-1"></a>network_table <span class="ot">&lt;-</span> <span class="fu">getTableColumns</span>()</span>
<span id="cb125-870"><a href="#cb125-870" aria-hidden="true" tabindex="-1"></a>clusterinfo <span class="ot">&lt;-</span> <span class="fu">table</span>(network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span>)</span>
<span id="cb125-871"><a href="#cb125-871" aria-hidden="true" tabindex="-1"></a>clusterinfo</span>
<span id="cb125-872"><a href="#cb125-872" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-873"><a href="#cb125-873" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-874"><a href="#cb125-874" aria-hidden="true" tabindex="-1"></a><span class="fu">## Members of each big cluster </span></span>
<span id="cb125-875"><a href="#cb125-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-878"><a href="#cb125-878" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-879"><a href="#cb125-879" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the members of clusters with &gt;5 nodes for subsequent analysis </span></span>
<span id="cb125-880"><a href="#cb125-880" aria-hidden="true" tabindex="-1"></a>bigclusters <span class="ot">&lt;-</span> <span class="fu">names</span>(clusterinfo)[(<span class="fu">table</span>(network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span>) <span class="sc">&gt;</span> <span class="dv">5</span>)]</span>
<span id="cb125-881"><a href="#cb125-881" aria-hidden="true" tabindex="-1"></a>bigclusternames <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"cluster"</span>, bigclusters)</span>
<span id="cb125-882"><a href="#cb125-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-883"><a href="#cb125-883" aria-hidden="true" tabindex="-1"></a><span class="co"># empty list of clusters to cycle through</span></span>
<span id="cb125-884"><a href="#cb125-884" aria-hidden="true" tabindex="-1"></a>clusterlist <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb125-885"><a href="#cb125-885" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-886"><a href="#cb125-886" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(bigclusters)){</span>
<span id="cb125-887"><a href="#cb125-887" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-888"><a href="#cb125-888" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create given clustername</span></span>
<span id="cb125-889"><a href="#cb125-889" aria-hidden="true" tabindex="-1"></a>  given_clustername <span class="ot">&lt;-</span> bigclusternames[i]</span>
<span id="cb125-890"><a href="#cb125-890" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-891"><a href="#cb125-891" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make a list of these lists </span></span>
<span id="cb125-892"><a href="#cb125-892" aria-hidden="true" tabindex="-1"></a>  clusterlist[[i]] <span class="ot">&lt;-</span> network_table<span class="sc">$</span>query.term[network_table<span class="sc">$</span><span class="st">`</span><span class="at">__glayCluster</span><span class="st">`</span> <span class="sc">==</span> bigclusters[i]]</span>
<span id="cb125-893"><a href="#cb125-893" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-894"><a href="#cb125-894" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-895"><a href="#cb125-895" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(clusterlist) <span class="ot">&lt;-</span> bigclusternames</span>
<span id="cb125-896"><a href="#cb125-896" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-897"><a href="#cb125-897" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-898"><a href="#cb125-898" aria-hidden="true" tabindex="-1"></a>Members of each cluster</span>
<span id="cb125-899"><a href="#cb125-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-902"><a href="#cb125-902" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-903"><a href="#cb125-903" aria-hidden="true" tabindex="-1"></a>clusterlist</span>
<span id="cb125-904"><a href="#cb125-904" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-905"><a href="#cb125-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-906"><a href="#cb125-906" aria-hidden="true" tabindex="-1"></a><span class="fu"># Enrichment analysis for each cluster (GO Biological Process)</span></span>
<span id="cb125-907"><a href="#cb125-907" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-910"><a href="#cb125-910" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-911"><a href="#cb125-911" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb125-912"><a href="#cb125-912" aria-hidden="true" tabindex="-1"></a>GO_BP_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb125-913"><a href="#cb125-913" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-914"><a href="#cb125-914" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb125-915"><a href="#cb125-915" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb125-916"><a href="#cb125-916" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb125-917"><a href="#cb125-917" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"BP"</span>,</span>
<span id="cb125-918"><a href="#cb125-918" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb125-919"><a href="#cb125-919" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb125-920"><a href="#cb125-920" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb125-921"><a href="#cb125-921" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-922"><a href="#cb125-922" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-923"><a href="#cb125-923" aria-hidden="true" tabindex="-1"></a>Create dotplots for results </span>
<span id="cb125-924"><a href="#cb125-924" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-927"><a href="#cb125-927" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-928"><a href="#cb125-928" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_BP_enrichments)){</span>
<span id="cb125-929"><a href="#cb125-929" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-930"><a href="#cb125-930" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb125-931"><a href="#cb125-931" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_BP_enrichments[[x]]</span>
<span id="cb125-932"><a href="#cb125-932" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-933"><a href="#cb125-933" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb125-934"><a href="#cb125-934" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb125-935"><a href="#cb125-935" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA BP for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb125-936"><a href="#cb125-936" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-937"><a href="#cb125-937" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb125-938"><a href="#cb125-938" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-939"><a href="#cb125-939" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-940"><a href="#cb125-940" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-941"><a href="#cb125-941" aria-hidden="true" tabindex="-1"></a><span class="fu">## Enrichment analysis for each cluster (GO Molecular Function)</span></span>
<span id="cb125-942"><a href="#cb125-942" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-945"><a href="#cb125-945" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-946"><a href="#cb125-946" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb125-947"><a href="#cb125-947" aria-hidden="true" tabindex="-1"></a>GO_MF_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb125-948"><a href="#cb125-948" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-949"><a href="#cb125-949" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb125-950"><a href="#cb125-950" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb125-951"><a href="#cb125-951" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb125-952"><a href="#cb125-952" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"MF"</span>,</span>
<span id="cb125-953"><a href="#cb125-953" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb125-954"><a href="#cb125-954" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb125-955"><a href="#cb125-955" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-956"><a href="#cb125-956" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb125-957"><a href="#cb125-957" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-958"><a href="#cb125-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-959"><a href="#cb125-959" aria-hidden="true" tabindex="-1"></a>Create dotplots for results </span>
<span id="cb125-960"><a href="#cb125-960" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-963"><a href="#cb125-963" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-964"><a href="#cb125-964" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_MF_enrichments)){</span>
<span id="cb125-965"><a href="#cb125-965" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-966"><a href="#cb125-966" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb125-967"><a href="#cb125-967" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_MF_enrichments[[x]]</span>
<span id="cb125-968"><a href="#cb125-968" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-969"><a href="#cb125-969" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb125-970"><a href="#cb125-970" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb125-971"><a href="#cb125-971" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA MF for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb125-972"><a href="#cb125-972" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-973"><a href="#cb125-973" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb125-974"><a href="#cb125-974" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-975"><a href="#cb125-975" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-976"><a href="#cb125-976" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-977"><a href="#cb125-977" aria-hidden="true" tabindex="-1"></a><span class="fu">## Enrichment analysis for each cluster (GO Cellular Component)</span></span>
<span id="cb125-978"><a href="#cb125-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-981"><a href="#cb125-981" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-982"><a href="#cb125-982" aria-hidden="true" tabindex="-1"></a><span class="co"># Run enrichment analysis on each cluster </span></span>
<span id="cb125-983"><a href="#cb125-983" aria-hidden="true" tabindex="-1"></a>GO_CC_enrichments <span class="ot">&lt;-</span> <span class="fu">lapply</span>(clusterlist, <span class="cf">function</span>(cluster) {</span>
<span id="cb125-984"><a href="#cb125-984" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-985"><a href="#cb125-985" aria-hidden="true" tabindex="-1"></a>  <span class="fu">enrichGO</span>(<span class="at">gene =</span> cluster,</span>
<span id="cb125-986"><a href="#cb125-986" aria-hidden="true" tabindex="-1"></a>           <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb125-987"><a href="#cb125-987" aria-hidden="true" tabindex="-1"></a>           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb125-988"><a href="#cb125-988" aria-hidden="true" tabindex="-1"></a>           <span class="at">ont =</span> <span class="st">"CC"</span>,</span>
<span id="cb125-989"><a href="#cb125-989" aria-hidden="true" tabindex="-1"></a>           <span class="at">pAdjustMethod =</span> <span class="st">"BH"</span>,</span>
<span id="cb125-990"><a href="#cb125-990" aria-hidden="true" tabindex="-1"></a>           <span class="at">qvalueCutoff =</span> <span class="fl">0.01</span>)</span>
<span id="cb125-991"><a href="#cb125-991" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb125-992"><a href="#cb125-992" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-993"><a href="#cb125-993" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-994"><a href="#cb125-994" aria-hidden="true" tabindex="-1"></a>Create dotplots for results </span>
<span id="cb125-995"><a href="#cb125-995" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-998"><a href="#cb125-998" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-999"><a href="#cb125-999" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(GO_CC_enrichments)){</span>
<span id="cb125-1000"><a href="#cb125-1000" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-1001"><a href="#cb125-1001" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pick out object with enrichments </span></span>
<span id="cb125-1002"><a href="#cb125-1002" aria-hidden="true" tabindex="-1"></a>  enrichresult <span class="ot">&lt;-</span> GO_CC_enrichments[[x]]</span>
<span id="cb125-1003"><a href="#cb125-1003" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-1004"><a href="#cb125-1004" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make dotplot object </span></span>
<span id="cb125-1005"><a href="#cb125-1005" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">dotplot</span>(<span class="at">object =</span> enrichresult) <span class="sc">+</span> </span>
<span id="cb125-1006"><a href="#cb125-1006" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">"ORA CC for "</span>, <span class="fu">names</span>(clusterlist)[x]))</span>
<span id="cb125-1007"><a href="#cb125-1007" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb125-1008"><a href="#cb125-1008" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p)</span>
<span id="cb125-1009"><a href="#cb125-1009" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-1010"><a href="#cb125-1010" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1011"><a href="#cb125-1011" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1012"><a href="#cb125-1012" aria-hidden="true" tabindex="-1"></a><span class="fu"># Random forest classification</span></span>
<span id="cb125-1013"><a href="#cb125-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1014"><a href="#cb125-1014" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prepare data</span></span>
<span id="cb125-1015"><a href="#cb125-1015" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1018"><a href="#cb125-1018" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1019"><a href="#cb125-1019" aria-hidden="true" tabindex="-1"></a><span class="co"># Load and prepare the data</span></span>
<span id="cb125-1020"><a href="#cb125-1020" aria-hidden="true" tabindex="-1"></a>data_raw <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb125-1021"><a href="#cb125-1021" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> data_raw <span class="sc">%&gt;%</span></span>
<span id="cb125-1022"><a href="#cb125-1022" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(GeneSymbol, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-1023"><a href="#cb125-1023" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>GeneID) <span class="sc">%&gt;%</span></span>
<span id="cb125-1024"><a href="#cb125-1024" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(GeneSymbol)) <span class="sc">%&gt;%</span></span>
<span id="cb125-1025"><a href="#cb125-1025" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"GeneSymbol"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-1026"><a href="#cb125-1026" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(<span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="dv">0</span>, .)) <span class="sc">%&gt;%</span></span>
<span id="cb125-1027"><a href="#cb125-1027" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_all</span>(as.integer)</span>
<span id="cb125-1028"><a href="#cb125-1028" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1029"><a href="#cb125-1029" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare metadata</span></span>
<span id="cb125-1030"><a href="#cb125-1030" aria-hidden="true" tabindex="-1"></a>metadata_raw <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb125-1031"><a href="#cb125-1031" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> metadata_raw <span class="sc">%&gt;%</span></span>
<span id="cb125-1032"><a href="#cb125-1032" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Index =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb125-1033"><a href="#cb125-1033" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"Index"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb125-1034"><a href="#cb125-1034" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">sample =</span> ID, <span class="at">disease =</span> Disease) <span class="sc">%&gt;%</span></span>
<span id="cb125-1035"><a href="#cb125-1035" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">gsub</span>(<span class="st">"_raw"</span>, <span class="st">""</span>, sample)) <span class="sc">%&gt;%</span></span>
<span id="cb125-1036"><a href="#cb125-1036" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, individual, disease) <span class="sc">%&gt;%</span></span>
<span id="cb125-1037"><a href="#cb125-1037" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">individual =</span> <span class="fu">factor</span>(individual), <span class="at">disease =</span> <span class="fu">factor</span>(disease))</span>
<span id="cb125-1038"><a href="#cb125-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1039"><a href="#cb125-1039" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure the data is in a matrix form with gene symbols as rownames</span></span>
<span id="cb125-1040"><a href="#cb125-1040" aria-hidden="true" tabindex="-1"></a>countData <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(data)</span>
<span id="cb125-1041"><a href="#cb125-1041" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1042"><a href="#cb125-1042" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare DESeqDataSet</span></span>
<span id="cb125-1043"><a href="#cb125-1043" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb125-1044"><a href="#cb125-1044" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> countData,</span>
<span id="cb125-1045"><a href="#cb125-1045" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata,</span>
<span id="cb125-1046"><a href="#cb125-1046" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> disease</span>
<span id="cb125-1047"><a href="#cb125-1047" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb125-1048"><a href="#cb125-1048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1049"><a href="#cb125-1049" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate size factors</span></span>
<span id="cb125-1050"><a href="#cb125-1050" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb125-1051"><a href="#cb125-1051" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1052"><a href="#cb125-1052" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate gene-wise dispersions</span></span>
<span id="cb125-1053"><a href="#cb125-1053" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateDispersionsGeneEst</span>(dds)</span>
<span id="cb125-1054"><a href="#cb125-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1055"><a href="#cb125-1055" aria-hidden="true" tabindex="-1"></a><span class="co"># Directly use the gene-wise estimates</span></span>
<span id="cb125-1056"><a href="#cb125-1056" aria-hidden="true" tabindex="-1"></a><span class="fu">dispersions</span>(dds) <span class="ot">&lt;-</span> <span class="fu">mcols</span>(dds)<span class="sc">$</span>dispGeneEst</span>
<span id="cb125-1057"><a href="#cb125-1057" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1058"><a href="#cb125-1058" aria-hidden="true" tabindex="-1"></a>normCounts <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-1059"><a href="#cb125-1059" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1060"><a href="#cb125-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1063"><a href="#cb125-1063" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1064"><a href="#cb125-1064" aria-hidden="true" tabindex="-1"></a><span class="co">#Define model fitting procedure</span></span>
<span id="cb125-1065"><a href="#cb125-1065" aria-hidden="true" tabindex="-1"></a><span class="co">#NB: you will also need the packages statmod and e1071 for this analysis</span></span>
<span id="cb125-1066"><a href="#cb125-1066" aria-hidden="true" tabindex="-1"></a><span class="co">#New score function</span></span>
<span id="cb125-1067"><a href="#cb125-1067" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>score <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y){</span>
<span id="cb125-1068"><a href="#cb125-1068" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(x) <span class="sc">/</span> <span class="fu">mean</span>(x)</span>
<span id="cb125-1069"><a href="#cb125-1069" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-1070"><a href="#cb125-1070" aria-hidden="true" tabindex="-1"></a><span class="co">#New filter function</span></span>
<span id="cb125-1071"><a href="#cb125-1071" aria-hidden="true" tabindex="-1"></a>rfSBF<span class="sc">$</span>filter <span class="ot">&lt;-</span> <span class="cf">function</span>(score, x, y){</span>
<span id="cb125-1072"><a href="#cb125-1072" aria-hidden="true" tabindex="-1"></a>meanLog2CPM <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">log2</span>(<span class="fu">cpm</span>(data) <span class="sc">+</span> <span class="dv">1</span>))</span>
<span id="cb125-1073"><a href="#cb125-1073" aria-hidden="true" tabindex="-1"></a>selection <span class="ot">&lt;-</span> score <span class="sc">&gt;</span> <span class="fu">quantile</span>(score, <span class="fl">0.9</span>) <span class="sc">&amp;</span> meanLog2CPM <span class="sc">&gt;</span> <span class="dv">1</span></span>
<span id="cb125-1074"><a href="#cb125-1074" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(score) <span class="sc">%in%</span> <span class="fu">names</span>(score)[selection]</span>
<span id="cb125-1075"><a href="#cb125-1075" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb125-1076"><a href="#cb125-1076" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1077"><a href="#cb125-1077" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1080"><a href="#cb125-1080" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1081"><a href="#cb125-1081" aria-hidden="true" tabindex="-1"></a><span class="co">#Create training set</span></span>
<span id="cb125-1082"><a href="#cb125-1082" aria-hidden="true" tabindex="-1"></a>trainingSet <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(<span class="fu">assay</span>(normCounts)))</span>
<span id="cb125-1083"><a href="#cb125-1083" aria-hidden="true" tabindex="-1"></a><span class="co">#Train and validate decision tree (C5.0)</span></span>
<span id="cb125-1084"><a href="#cb125-1084" aria-hidden="true" tabindex="-1"></a>dtControl <span class="ot">&lt;-</span> <span class="fu">sbfControl</span>(</span>
<span id="cb125-1085"><a href="#cb125-1085" aria-hidden="true" tabindex="-1"></a>  <span class="at">functions =</span> rfSBF,</span>
<span id="cb125-1086"><a href="#cb125-1086" aria-hidden="true" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">"loocv"</span>,</span>
<span id="cb125-1087"><a href="#cb125-1087" aria-hidden="true" tabindex="-1"></a>  <span class="at">saveDetails =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-1088"><a href="#cb125-1088" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb125-1089"><a href="#cb125-1089" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1090"><a href="#cb125-1090" aria-hidden="true" tabindex="-1"></a>dtModel <span class="ot">&lt;-</span> <span class="fu">sbf</span>(</span>
<span id="cb125-1091"><a href="#cb125-1091" aria-hidden="true" tabindex="-1"></a>  trainingSet,</span>
<span id="cb125-1092"><a href="#cb125-1092" aria-hidden="true" tabindex="-1"></a>  metadata<span class="sc">$</span>disease,</span>
<span id="cb125-1093"><a href="#cb125-1093" aria-hidden="true" tabindex="-1"></a>  <span class="at">sbfControl =</span> dtControl)</span>
<span id="cb125-1094"><a href="#cb125-1094" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1095"><a href="#cb125-1095" aria-hidden="true" tabindex="-1"></a><span class="co">#Show model</span></span>
<span id="cb125-1096"><a href="#cb125-1096" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(dtModel$fit)</span></span>
<span id="cb125-1097"><a href="#cb125-1097" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1098"><a href="#cb125-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1099"><a href="#cb125-1099" aria-hidden="true" tabindex="-1"></a><span class="fu">## Performance evaluation</span></span>
<span id="cb125-1100"><a href="#cb125-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1103"><a href="#cb125-1103" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1104"><a href="#cb125-1104" aria-hidden="true" tabindex="-1"></a><span class="co">#Print confusion matrix from validation</span></span>
<span id="cb125-1105"><a href="#cb125-1105" aria-hidden="true" tabindex="-1"></a><span class="fu">confusionMatrix</span>(dtModel<span class="sc">$</span>pred<span class="sc">$</span>predictions<span class="sc">$</span>pred, metadata<span class="sc">$</span>disease)</span>
<span id="cb125-1106"><a href="#cb125-1106" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1107"><a href="#cb125-1107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1108"><a href="#cb125-1108" aria-hidden="true" tabindex="-1"></a><span class="fu"># Spearman correlation analysis</span></span>
<span id="cb125-1109"><a href="#cb125-1109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1110"><a href="#cb125-1110" aria-hidden="true" tabindex="-1"></a>Loading the data for the correlaton analysis</span>
<span id="cb125-1111"><a href="#cb125-1111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1114"><a href="#cb125-1114" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1115"><a href="#cb125-1115" aria-hidden="true" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/voom_normalized.tsv"</span>)</span>
<span id="cb125-1116"><a href="#cb125-1116" aria-hidden="true" tabindex="-1"></a>de_results <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">"data/significant_genes.tsv"</span>)</span>
<span id="cb125-1117"><a href="#cb125-1117" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">"data/ad_metadata.xlsx"</span>)</span>
<span id="cb125-1118"><a href="#cb125-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1119"><a href="#cb125-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1120"><a href="#cb125-1120" aria-hidden="true" tabindex="-1"></a>Performing Spearman correlation of the 63 upregulated genes on selected clinical variables of interest and comparing the correlation between Alzheimer's disease and control  samples.</span>
<span id="cb125-1121"><a href="#cb125-1121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1122"><a href="#cb125-1122" aria-hidden="true" tabindex="-1"></a><span class="in">```{r , fig.width=20, fig.height=10}</span></span>
<span id="cb125-1123"><a href="#cb125-1123" aria-hidden="true" tabindex="-1"></a><span class="in"># Function to create correlation data and plot</span></span>
<span id="cb125-1124"><a href="#cb125-1124" aria-hidden="true" tabindex="-1"></a><span class="in">create_correlation_plot &lt;- function(data, metadata, gene_symbols, variable, title) {</span></span>
<span id="cb125-1125"><a href="#cb125-1125" aria-hidden="true" tabindex="-1"></a><span class="in">  correlation_data &lt;- data %&gt;%</span></span>
<span id="cb125-1126"><a href="#cb125-1126" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(GeneSymbol %in% gene_symbols) %&gt;%</span></span>
<span id="cb125-1127"><a href="#cb125-1127" aria-hidden="true" tabindex="-1"></a><span class="in">    pivot_longer(cols = -GeneSymbol, names_to = "ID", values_to = "value") %&gt;%</span></span>
<span id="cb125-1128"><a href="#cb125-1128" aria-hidden="true" tabindex="-1"></a><span class="in">    right_join(metadata, by = "ID") %&gt;%</span></span>
<span id="cb125-1129"><a href="#cb125-1129" aria-hidden="true" tabindex="-1"></a><span class="in">    group_by(GeneSymbol, Disease) %&gt;%</span></span>
<span id="cb125-1130"><a href="#cb125-1130" aria-hidden="true" tabindex="-1"></a><span class="in">    summarise(correlation = cor(value, !!rlang::sym(variable), method = "spearman"), .groups = "drop") %&gt;%</span></span>
<span id="cb125-1131"><a href="#cb125-1131" aria-hidden="true" tabindex="-1"></a><span class="in">    ungroup()</span></span>
<span id="cb125-1132"><a href="#cb125-1132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1133"><a href="#cb125-1133" aria-hidden="true" tabindex="-1"></a><span class="in">  # Extract the order of genes based on their correlation in AD</span></span>
<span id="cb125-1134"><a href="#cb125-1134" aria-hidden="true" tabindex="-1"></a><span class="in">  ad_correlations &lt;- correlation_data %&gt;%</span></span>
<span id="cb125-1135"><a href="#cb125-1135" aria-hidden="true" tabindex="-1"></a><span class="in">    filter(Disease == "AD") %&gt;%</span></span>
<span id="cb125-1136"><a href="#cb125-1136" aria-hidden="true" tabindex="-1"></a><span class="in">    arrange(desc(correlation)) %&gt;%</span></span>
<span id="cb125-1137"><a href="#cb125-1137" aria-hidden="true" tabindex="-1"></a><span class="in">    pull(GeneSymbol)</span></span>
<span id="cb125-1138"><a href="#cb125-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1139"><a href="#cb125-1139" aria-hidden="true" tabindex="-1"></a><span class="in">  # Ensure that all GeneSymbols are included, even those not present in AD data</span></span>
<span id="cb125-1140"><a href="#cb125-1140" aria-hidden="true" tabindex="-1"></a><span class="in">  full_gene_list &lt;- unique(correlation_data$GeneSymbol)</span></span>
<span id="cb125-1141"><a href="#cb125-1141" aria-hidden="true" tabindex="-1"></a><span class="in">  ad_order &lt;- unique(c(ad_correlations, full_gene_list)) # This ensures all genes are included and in order</span></span>
<span id="cb125-1142"><a href="#cb125-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1143"><a href="#cb125-1143" aria-hidden="true" tabindex="-1"></a><span class="in">  # Set the levels of GeneSymbol according to their order in AD</span></span>
<span id="cb125-1144"><a href="#cb125-1144" aria-hidden="true" tabindex="-1"></a><span class="in">  correlation_data$GeneSymbol &lt;- factor(correlation_data$GeneSymbol, levels = ad_order)</span></span>
<span id="cb125-1145"><a href="#cb125-1145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1146"><a href="#cb125-1146" aria-hidden="true" tabindex="-1"></a><span class="in">  # Plot the heatmap with genes ordered by their correlation in AD</span></span>
<span id="cb125-1147"><a href="#cb125-1147" aria-hidden="true" tabindex="-1"></a><span class="in">  heatmap_plot &lt;- ggplot(correlation_data, aes(x = Disease, y = GeneSymbol, fill = correlation)) +</span></span>
<span id="cb125-1148"><a href="#cb125-1148" aria-hidden="true" tabindex="-1"></a><span class="in">    geom_tile() +</span></span>
<span id="cb125-1149"><a href="#cb125-1149" aria-hidden="true" tabindex="-1"></a><span class="in">    scale_fill_gradient2(low = "blue", high = "red", mid = "white", midpoint = 0) +</span></span>
<span id="cb125-1150"><a href="#cb125-1150" aria-hidden="true" tabindex="-1"></a><span class="in">    theme_minimal() +</span></span>
<span id="cb125-1151"><a href="#cb125-1151" aria-hidden="true" tabindex="-1"></a><span class="in">    theme(axis.text.x = element_text(angle = 45, hjust = 1)) +</span></span>
<span id="cb125-1152"><a href="#cb125-1152" aria-hidden="true" tabindex="-1"></a><span class="in">    labs(fill = "Correlation", y = "Gene (ordered by AD correlation)", x = "Disease", title = title)</span></span>
<span id="cb125-1153"><a href="#cb125-1153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1154"><a href="#cb125-1154" aria-hidden="true" tabindex="-1"></a><span class="in">  return(heatmap_plot)</span></span>
<span id="cb125-1155"><a href="#cb125-1155" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb125-1156"><a href="#cb125-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1157"><a href="#cb125-1157" aria-hidden="true" tabindex="-1"></a><span class="in"># Variables of interest</span></span>
<span id="cb125-1158"><a href="#cb125-1158" aria-hidden="true" tabindex="-1"></a><span class="in">variables_of_interest &lt;- c("Age", "PMI", "Mean_plaque", "CDR")</span></span>
<span id="cb125-1159"><a href="#cb125-1159" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1160"><a href="#cb125-1160" aria-hidden="true" tabindex="-1"></a><span class="in"># Plot list</span></span>
<span id="cb125-1161"><a href="#cb125-1161" aria-hidden="true" tabindex="-1"></a><span class="in">plot_list &lt;- list()</span></span>
<span id="cb125-1162"><a href="#cb125-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1163"><a href="#cb125-1163" aria-hidden="true" tabindex="-1"></a><span class="in"># Loop over variables and create plots</span></span>
<span id="cb125-1164"><a href="#cb125-1164" aria-hidden="true" tabindex="-1"></a><span class="in">for (variable in variables_of_interest) {</span></span>
<span id="cb125-1165"><a href="#cb125-1165" aria-hidden="true" tabindex="-1"></a><span class="in">  plot_list[[variable]] &lt;- create_correlation_plot(data, metadata, de_results$GeneSymbol, variable, paste("Correlation with", variable))</span></span>
<span id="cb125-1166"><a href="#cb125-1166" aria-hidden="true" tabindex="-1"></a><span class="in">}</span></span>
<span id="cb125-1167"><a href="#cb125-1167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1168"><a href="#cb125-1168" aria-hidden="true" tabindex="-1"></a><span class="in"># Combine plots</span></span>
<span id="cb125-1169"><a href="#cb125-1169" aria-hidden="true" tabindex="-1"></a><span class="in">combined_plot &lt;- wrap_plots(plot_list, nrow = 1)</span></span>
<span id="cb125-1170"><a href="#cb125-1170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1171"><a href="#cb125-1171" aria-hidden="true" tabindex="-1"></a><span class="in"># Print the combined plot</span></span>
<span id="cb125-1172"><a href="#cb125-1172" aria-hidden="true" tabindex="-1"></a><span class="in">print(combined_plot)</span></span>
<span id="cb125-1173"><a href="#cb125-1173" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb125-1174"><a href="#cb125-1174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1175"><a href="#cb125-1175" aria-hidden="true" tabindex="-1"></a><span class="fu"># Final reflection (1/2 A4)</span></span>
<span id="cb125-1176"><a href="#cb125-1176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1177"><a href="#cb125-1177" aria-hidden="true" tabindex="-1"></a>Statistical challenges:</span>
<span id="cb125-1178"><a href="#cb125-1178" aria-hidden="true" tabindex="-1"></a>I think for our project there was a few challanges that we tried to overcome. The first one was when picking a completely new dataset unfamiliar to us. Even though some of the group members had worked on RNA-seq data before, it was a challenge for me to understand all the important steps during the pre-processing and considerations when analyzing. Since my background is in plasma proteomics, I found some similarities in the dataframe and how I could analyze the data. For instance, I was not used to the idea of count data because I normally work with either normalized protein expression (NPX) or medium florecence intensity (MFI). </span>
<span id="cb125-1179"><a href="#cb125-1179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1180"><a href="#cb125-1180" aria-hidden="true" tabindex="-1"></a>My largest contributions to this work was the attempt to classify Alzheimer's Disease (AD) using a leave-one-out (LOO) cross-validation (CV). In the short time limit that we had, I chose to apply a similar pipeline that was presented during the course. There was an issue with the vst() transformation funciton, which led me to perform some extra steps before I could apply the spelled out version of the function, which gave me a functional object from which I could proceed with this pipeline. In the end, the random forest classification gave an AUC of 0.41, which is worse than a random guess. There could have been several reasons behind this. On the one hand, since we may have picked the wrong data set or performed unlegitimate pre-processing, the data could have been more noisy. </span>
<span id="cb125-1181"><a href="#cb125-1181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1182"><a href="#cb125-1182" aria-hidden="true" tabindex="-1"></a>The other contribution was to perform Spearman correlation analysis of the significant upregulated genes from the differential expression analysis on the AD patients and the controls. Some biological relevancies was found like the higher correlation to CDR. However, it didn't make any sense to me that the top markers would have a negative correlation with age since age is a big factor behind AD, but this could be because of the noisy data.</span>
<span id="cb125-1183"><a href="#cb125-1183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1184"><a href="#cb125-1184" aria-hidden="true" tabindex="-1"></a>Final reflection:</span>
<span id="cb125-1185"><a href="#cb125-1185" aria-hidden="true" tabindex="-1"></a>In light of my own PhD project and what I have learnt from this course, I now feel more confortable applying these concept to my plasma proteomics pipelines and I now have more thoughts on how I could improve my current projects with methods such as canonical correlation analysis (CCA). My goal is to characterize the cardiovascular proteome and I think it will include some more multivariate analysis, therefore many of these concept I learnt during the course will be of great help.</span>
<span id="cb125-1186"><a href="#cb125-1186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1187"><a href="#cb125-1187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1188"><a href="#cb125-1188" aria-hidden="true" tabindex="-1"></a><span class="fu"># Session info</span></span>
<span id="cb125-1189"><a href="#cb125-1189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-1192"><a href="#cb125-1192" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb125-1193"><a href="#cb125-1193" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span>
<span id="cb125-1194"><a href="#cb125-1194" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




<script src="group2_final_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>